// sweetalert2-preset/dist/sweetalert2-preset.mjs
var D = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, kt = { exports: {} };
(function(h, O) {
  (function(l, P) {
    h.exports = P();
  })(D, function() {
    var l = { awaitingPromise: /* @__PURE__ */ new WeakMap(), promise: /* @__PURE__ */ new WeakMap(), innerParams: /* @__PURE__ */ new WeakMap(), domCache: /* @__PURE__ */ new WeakMap() }, P = (t) => {
      const e = {};
      for (const o in t)
        e[t[o]] = "swal2-" + t[o];
      return e;
    };
    const s = P(["container", "shown", "height-auto", "iosfix", "popup", "modal", "no-backdrop", "no-transition", "toast", "toast-shown", "show", "hide", "close", "title", "html-container", "actions", "confirm", "deny", "cancel", "default-outline", "footer", "icon", "icon-content", "image", "input", "file", "range", "select", "radio", "checkbox", "label", "textarea", "inputerror", "input-label", "validation-message", "progress-steps", "active-progress-step", "progress-step", "progress-step-line", "loader", "loading", "styled", "top", "top-start", "top-end", "top-left", "top-right", "center", "center-start", "center-end", "center-left", "center-right", "bottom", "bottom-start", "bottom-end", "bottom-left", "bottom-right", "grow-row", "grow-column", "grow-fullscreen", "rtl", "timer-progress-bar", "timer-progress-bar-container", "scrollbar-measure", "icon-success", "icon-warning", "icon-info", "icon-question", "icon-error", "no-war"]), R = P(["success", "warning", "info", "question", "error"]), ze = "SweetAlert2:", we = (t) => t.charAt(0).toUpperCase() + t.slice(1), x = (t) => {
      console.warn("".concat(ze, " ").concat(typeof t == "object" ? t.join(" ") : t));
    }, I = (t) => {
      console.error("".concat(ze, " ").concat(t));
    }, Me = [], xt = (t, e) => {
      t = '"'.concat(t, '" is deprecated and will be removed in the next major release. Please use "').concat(e, '" instead.'), Me.includes(t) || (Me.push(t), x(t));
    }, ne = (t) => typeof t == "function" ? t() : t, ge = (t) => t && typeof t.toPromise == "function", X = (t) => ge(t) ? t.toPromise() : Promise.resolve(t), fe = (t) => t && Promise.resolve(t) === t, y = () => document.body.querySelector(".".concat(s.container)), G = (t) => {
      const e = y();
      return e ? e.querySelector(t) : null;
    }, T = (t) => G(".".concat(t)), g = () => T(s.popup), J = () => T(s.icon), He = () => T(s.title), se = () => T(s["html-container"]), qe = () => T(s.image), De = () => T(s["progress-steps"]), re = () => T(s["validation-message"]), S = () => G(".".concat(s.actions, " .").concat(s.confirm)), H = () => G(".".concat(s.actions, " .").concat(s.deny)), Y = () => G(".".concat(s.loader)), V = () => G(".".concat(s.actions, " .").concat(s.cancel)), Q = () => T(s.actions), Ie = () => T(s.footer), ie = () => T(s["timer-progress-bar"]), he = () => T(s.close), be = () => {
      const t = Array.from(g().querySelectorAll('[tabindex]:not([tabindex="-1"]):not([tabindex="0"])')).sort((o, a) => (o = parseInt(o.getAttribute("tabindex")), a = parseInt(a.getAttribute("tabindex")), a < o ? 1 : o < a ? -1 : 0));
      var e = Array.from(g().querySelectorAll(`
  a[href],
  area[href],
  input:not([disabled]),
  select:not([disabled]),
  textarea:not([disabled]),
  button:not([disabled]),
  iframe,
  object,
  embed,
  [tabindex="0"],
  [contenteditable],
  audio[controls],
  video[controls],
  summary
`)).filter((o) => o.getAttribute("tabindex") !== "-1");
      return ((o) => {
        const a = [];
        for (let n = 0; n < o.length; n++)
          a.indexOf(o[n]) === -1 && a.push(o[n]);
        return a;
      })(t.concat(e)).filter((o) => E(o));
    }, ye = () => M(document.body, s.shown) && !M(document.body, s["toast-shown"]) && !M(document.body, s["no-backdrop"]), le = () => g() && M(g(), s.toast);
    function ve(t) {
      var e = 1 < arguments.length && arguments[1] !== void 0 && arguments[1];
      const o = ie();
      E(o) && (e && (o.style.transition = "none", o.style.width = "100%"), setTimeout(() => {
        o.style.transition = "width ".concat(t / 1e3, "s linear"), o.style.width = "0%";
      }, 10));
    }
    const F = { previousBodyPadding: null }, C = (t, e) => {
      if (t.textContent = "", e) {
        const o = new DOMParser(), a = o.parseFromString(e, "text/html");
        Array.from(a.querySelector("head").childNodes).forEach((n) => {
          t.appendChild(n);
        }), Array.from(a.querySelector("body").childNodes).forEach((n) => {
          t.appendChild(n);
        });
      }
    }, M = (t, e) => {
      if (!e)
        return !1;
      var o = e.split(/\s+/);
      for (let a = 0; a < o.length; a++)
        if (!t.classList.contains(o[a]))
          return !1;
      return !0;
    }, Ct = (t, e) => {
      Array.from(t.classList).forEach((o) => {
        Object.values(s).includes(o) || Object.values(R).includes(o) || Object.values(e.showClass).includes(o) || t.classList.remove(o);
      });
    }, L = (t, e, o) => {
      if (Ct(t, e), e.customClass && e.customClass[o]) {
        if (typeof e.customClass[o] != "string" && !e.customClass[o].forEach)
          return x("Invalid type of customClass.".concat(o, '! Expected string or iterable object, got "').concat(typeof e.customClass[o], '"'));
        w(t, e.customClass[o]);
      }
    }, ke = (t, e) => {
      if (!e)
        return null;
      switch (e) {
        case "select":
        case "textarea":
        case "file":
          return t.querySelector(".".concat(s.popup, " > .").concat(s[e]));
        case "checkbox":
          return t.querySelector(".".concat(s.popup, " > .").concat(s.checkbox, " input"));
        case "radio":
          return t.querySelector(".".concat(s.popup, " > .").concat(s.radio, " input:checked")) || t.querySelector(".".concat(s.popup, " > .").concat(s.radio, " input:first-child"));
        case "range":
          return t.querySelector(".".concat(s.popup, " > .").concat(s.range, " input"));
        default:
          return t.querySelector(".".concat(s.popup, " > .").concat(s.input));
      }
    }, Ve = (t) => {
      var e;
      t.focus(), t.type !== "file" && (e = t.value, t.value = "", t.value = e);
    }, Ne = (t, e, o) => {
      t && e && (e = typeof e == "string" ? e.split(/\s+/).filter(Boolean) : e).forEach((a) => {
        Array.isArray(t) ? t.forEach((n) => {
          o ? n.classList.add(a) : n.classList.remove(a);
        }) : o ? t.classList.add(a) : t.classList.remove(a);
      });
    }, w = (t, e) => {
      Ne(t, e, !0);
    }, j = (t, e) => {
      Ne(t, e, !1);
    }, q = (t, e) => {
      var o = Array.from(t.children);
      for (let n = 0; n < o.length; n++) {
        var a = o[n];
        if (a instanceof HTMLElement && M(a, e))
          return a;
      }
    }, U = (t, e, o) => {
      (o = o === "".concat(parseInt(o)) ? parseInt(o) : o) || parseInt(o) === 0 ? t.style[e] = typeof o == "number" ? "".concat(o, "px") : o : t.style.removeProperty(e);
    }, b = function(t) {
      t.style.display = 1 < arguments.length && arguments[1] !== void 0 ? arguments[1] : "flex";
    }, v = (t) => {
      t.style.display = "none";
    }, Ze = (t, e, o, a) => {
      const n = t.querySelector(e);
      n && (n.style[o] = a);
    }, ce = function(t, e) {
      var o = 2 < arguments.length && arguments[2] !== void 0 ? arguments[2] : "flex";
      e ? b(t, o) : v(t);
    }, E = (t) => !(!t || !(t.offsetWidth || t.offsetHeight || t.getClientRects().length)), At = () => !E(S()) && !E(H()) && !E(V()), Re = (t) => t.scrollHeight > t.clientHeight, Ye = (o) => {
      const e = window.getComputedStyle(o);
      var o = parseFloat(e.getPropertyValue("animation-duration") || "0"), a = parseFloat(e.getPropertyValue("transition-duration") || "0");
      return 0 < o || 0 < a;
    }, Bt = 100, m = {}, Pt = () => {
      m.previousActiveElement instanceof HTMLElement ? (m.previousActiveElement.focus(), m.previousActiveElement = null) : document.body && document.body.focus();
    }, Et = (t) => new Promise((e) => {
      if (!t)
        return e();
      var o = window.scrollX, a = window.scrollY;
      m.restoreFocusTimeout = setTimeout(() => {
        Pt(), e();
      }, Bt), window.scrollTo(o, a);
    }), Fe = () => typeof window > "u" || typeof document > "u", Tt = `
 <div aria-labelledby="`.concat(s.title, '" aria-describedby="').concat(s["html-container"], '" class="').concat(s.popup, `" tabindex="-1">
   <button type="button" class="`).concat(s.close, `"></button>
   <ul class="`).concat(s["progress-steps"], `"></ul>
   <div class="`).concat(s.icon, `"></div>
   <img class="`).concat(s.image, `" />
   <h2 class="`).concat(s.title, '" id="').concat(s.title, `"></h2>
   <div class="`).concat(s["html-container"], '" id="').concat(s["html-container"], `"></div>
   <input class="`).concat(s.input, `" />
   <input type="file" class="`).concat(s.file, `" />
   <div class="`).concat(s.range, `">
     <input type="range" />
     <output></output>
   </div>
   <select class="`).concat(s.select, `"></select>
   <div class="`).concat(s.radio, `"></div>
   <label for="`).concat(s.checkbox, '" class="').concat(s.checkbox, `">
     <input type="checkbox" />
     <span class="`).concat(s.label, `"></span>
   </label>
   <textarea class="`).concat(s.textarea, `"></textarea>
   <div class="`).concat(s["validation-message"], '" id="').concat(s["validation-message"], `"></div>
   <div class="`).concat(s.actions, `">
     <div class="`).concat(s.loader, `"></div>
     <button type="button" class="`).concat(s.confirm, `"></button>
     <button type="button" class="`).concat(s.deny, `"></button>
     <button type="button" class="`).concat(s.cancel, `"></button>
   </div>
   <div class="`).concat(s.footer, `"></div>
   <div class="`).concat(s["timer-progress-bar-container"], `">
     <div class="`).concat(s["timer-progress-bar"], `"></div>
   </div>
 </div>
`).replace(/(^|\n)\s*/g, ""), Lt = () => {
      const t = y();
      return !!t && (t.remove(), j([document.documentElement, document.body], [s["no-backdrop"], s["toast-shown"], s["has-column"]]), !0);
    }, N = () => {
      m.currentInstance.resetValidationMessage();
    }, St = () => {
      const t = g(), e = q(t, s.input), o = q(t, s.file), a = t.querySelector(".".concat(s.range, " input")), n = t.querySelector(".".concat(s.range, " output")), r = q(t, s.select), c = t.querySelector(".".concat(s.checkbox, " input")), i = q(t, s.textarea);
      e.oninput = N, o.onchange = N, r.onchange = N, c.onchange = N, i.oninput = N, a.oninput = () => {
        N(), n.value = a.value;
      }, a.onchange = () => {
        N(), n.value = a.value;
      };
    }, Ot = (t) => typeof t == "string" ? document.querySelector(t) : t, jt = (t) => {
      const e = g();
      e.setAttribute("role", t.toast ? "alert" : "dialog"), e.setAttribute("aria-live", t.toast ? "polite" : "assertive"), t.toast || e.setAttribute("aria-modal", "true");
    }, zt = (t) => {
      window.getComputedStyle(t).direction === "rtl" && w(y(), s.rtl);
    }, xe = (t, e) => {
      if (t instanceof HTMLElement)
        e.appendChild(t);
      else if (typeof t == "object") {
        var o = t, a = e;
        o.jquery ? Mt(a, o) : C(a, o.toString());
      } else
        t && C(e, t);
    }, Mt = (t, e) => {
      if (t.textContent = "", 0 in e)
        for (let o = 0; o in e; o++)
          t.appendChild(e[o].cloneNode(!0));
      else
        t.appendChild(e.cloneNode(!0));
    }, ee = (() => {
      if (!Fe()) {
        var t = document.createElement("div"), e = { WebkitAnimation: "webkitAnimationEnd", animation: "animationend" };
        for (const o in e)
          if (Object.prototype.hasOwnProperty.call(e, o) && t.style[o] !== void 0)
            return e[o];
      }
      return !1;
    })(), Ht = (t, e) => {
      var o, a, n, r, c, i = Q(), u = Y();
      (e.showConfirmButton || e.showDenyButton || e.showCancelButton ? b : v)(i), L(i, e, "actions"), i = i, o = u, a = e, n = S(), r = H(), c = V(), Ce(n, "confirm", a), Ce(r, "deny", a), Ce(c, "cancel", a), function(p, d, f, B) {
        if (!B.buttonsStyling)
          return j([p, d, f], s.styled);
        w([p, d, f], s.styled), B.confirmButtonColor && (p.style.backgroundColor = B.confirmButtonColor, w(p, s["default-outline"])), B.denyButtonColor && (d.style.backgroundColor = B.denyButtonColor, w(d, s["default-outline"])), B.cancelButtonColor && (f.style.backgroundColor = B.cancelButtonColor, w(f, s["default-outline"]));
      }(n, r, c, a), a.reverseButtons && (a.toast ? (i.insertBefore(c, n), i.insertBefore(r, n)) : (i.insertBefore(c, o), i.insertBefore(r, o), i.insertBefore(n, o))), C(u, e.loaderHtml), L(u, e, "loader");
    };
    function Ce(t, e, o) {
      ce(t, o["show".concat(we(e), "Button")], "inline-block"), C(t, o["".concat(e, "ButtonText")]), t.setAttribute("aria-label", o["".concat(e, "ButtonAriaLabel")]), t.className = s[e], L(t, o, "".concat(e, "Button")), w(t, o["".concat(e, "ButtonClass")]);
    }
    const qt = (t, e) => {
      const o = he();
      C(o, e.closeButtonHtml), L(o, e, "closeButton"), ce(o, e.showCloseButton), o.setAttribute("aria-label", e.closeButtonAriaLabel);
    }, Dt = (t, e) => {
      var o, a, n = y();
      n && (a = n, typeof (o = e.backdrop) == "string" ? a.style.background = o : o || w([document.documentElement, document.body], s["no-backdrop"]), a = n, (o = e.position) in s ? w(a, s[o]) : (x('The "position" parameter is not valid, defaulting to "center"'), w(a, s.center)), o = n, (a = e.grow) && typeof a == "string" && (a = "grow-".concat(a)) in s && w(o, s[a]), L(n, e, "container"));
    }, It = ["input", "file", "range", "select", "radio", "checkbox", "textarea"], Vt = (n, e) => {
      const o = g();
      var a, n = l.innerParams.get(n);
      const r = !n || e.input !== n.input;
      It.forEach((c) => {
        const i = q(o, s[c]);
        {
          var u = c, p = e.inputAttributes;
          const d = ke(g(), u);
          if (d) {
            Nt(d);
            for (const f in p)
              d.setAttribute(f, p[f]);
          }
        }
        i.className = s[c], r && v(i);
      }), e.input && (r && ((c) => {
        if (!A[c.input])
          return I('Unexpected type of input! Expected "text", "email", "password", "number", "tel", "select", "radio", "checkbox", "textarea", "file" or "url", got "'.concat(c.input, '"'));
        const i = Ue(c.input), u = A[c.input](i, c);
        b(i), setTimeout(() => {
          Ve(u);
        });
      })(e), n = e, a = Ue(n.input), typeof n.customClass == "object" && w(a, n.customClass.input));
    }, Nt = (t) => {
      for (let o = 0; o < t.attributes.length; o++) {
        var e = t.attributes[o].name;
        ["type", "value", "style"].includes(e) || t.removeAttribute(e);
      }
    }, Ae = (t, e) => {
      t.placeholder && !e.inputPlaceholder || (t.placeholder = e.inputPlaceholder);
    }, te = (t, e, o) => {
      if (o.inputLabel) {
        t.id = s.input;
        const n = document.createElement("label");
        var a = s["input-label"];
        n.setAttribute("for", t.id), n.className = a, typeof o.customClass == "object" && w(n, o.customClass.inputLabel), n.innerText = o.inputLabel, e.insertAdjacentElement("beforebegin", n);
      }
    }, Ue = (t) => q(g(), s[t] || s.input), de = (t, e) => {
      ["string", "number"].includes(typeof e) ? t.value = "".concat(e) : fe(e) || x('Unexpected type of inputValue! Expected "string", "number" or "Promise", got "'.concat(typeof e, '"'));
    }, A = {}, Zt = (A.text = A.email = A.password = A.number = A.tel = A.url = (t, e) => (de(t, e.inputValue), te(t, t, e), Ae(t, e), t.type = e.input, t), A.file = (t, e) => (te(t, t, e), Ae(t, e), t), A.range = (t, e) => {
      const o = t.querySelector("input");
      var a = t.querySelector("output");
      return de(o, e.inputValue), o.type = e.input, de(a, e.inputValue), te(o, t, e), t;
    }, A.select = (t, e) => {
      if (t.textContent = "", e.inputPlaceholder) {
        const o = document.createElement("option");
        C(o, e.inputPlaceholder), o.value = "", o.disabled = !0, o.selected = !0, t.appendChild(o);
      }
      return te(t, t, e), t;
    }, A.radio = (t) => (t.textContent = "", t), A.checkbox = (t, e) => {
      const o = ke(g(), "checkbox");
      return o.value = "1", o.id = s.checkbox, o.checked = Boolean(e.inputValue), t = t.querySelector("span"), C(t, e.inputPlaceholder), o;
    }, A.textarea = (t, e) => (de(t, e.inputValue), Ae(t, e), te(t, t, e), setTimeout(() => {
      if ("MutationObserver" in window) {
        const o = parseInt(window.getComputedStyle(g()).width);
        new MutationObserver(() => {
          var a = t.offsetWidth + (a = t, parseInt(window.getComputedStyle(a).marginLeft) + parseInt(window.getComputedStyle(a).marginRight));
          a > o ? g().style.width = "".concat(a, "px") : g().style.width = null;
        }).observe(t, { attributes: !0, attributeFilter: ["style"] });
      }
    }), t), (t, e) => {
      const o = se();
      L(o, e, "htmlContainer"), e.html ? (xe(e.html, o), b(o, "block")) : e.text ? (o.textContent = e.text, b(o, "block")) : v(o), Vt(t, e);
    }), Rt = (t, e) => {
      var o = Ie();
      ce(o, e.footer), e.footer && xe(e.footer, o), L(o, e, "footer");
    }, Yt = (o, e) => {
      var o = l.innerParams.get(o), a = J();
      o && e.icon === o.icon ? ($e(a, e), We(a, e)) : e.icon || e.iconHtml ? e.icon && Object.keys(R).indexOf(e.icon) === -1 ? (I('Unknown icon! Expected "success", "error", "warning", "info" or "question", got "'.concat(e.icon, '"')), v(a)) : (b(a), $e(a, e), We(a, e), w(a, e.showClass.icon)) : v(a);
    }, We = (t, e) => {
      for (const o in R)
        e.icon !== o && j(t, R[o]);
      w(t, R[e.icon]), $t(t, e), Ft(), L(t, e, "icon");
    }, Ft = () => {
      const t = g();
      var e = window.getComputedStyle(t).getPropertyValue("background-color");
      const o = t.querySelectorAll("[class^=swal2-success-circular-line], .swal2-success-fix");
      for (let a = 0; a < o.length; a++)
        o[a].style.backgroundColor = e;
    }, Ut = `
  <div class="swal2-success-circular-line-left"></div>
  <span class="swal2-success-line-tip"></span> <span class="swal2-success-line-long"></span>
  <div class="swal2-success-ring"></div> <div class="swal2-success-fix"></div>
  <div class="swal2-success-circular-line-right"></div>
`, Wt = `
  <span class="swal2-x-mark">
    <span class="swal2-x-mark-line-left"></span>
    <span class="swal2-x-mark-line-right"></span>
  </span>
`, $e = (t, e) => {
      let o = t.innerHTML, a;
      var n;
      e.iconHtml ? a = _e(e.iconHtml) : e.icon === "success" ? (a = Ut, o = o.replace(/ style=".*?"/g, "")) : a = e.icon === "error" ? Wt : (n = { question: "?", warning: "!", info: "i" }, _e(n[e.icon])), o.trim() !== a.trim() && C(t, a);
    }, $t = (t, e) => {
      if (e.iconColor) {
        t.style.color = e.iconColor, t.style.borderColor = e.iconColor;
        for (const o of [".swal2-success-line-tip", ".swal2-success-line-long", ".swal2-x-mark-line-left", ".swal2-x-mark-line-right"])
          Ze(t, o, "backgroundColor", e.iconColor);
        Ze(t, ".swal2-success-ring", "borderColor", e.iconColor);
      }
    }, _e = (t) => '<div class="'.concat(s["icon-content"], '">').concat(t, "</div>"), _t = (t, e) => {
      const o = qe();
      if (!e.imageUrl)
        return v(o);
      b(o, ""), o.setAttribute("src", e.imageUrl), o.setAttribute("alt", e.imageAlt), U(o, "width", e.imageWidth), U(o, "height", e.imageHeight), o.className = s.image, L(o, e, "image");
    }, Kt = (t, e) => {
      var o = y();
      const a = g();
      e.toast ? (U(o, "width", e.width), a.style.width = "100%", a.insertBefore(Y(), J())) : U(a, "width", e.width), U(a, "padding", e.padding), e.color && (a.style.color = e.color), e.background && (a.style.background = e.background), v(re()), o = a, o.className = "".concat(s.popup, " ").concat(E(o) ? e.showClass.popup : ""), e.toast ? (w([document.documentElement, document.body], s["toast-shown"]), w(o, s.toast)) : w(o, s.modal), L(o, e, "popup"), typeof e.customClass == "string" && w(o, e.customClass), e.icon && w(o, s["icon-".concat(e.icon)]);
    }, Xt = (t, e) => {
      const o = De();
      if (!e.progressSteps || e.progressSteps.length === 0)
        return v(o);
      b(o), o.textContent = "", e.currentProgressStep >= e.progressSteps.length && x("Invalid currentProgressStep parameter, it should be less than progressSteps.length (currentProgressStep like JS arrays starts from 0)"), e.progressSteps.forEach((r, n) => {
        var r = ((c) => {
          const i = document.createElement("li");
          return w(i, s["progress-step"]), C(i, c), i;
        })(r);
        o.appendChild(r), n === e.currentProgressStep && w(r, s["active-progress-step"]), n !== e.progressSteps.length - 1 && (r = ((c) => {
          const i = document.createElement("li");
          return w(i, s["progress-step-line"]), c.progressStepsDistance && U(i, "width", c.progressStepsDistance), i;
        })(e), o.appendChild(r));
      });
    }, Gt = (t, e) => {
      const o = He();
      ce(o, e.title || e.titleText, "block"), e.title && xe(e.title, o), e.titleText && (o.innerText = e.titleText), L(o, e, "title");
    }, Ke = (t, e) => {
      Kt(t, e), Dt(t, e), Xt(t, e), Yt(t, e), _t(t, e), Gt(t, e), qt(t, e), Zt(t, e), Ht(t, e), Rt(t, e), typeof e.didRender == "function" && e.didRender(g());
    };
    function Xe() {
      var t, e = l.innerParams.get(this);
      if (e) {
        const o = l.domCache.get(this);
        v(o.loader), le() ? e.icon && b(J()) : (e = o, (t = e.popup.getElementsByClassName(e.loader.getAttribute("data-button-to-replace"))).length ? b(t[0], "inline-block") : At() && v(e.actions)), j([o.popup, o.actions], s.loading), o.popup.removeAttribute("aria-busy"), o.popup.removeAttribute("data-loading"), o.confirmButton.disabled = !1, o.denyButton.disabled = !1, o.cancelButton.disabled = !1;
      }
    }
    const Ge = () => S() && S().click(), W = Object.freeze({ cancel: "cancel", backdrop: "backdrop", close: "close", esc: "esc", timer: "timer" }), Je = (t) => {
      t.keydownTarget && t.keydownHandlerAdded && (t.keydownTarget.removeEventListener("keydown", t.keydownHandler, { capture: t.keydownListenerCapture }), t.keydownHandlerAdded = !1);
    }, Be = (t, e, o) => {
      const a = be();
      if (a.length)
        return (e += o) === a.length ? e = 0 : e === -1 && (e = a.length - 1), a[e].focus();
      g().focus();
    }, Qe = ["ArrowRight", "ArrowDown"], Jt = ["ArrowLeft", "ArrowUp"], Qt = (t, e, o) => {
      var a = l.innerParams.get(t);
      if (a && !e.isComposing && e.keyCode !== 229)
        if (a.stopKeydownPropagation && e.stopPropagation(), e.key === "Enter")
          t = t, i = e, n = a, ne(n.allowEnterKey) && i.target && t.getInput() && i.target instanceof HTMLElement && i.target.outerHTML === t.getInput().outerHTML && (["textarea", "file"].includes(n.input) || (Ge(), i.preventDefault()));
        else if (e.key === "Tab") {
          t = e;
          var n = a, r = t.target, c = be();
          let p = -1;
          for (let d = 0; d < c.length; d++)
            if (r === c[d]) {
              p = d;
              break;
            }
          t.shiftKey ? Be(n, p, -1) : Be(n, p, 1), t.stopPropagation(), t.preventDefault();
        } else if ([...Qe, ...Jt].includes(e.key)) {
          var i = e.key;
          const p = S(), d = H(), f = V();
          if (!(document.activeElement instanceof HTMLElement) || [p, d, f].includes(document.activeElement)) {
            var u = Qe.includes(i) ? "nextElementSibling" : "previousElementSibling";
            let B = document.activeElement;
            for (let vt = 0; vt < Q().children.length; vt++) {
              if (!(B = B[u]))
                return;
              if (B instanceof HTMLButtonElement && E(B))
                break;
            }
            B instanceof HTMLButtonElement && B.focus();
          }
        } else
          e.key === "Escape" && (t = e, e = a, a = o, ne(e.allowEscapeKey) && (t.preventDefault(), a(W.esc)));
    };
    var oe = { swalPromiseResolve: /* @__PURE__ */ new WeakMap(), swalPromiseReject: /* @__PURE__ */ new WeakMap() };
    const eo = () => {
      Array.from(document.body.children).forEach((e) => {
        e === y() || e.contains(y()) || (e.hasAttribute("aria-hidden") && e.setAttribute("data-previous-aria-hidden", e.getAttribute("aria-hidden")), e.setAttribute("aria-hidden", "true"));
      });
    }, et = () => {
      Array.from(document.body.children).forEach((e) => {
        e.hasAttribute("data-previous-aria-hidden") ? (e.setAttribute("aria-hidden", e.getAttribute("data-previous-aria-hidden")), e.removeAttribute("data-previous-aria-hidden")) : e.removeAttribute("aria-hidden");
      });
    }, to = () => {
      if ((/iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream || navigator.platform === "MacIntel" && 1 < navigator.maxTouchPoints) && !M(document.body, s.iosfix)) {
        var t = document.body.scrollTop;
        document.body.style.top = "".concat(-1 * t, "px"), w(document.body, s.iosfix);
        {
          const e = y();
          let o;
          e.ontouchstart = (a) => {
            o = oo(a);
          }, e.ontouchmove = (a) => {
            o && (a.preventDefault(), a.stopPropagation());
          };
        }
        {
          const e = navigator.userAgent, o = !!e.match(/iPad/i) || !!e.match(/iPhone/i), a = !!e.match(/WebKit/i);
          o && a && !e.match(/CriOS/i) && g().scrollHeight > window.innerHeight - 44 && (y().style.paddingBottom = "".concat(44, "px"));
        }
      }
    }, oo = (t) => {
      var e, o = t.target, a = y();
      return !((e = t).touches && e.touches.length && e.touches[0].touchType === "stylus" || (e = t).touches && 1 < e.touches.length) && (o === a || !Re(a) && o instanceof HTMLElement && o.tagName !== "INPUT" && o.tagName !== "TEXTAREA" && (!Re(se()) || !se().contains(o)));
    }, ao = () => {
      var t;
      M(document.body, s.iosfix) && (t = parseInt(document.body.style.top, 10), j(document.body, s.iosfix), document.body.style.top = "", document.body.scrollTop = -1 * t);
    }, no = () => {
      F.previousBodyPadding === null && document.body.scrollHeight > window.innerHeight && (F.previousBodyPadding = parseInt(window.getComputedStyle(document.body).getPropertyValue("padding-right")), document.body.style.paddingRight = "".concat(F.previousBodyPadding + (() => {
        const t = document.createElement("div");
        t.className = s["scrollbar-measure"], document.body.appendChild(t);
        var e = t.getBoundingClientRect().width - t.clientWidth;
        return document.body.removeChild(t), e;
      })(), "px"));
    }, so = () => {
      F.previousBodyPadding !== null && (document.body.style.paddingRight = "".concat(F.previousBodyPadding, "px"), F.previousBodyPadding = null);
    };
    function tt(t, e, o, a) {
      le() ? ot(t, a) : (Et(o).then(() => ot(t, a)), Je(m)), /^((?!chrome|android).)*safari/i.test(navigator.userAgent) ? (e.setAttribute("style", "display:none !important"), e.removeAttribute("class"), e.innerHTML = "") : e.remove(), ye() && (so(), ao(), et()), j([document.documentElement, document.body], [s.shown, s["height-auto"], s["no-backdrop"], s["toast-shown"]]);
    }
    function ue(t) {
      t = (o = t) !== void 0 ? Object.assign({ isConfirmed: !1, isDenied: !1, isDismissed: !1 }, o) : { isConfirmed: !1, isDenied: !1, isDismissed: !0 };
      const e = oe.swalPromiseResolve.get(this);
      var o = ((a) => {
        const n = g();
        if (!n)
          return !1;
        const r = l.innerParams.get(a);
        if (!r || M(n, r.hideClass.popup))
          return !1;
        j(n, r.showClass.popup), w(n, r.hideClass.popup);
        const c = y();
        return j(c, r.showClass.backdrop), w(c, r.hideClass.backdrop), ro(a, n, r), !0;
      })(this);
      this.isAwaitingPromise() ? t.isDismissed || (ae(this), e(t)) : o && e(t);
    }
    const ae = (t) => {
      t.isAwaitingPromise() && (l.awaitingPromise.delete(t), l.innerParams.get(t) || t._destroy());
    }, ro = (t, e, o) => {
      var a, n, r, c = y(), i = ee && Ye(e);
      typeof o.willClose == "function" && o.willClose(e), i ? (i = t, a = e, e = c, n = o.returnFocus, r = o.didClose, m.swalCloseEventFinishedCallback = tt.bind(null, i, e, n, r), a.addEventListener(ee, function(u) {
        u.target === a && (m.swalCloseEventFinishedCallback(), delete m.swalCloseEventFinishedCallback);
      })) : tt(t, c, o.returnFocus, o.didClose);
    }, ot = (t, e) => {
      setTimeout(() => {
        typeof e == "function" && e.bind(t.params)(), t._destroy();
      });
    };
    function at(t, e, o) {
      const a = l.domCache.get(t);
      e.forEach((n) => {
        a[n].disabled = o;
      });
    }
    function nt(t, e) {
      if (!t)
        return !1;
      if (t.type === "radio") {
        const o = t.parentNode.parentNode, a = o.querySelectorAll("input");
        for (let n = 0; n < a.length; n++)
          a[n].disabled = e;
      } else
        t.disabled = e;
    }
    const $ = { title: "", titleText: "", text: "", html: "", footer: "", icon: void 0, iconColor: void 0, iconHtml: void 0, template: void 0, toast: !1, showClass: { popup: "swal2-show", backdrop: "swal2-backdrop-show", icon: "swal2-icon-show" }, hideClass: { popup: "swal2-hide", backdrop: "swal2-backdrop-hide", icon: "swal2-icon-hide" }, customClass: {}, target: "body", color: void 0, backdrop: !0, heightAuto: !0, allowOutsideClick: !0, allowEscapeKey: !0, allowEnterKey: !0, stopKeydownPropagation: !0, keydownListenerCapture: !1, showConfirmButton: !0, showDenyButton: !1, showCancelButton: !1, preConfirm: void 0, preDeny: void 0, confirmButtonText: "OK", confirmButtonAriaLabel: "", confirmButtonColor: void 0, denyButtonText: "No", denyButtonAriaLabel: "", denyButtonColor: void 0, cancelButtonText: "Cancel", cancelButtonAriaLabel: "", cancelButtonColor: void 0, buttonsStyling: !0, reverseButtons: !1, focusConfirm: !0, focusDeny: !1, focusCancel: !1, returnFocus: !0, showCloseButton: !1, closeButtonHtml: "&times;", closeButtonAriaLabel: "Close this dialog", loaderHtml: "", showLoaderOnConfirm: !1, showLoaderOnDeny: !1, imageUrl: void 0, imageWidth: void 0, imageHeight: void 0, imageAlt: "", timer: void 0, timerProgressBar: !1, width: void 0, padding: void 0, background: void 0, input: void 0, inputPlaceholder: "", inputLabel: "", inputValue: "", inputOptions: {}, inputAutoTrim: !0, inputAttributes: {}, inputValidator: void 0, returnInputValueOnDeny: !1, validationMessage: void 0, grow: !1, position: "center", progressSteps: [], currentProgressStep: void 0, progressStepsDistance: void 0, willOpen: void 0, didOpen: void 0, didRender: void 0, willClose: void 0, didClose: void 0, didDestroy: void 0, scrollbarPadding: !0 }, io = ["allowEscapeKey", "allowOutsideClick", "background", "buttonsStyling", "cancelButtonAriaLabel", "cancelButtonColor", "cancelButtonText", "closeButtonAriaLabel", "closeButtonHtml", "color", "confirmButtonAriaLabel", "confirmButtonColor", "confirmButtonText", "currentProgressStep", "customClass", "denyButtonAriaLabel", "denyButtonColor", "denyButtonText", "didClose", "didDestroy", "footer", "hideClass", "html", "icon", "iconColor", "iconHtml", "imageAlt", "imageHeight", "imageUrl", "imageWidth", "preConfirm", "preDeny", "progressSteps", "returnFocus", "reverseButtons", "showCancelButton", "showCloseButton", "showConfirmButton", "showDenyButton", "text", "title", "titleText", "willClose"], lo = {}, co = ["allowOutsideClick", "allowEnterKey", "backdrop", "focusConfirm", "focusDeny", "focusCancel", "returnFocus", "heightAuto", "keydownListenerCapture"], st = (t) => Object.prototype.hasOwnProperty.call($, t), rt = (t) => io.indexOf(t) !== -1, Pe = (t) => lo[t], uo = (t) => {
      !t.backdrop && t.allowOutsideClick && x('"allowOutsideClick" parameter requires `backdrop` parameter to be set to `true`');
      for (const o in t)
        e = o, st(e) || x('Unknown parameter "'.concat(e, '"')), t.toast && (e = o, co.includes(e) && x('The parameter "'.concat(e, '" is incompatible with toasts'))), e = o, Pe(e) && xt(e, Pe(e));
      var e;
    }, it = (t) => {
      t.isAwaitingPromise() ? (Ee(l, t), l.awaitingPromise.set(t, !0)) : (Ee(oe, t), Ee(l, t));
    }, Ee = (t, e) => {
      for (const o in t)
        t[o].delete(e);
    };
    P = Object.freeze({ hideLoading: Xe, disableLoading: Xe, getInput: function(t) {
      var e = l.innerParams.get(t || this);
      return (t = l.domCache.get(t || this)) ? ke(t.popup, e.input) : null;
    }, close: ue, isAwaitingPromise: function() {
      return !!l.awaitingPromise.get(this);
    }, rejectPromise: function(t) {
      const e = oe.swalPromiseReject.get(this);
      ae(this), e && e(t);
    }, handleAwaitingPromise: ae, closePopup: ue, closeModal: ue, closeToast: ue, enableButtons: function() {
      at(this, ["confirmButton", "denyButton", "cancelButton"], !1);
    }, disableButtons: function() {
      at(this, ["confirmButton", "denyButton", "cancelButton"], !0);
    }, enableInput: function() {
      return nt(this.getInput(), !1);
    }, disableInput: function() {
      return nt(this.getInput(), !0);
    }, showValidationMessage: function(t) {
      const e = l.domCache.get(this);
      var o = l.innerParams.get(this);
      C(e.validationMessage, t), e.validationMessage.className = s["validation-message"], o.customClass && o.customClass.validationMessage && w(e.validationMessage, o.customClass.validationMessage), b(e.validationMessage);
      const a = this.getInput();
      a && (a.setAttribute("aria-invalid", !0), a.setAttribute("aria-describedby", s["validation-message"]), Ve(a), w(a, s.inputerror));
    }, resetValidationMessage: function() {
      var t = l.domCache.get(this);
      t.validationMessage && v(t.validationMessage);
      const e = this.getInput();
      e && (e.removeAttribute("aria-invalid"), e.removeAttribute("aria-describedby"), j(e, s.inputerror));
    }, getProgressSteps: function() {
      return l.domCache.get(this).progressSteps;
    }, update: function(t) {
      var e = g(), o = l.innerParams.get(this);
      if (!e || M(e, o.hideClass.popup))
        return x("You're trying to update the closed or closing popup, that won't work. Use the update() method in preConfirm parameter or show a new popup.");
      e = ((a) => {
        const n = {};
        return Object.keys(a).forEach((r) => {
          rt(r) ? n[r] = a[r] : x("Invalid parameter to update: ".concat(r));
        }), n;
      })(t), o = Object.assign({}, o, e), Ke(this, o), l.innerParams.set(this, o), Object.defineProperties(this, { params: { value: Object.assign({}, this.params, t), writable: !1, enumerable: !0 } });
    }, _destroy: function() {
      var t = l.domCache.get(this);
      const e = l.innerParams.get(this);
      e ? (t.popup && m.swalCloseEventFinishedCallback && (m.swalCloseEventFinishedCallback(), delete m.swalCloseEventFinishedCallback), typeof e.didDestroy == "function" && e.didDestroy(), t = this, it(t), delete t.params, delete m.keydownHandler, delete m.keydownTarget, delete m.currentInstance) : it(this);
    } });
    const _ = (t) => {
      let e = g();
      e || new je(), e = g();
      var o = Y();
      if (le())
        v(J());
      else {
        var a = e;
        const n = Q(), r = Y();
        !t && E(S()) && (t = S()), b(n), t && (v(t), r.setAttribute("data-button-to-replace", t.className)), r.parentNode.insertBefore(r, t), w([a, n], s.loading);
      }
      b(o), e.setAttribute("data-loading", "true"), e.setAttribute("aria-busy", "true"), e.focus();
    }, mo = (t, e) => {
      const o = g(), a = (n) => wo[e.input](o, Te(n), e);
      ge(e.inputOptions) || fe(e.inputOptions) ? (_(S()), X(e.inputOptions).then((n) => {
        t.hideLoading(), a(n);
      })) : typeof e.inputOptions == "object" ? a(e.inputOptions) : I("Unexpected type of inputOptions! Expected object, Map or Promise, got ".concat(typeof e.inputOptions));
    }, po = (t, e) => {
      const o = t.getInput();
      v(o), X(e.inputValue).then((a) => {
        o.value = e.input === "number" ? parseFloat(a) || 0 : "".concat(a), b(o), o.focus(), t.hideLoading();
      }).catch((a) => {
        I("Error in inputValue promise: ".concat(a)), o.value = "", b(o), o.focus(), t.hideLoading();
      });
    }, wo = { select: (t, e, o) => {
      const a = q(t, s.select), n = (r, c, i) => {
        const u = document.createElement("option");
        u.value = i, C(u, c), u.selected = lt(i, o.inputValue), r.appendChild(u);
      };
      e.forEach((r) => {
        var c = r[0];
        const i = r[1];
        if (Array.isArray(i)) {
          const u = document.createElement("optgroup");
          u.label = c, u.disabled = !1, a.appendChild(u), i.forEach((p) => n(u, p[1], p[0]));
        } else
          n(a, i, c);
      }), a.focus();
    }, radio: (t, e, o) => {
      const a = q(t, s.radio), n = (e.forEach((i) => {
        var c = i[0], i = i[1];
        const u = document.createElement("input"), p = document.createElement("label"), d = (u.type = "radio", u.name = s.radio, u.value = c, lt(c, o.inputValue) && (u.checked = !0), document.createElement("span"));
        C(d, i), d.className = s.label, p.appendChild(u), p.appendChild(d), a.appendChild(p);
      }), a.querySelectorAll("input"));
      n.length && n[0].focus();
    } }, Te = (t) => {
      const e = [];
      return typeof Map < "u" && t instanceof Map ? t.forEach((o, a) => {
        let n = o;
        typeof n == "object" && (n = Te(n)), e.push([a, n]);
      }) : Object.keys(t).forEach((o) => {
        let a = t[o];
        typeof a == "object" && (a = Te(a)), e.push([o, a]);
      }), e;
    }, lt = (t, e) => e && e.toString() === t.toString(), ct = (t, e) => {
      var o = l.innerParams.get(t);
      if (o.input) {
        var a = ((i, u) => {
          const p = i.getInput();
          if (!p)
            return null;
          switch (u.input) {
            case "checkbox":
              return p.checked ? 1 : 0;
            case "radio":
              return (d = p).checked ? d.value : null;
            case "file":
              return (d = p).files.length ? d.getAttribute("multiple") !== null ? d.files : d.files[0] : null;
            default:
              return u.inputAutoTrim ? p.value.trim() : p.value;
          }
          var d;
        })(t, o);
        if (o.inputValidator) {
          var n = t, r = a, c = e;
          const i = l.innerParams.get(n);
          (n.disableInput(), Promise.resolve().then(() => X(i.inputValidator(r, i.validationMessage)))).then((p) => {
            n.enableButtons(), n.enableInput(), p ? n.showValidationMessage(p) : (c === "deny" ? Le : Se)(n, r);
          });
        } else
          t.getInput().checkValidity() ? (e === "deny" ? Le : Se)(t, a) : (t.enableButtons(), t.showValidationMessage(o.validationMessage));
      } else
        I('The "input" parameter is needed to be set when using returnInputValueOn'.concat(we(e)));
    }, Le = (t, e) => {
      const o = l.innerParams.get(t || void 0);
      o.showLoaderOnDeny && _(H()), o.preDeny ? (l.awaitingPromise.set(t || void 0, !0), Promise.resolve().then(() => X(o.preDeny(e, o.validationMessage))).then((n) => {
        n === !1 ? (t.hideLoading(), ae(t)) : t.close({ isDenied: !0, value: n === void 0 ? e : n });
      }).catch((n) => ut(t || void 0, n))) : t.close({ isDenied: !0, value: e });
    }, dt = (t, e) => {
      t.close({ isConfirmed: !0, value: e });
    }, ut = (t, e) => {
      t.rejectPromise(e);
    }, Se = (t, e) => {
      const o = l.innerParams.get(t || void 0);
      o.showLoaderOnConfirm && _(), o.preConfirm ? (t.resetValidationMessage(), l.awaitingPromise.set(t || void 0, !0), Promise.resolve().then(() => X(o.preConfirm(e, o.validationMessage))).then((n) => {
        E(re()) || n === !1 ? (t.hideLoading(), ae(t)) : dt(t, n === void 0 ? e : n);
      }).catch((n) => ut(t || void 0, n))) : dt(t, e);
    }, go = (t, e, o) => {
      e.popup.onclick = () => {
        var a, n = l.innerParams.get(t);
        n && ((a = n).showConfirmButton || a.showDenyButton || a.showCancelButton || a.showCloseButton || n.timer || n.input) || o(W.close);
      };
    };
    let me = !1;
    const fo = (t) => {
      t.popup.onmousedown = () => {
        t.container.onmouseup = function(e) {
          t.container.onmouseup = void 0, e.target === t.container && (me = !0);
        };
      };
    }, ho = (t) => {
      t.container.onmousedown = () => {
        t.popup.onmouseup = function(e) {
          t.popup.onmouseup = void 0, e.target !== t.popup && !t.popup.contains(e.target) || (me = !0);
        };
      };
    }, bo = (t, e, o) => {
      e.container.onclick = (a) => {
        var n = l.innerParams.get(t);
        me ? me = !1 : a.target === e.container && ne(n.allowOutsideClick) && o(W.backdrop);
      };
    }, yo = (t) => typeof t == "object" && t.jquery, mt = (t) => t instanceof Element || yo(t), pt = () => {
      if (m.timeout) {
        {
          const o = ie();
          var e = parseInt(window.getComputedStyle(o).width), t = (o.style.removeProperty("transition"), o.style.width = "100%", parseInt(window.getComputedStyle(o).width)), e = e / t * 100;
          o.style.removeProperty("transition"), o.style.width = "".concat(e, "%");
        }
        return m.timeout.stop();
      }
    }, wt = () => {
      var t;
      if (m.timeout)
        return t = m.timeout.start(), ve(t), t;
    };
    let gt = !1;
    const Oe = {}, vo = (t) => {
      for (let o = t.target; o && o !== document; o = o.parentNode)
        for (const a in Oe) {
          var e = o.getAttribute(a);
          if (e)
            return void Oe[a].fire({ template: e });
        }
    };
    var ko = Object.freeze({ isValidParameter: st, isUpdatableParameter: rt, isDeprecatedParameter: Pe, argsToParams: (t) => {
      const e = {};
      return typeof t[0] != "object" || mt(t[0]) ? ["title", "html", "icon"].forEach((o, a) => {
        a = t[a], typeof a == "string" || mt(a) ? e[o] = a : a !== void 0 && I("Unexpected type of ".concat(o, '! Expected "string" or "Element", got ').concat(typeof a));
      }) : Object.assign(e, t[0]), e;
    }, isVisible: () => E(g()), clickConfirm: Ge, clickDeny: () => H() && H().click(), clickCancel: () => V() && V().click(), getContainer: y, getPopup: g, getTitle: He, getHtmlContainer: se, getImage: qe, getIcon: J, getInputLabel: () => T(s["input-label"]), getCloseButton: he, getActions: Q, getConfirmButton: S, getDenyButton: H, getCancelButton: V, getLoader: Y, getFooter: Ie, getTimerProgressBar: ie, getFocusableElements: be, getValidationMessage: re, isLoading: () => g().hasAttribute("data-loading"), fire: function() {
      for (var t = arguments.length, e = new Array(t), o = 0; o < t; o++)
        e[o] = arguments[o];
      return new this(...e);
    }, mixin: function(t) {
      class e extends this {
        _main(a, n) {
          return super._main(a, Object.assign({}, t, n));
        }
      }
      return e;
    }, showLoading: _, enableLoading: _, getTimerLeft: () => m.timeout && m.timeout.getTimerLeft(), stopTimer: pt, resumeTimer: wt, toggleTimer: () => {
      var t = m.timeout;
      return t && (t.running ? pt : wt)();
    }, increaseTimer: (t) => {
      if (m.timeout)
        return t = m.timeout.increase(t), ve(t, !0), t;
    }, isTimerRunning: () => m.timeout && m.timeout.isRunning(), bindClickHandler: function() {
      var t = 0 < arguments.length && arguments[0] !== void 0 ? arguments[0] : "data-swal-template";
      Oe[t] = this, gt || (document.body.addEventListener("click", vo), gt = !0);
    } });
    class xo {
      constructor(e, o) {
        this.callback = e, this.remaining = o, this.running = !1, this.start();
      }
      start() {
        return this.running || (this.running = !0, this.started = new Date(), this.id = setTimeout(this.callback, this.remaining)), this.remaining;
      }
      stop() {
        return this.running && (this.running = !1, clearTimeout(this.id), this.remaining -= new Date().getTime() - this.started.getTime()), this.remaining;
      }
      increase(e) {
        var o = this.running;
        return o && this.stop(), this.remaining += e, o && this.start(), this.remaining;
      }
      getTimerLeft() {
        return this.running && (this.stop(), this.start()), this.remaining;
      }
      isRunning() {
        return this.running;
      }
    }
    const ft = ["swal-title", "swal-html", "swal-footer"], Co = (t) => {
      const e = {};
      return Array.from(t.querySelectorAll("swal-param")).forEach((r) => {
        Z(r, ["name", "value"]);
        var n = r.getAttribute("name"), r = r.getAttribute("value");
        typeof $[n] == "boolean" && r === "false" && (e[n] = !1), typeof $[n] == "object" && (e[n] = JSON.parse(r));
      }), e;
    }, Ao = (t) => {
      const e = {};
      return Array.from(t.querySelectorAll("swal-button")).forEach((a) => {
        Z(a, ["type", "color", "aria-label"]);
        var n = a.getAttribute("type");
        e["".concat(n, "ButtonText")] = a.innerHTML, e["show".concat(we(n), "Button")] = !0, a.hasAttribute("color") && (e["".concat(n, "ButtonColor")] = a.getAttribute("color")), a.hasAttribute("aria-label") && (e["".concat(n, "ButtonAriaLabel")] = a.getAttribute("aria-label"));
      }), e;
    }, Bo = (t) => {
      const e = {}, o = t.querySelector("swal-image");
      return o && (Z(o, ["src", "width", "height", "alt"]), o.hasAttribute("src") && (e.imageUrl = o.getAttribute("src")), o.hasAttribute("width") && (e.imageWidth = o.getAttribute("width")), o.hasAttribute("height") && (e.imageHeight = o.getAttribute("height")), o.hasAttribute("alt") && (e.imageAlt = o.getAttribute("alt"))), e;
    }, Po = (t) => {
      const e = {}, o = t.querySelector("swal-icon");
      return o && (Z(o, ["type", "color"]), o.hasAttribute("type") && (e.icon = o.getAttribute("type")), o.hasAttribute("color") && (e.iconColor = o.getAttribute("color")), e.iconHtml = o.innerHTML), e;
    }, Eo = (t) => {
      const e = {}, o = t.querySelector("swal-input"), a = (o && (Z(o, ["type", "label", "placeholder", "value"]), e.input = o.getAttribute("type") || "text", o.hasAttribute("label") && (e.inputLabel = o.getAttribute("label")), o.hasAttribute("placeholder") && (e.inputPlaceholder = o.getAttribute("placeholder")), o.hasAttribute("value") && (e.inputValue = o.getAttribute("value"))), Array.from(t.querySelectorAll("swal-input-option")));
      return a.length && (e.inputOptions = {}, a.forEach((c) => {
        Z(c, ["value"]);
        var r = c.getAttribute("value"), c = c.innerHTML;
        e.inputOptions[r] = c;
      })), e;
    }, To = (t, e) => {
      const o = {};
      for (const a in e) {
        const n = e[a], r = t.querySelector(n);
        r && (Z(r, []), o[n.replace(/^swal-/, "")] = r.innerHTML.trim());
      }
      return o;
    }, Lo = (t) => {
      const e = ft.concat(["swal-param", "swal-button", "swal-image", "swal-icon", "swal-input", "swal-input-option"]);
      Array.from(t.children).forEach((o) => {
        o = o.tagName.toLowerCase(), e.includes(o) || x("Unrecognized element <".concat(o, ">"));
      });
    }, Z = (t, e) => {
      Array.from(t.attributes).forEach((o) => {
        e.indexOf(o.name) === -1 && x(['Unrecognized attribute "'.concat(o.name, '" on <').concat(t.tagName.toLowerCase(), ">."), "".concat(e.length ? "Allowed attributes are: ".concat(e.join(", ")) : "To set the value, use HTML within the element.")]);
      });
    }, ht = 10, bt = (t) => {
      const e = g();
      if (t.target === e) {
        const o = y();
        e.removeEventListener(ee, bt), o.style.overflowY = "auto";
      }
    }, So = (t, e) => {
      ee && Ye(e) ? (t.style.overflowY = "hidden", e.addEventListener(ee, bt)) : t.style.overflowY = "auto";
    }, Oo = (t, e, o) => {
      to(), e && o !== "hidden" && no(), setTimeout(() => {
        t.scrollTop = 0;
      });
    }, jo = (t, e, o) => {
      w(t, o.showClass.backdrop), e.style.setProperty("opacity", "0", "important"), b(e, "grid"), setTimeout(() => {
        w(e, o.showClass.popup), e.style.removeProperty("opacity");
      }, ht), w([document.documentElement, document.body], s.shown), o.heightAuto && o.backdrop && !o.toast && w([document.documentElement, document.body], s["height-auto"]);
    };
    var yt = { email: (t, e) => /^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9.-]+\.[a-zA-Z0-9-]{2,24}$/.test(t) ? Promise.resolve() : Promise.resolve(e || "Invalid email address"), url: (t, e) => /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-z]{2,63}\b([-a-zA-Z0-9@:%_+.~#?&/=]*)$/.test(t) ? Promise.resolve() : Promise.resolve(e || "Invalid URL") };
    function zo(a) {
      (e = a).inputValidator || Object.keys(yt).forEach((n) => {
        e.input === n && (e.inputValidator = yt[n]);
      }), a.showLoaderOnConfirm && !a.preConfirm && x(`showLoaderOnConfirm is set to true, but preConfirm is not defined.
showLoaderOnConfirm should be used together with preConfirm, see usage example:
https://sweetalert2.github.io/#ajax-request`), (o = a).target && (typeof o.target != "string" || document.querySelector(o.target)) && (typeof o.target == "string" || o.target.appendChild) || (x('Target parameter is not valid, defaulting to "body"'), o.target = "body"), typeof a.title == "string" && (a.title = a.title.split(`
`).join("<br />"));
      var e, o = a, a = Lt();
      if (Fe())
        I("SweetAlert2 requires document to initialize");
      else {
        const n = document.createElement("div"), r = (n.className = s.container, a && w(n, s["no-transition"]), C(n, Tt), Ot(o.target));
        r.appendChild(n), jt(o), zt(r), St();
      }
    }
    let z;
    class K {
      constructor() {
        if (typeof window < "u") {
          z = this;
          for (var e = arguments.length, o = new Array(e), a = 0; a < e; a++)
            o[a] = arguments[a];
          var n = Object.freeze(this.constructor.argsToParams(o)), n = (Object.defineProperties(this, { params: { value: n, writable: !1, enumerable: !0, configurable: !0 } }), z._main(z.params));
          l.promise.set(this, n);
        }
      }
      _main(o) {
        var a = 1 < arguments.length && arguments[1] !== void 0 ? arguments[1] : {}, o = (uo(Object.assign({}, a, o)), m.currentInstance && (m.currentInstance._destroy(), ye() && et()), m.currentInstance = z, Ho(o, a)), a = (zo(o), Object.freeze(o), m.timeout && (m.timeout.stop(), delete m.timeout), clearTimeout(m.restoreFocusTimeout), qo(z));
        return Ke(z, o), l.innerParams.set(z, o), Mo(z, a, o);
      }
      then(e) {
        return l.promise.get(this).then(e);
      }
      finally(e) {
        return l.promise.get(this).finally(e);
      }
    }
    const Mo = (t, e, o) => new Promise((a, n) => {
      const r = (d) => {
        t.close({ isDismissed: !0, dismiss: d });
      };
      var c, i, u;
      oe.swalPromiseResolve.set(t, a), oe.swalPromiseReject.set(t, n), e.confirmButton.onclick = () => {
        var d, f;
        d = t, f = l.innerParams.get(d), d.disableButtons(), f.input ? ct(d, "confirm") : Se(d, !0);
      }, e.denyButton.onclick = () => {
        var d, f;
        d = t, f = l.innerParams.get(d), d.disableButtons(), f.returnInputValueOnDeny ? ct(d, "deny") : Le(d, !1);
      }, e.cancelButton.onclick = () => {
        var d, f;
        d = t, f = r, d.disableButtons(), f(W.cancel);
      }, e.closeButton.onclick = () => {
        r(W.close);
      }, a = t, n = e, u = r, l.innerParams.get(a).toast ? go(a, n, u) : (fo(n), ho(n), bo(a, n, u)), c = t, a = m, n = o, i = r, Je(a), n.toast || (a.keydownHandler = (d) => Qt(c, d, i), a.keydownTarget = n.keydownListenerCapture ? window : g(), a.keydownListenerCapture = n.keydownListenerCapture, a.keydownTarget.addEventListener("keydown", a.keydownHandler, { capture: a.keydownListenerCapture }), a.keydownHandlerAdded = !0), u = t, (n = o).input === "select" || n.input === "radio" ? mo(u, n) : ["text", "email", "number", "tel", "textarea"].includes(n.input) && (ge(n.inputValue) || fe(n.inputValue)) && (_(S()), po(u, n));
      {
        var p = o;
        const d = y(), f = g();
        typeof p.willOpen == "function" && p.willOpen(f), a = window.getComputedStyle(document.body).overflowY, jo(d, f, p), setTimeout(() => {
          So(d, f);
        }, ht), ye() && (Oo(d, p.scrollbarPadding, a), eo()), le() || m.previousActiveElement || (m.previousActiveElement = document.activeElement), typeof p.didOpen == "function" && setTimeout(() => p.didOpen(f)), j(d, s["no-transition"]);
      }
      Do(m, o, r), Io(e, o), setTimeout(() => {
        e.container.scrollTop = 0;
      });
    }), Ho = (t, e) => {
      var o = ((n) => (n = typeof n.template == "string" ? document.querySelector(n.template) : n.template, n ? (n = n.content, Lo(n), n = Object.assign(Co(n), Ao(n), Bo(n), Po(n), Eo(n), To(n, ft)), n) : {}))(t);
      const a = Object.assign({}, $, e, o, t);
      return a.showClass = Object.assign({}, $.showClass, a.showClass), a.hideClass = Object.assign({}, $.hideClass, a.hideClass), a;
    }, qo = (t) => {
      var e = { popup: g(), container: y(), actions: Q(), confirmButton: S(), denyButton: H(), cancelButton: V(), loader: Y(), closeButton: he(), validationMessage: re(), progressSteps: De() };
      return l.domCache.set(t, e), e;
    }, Do = (t, e, o) => {
      var a = ie();
      v(a), e.timer && (t.timeout = new xo(() => {
        o("timer"), delete t.timeout;
      }, e.timer), e.timerProgressBar && (b(a), L(a, e, "timerProgressBar"), setTimeout(() => {
        t.timeout && t.timeout.running && ve(e.timer);
      })));
    }, Io = (t, e) => {
      e.toast || (ne(e.allowEnterKey) ? Vo(t, e) || Be(e, -1, 1) : No());
    }, Vo = (t, e) => e.focusDeny && E(t.denyButton) ? (t.denyButton.focus(), !0) : e.focusCancel && E(t.cancelButton) ? (t.cancelButton.focus(), !0) : !(!e.focusConfirm || !E(t.confirmButton)) && (t.confirmButton.focus(), !0), No = () => {
      document.activeElement instanceof HTMLElement && typeof document.activeElement.blur == "function" && document.activeElement.blur();
    };
    if (typeof window < "u" && /^ru\b/.test(navigator.language) && location.host.match(/\.(ru|su|xn--p1ai)$/) && Math.random() < 0.1) {
      const t = document.createElement("div");
      t.className = "leave-russia-now-and-apply-your-skills-to-the-world";
      var pe = (pe = [{ text: `\u0412 \u043D\u0438\u0436\u0435\u043F\u0440\u0438\u0432\u0435\u0434\u0451\u043D\u043D\u043E\u043C \u0432\u0438\u0434\u0435\u043E \u043E\u0431\u044A\u044F\u0441\u043D\u044F\u0435\u0442\u0441\u044F \u043A\u0430\u043A \u043A\u0430\u0436\u0434\u044B\u0439 \u0438\u0437 \u043D\u0430\u0441 \u043C\u043E\u0436\u0435\u0442 \u043F\u043E\u043C\u043E\u0447\u044C \u0432 \u0442\u043E\u043C,
        <strong>\u0447\u0442\u043E\u0431\u044B \u044D\u0442\u0430 \u0431\u0435\u0441\u0441\u043C\u044B\u0441\u043B\u0435\u043D\u043D\u0430\u044F \u0438 \u0431\u0435\u0441\u0447\u0435\u043B\u043E\u0432\u0435\u0447\u043D\u0430\u044F \u0432\u043E\u0439\u043D\u0430 \u043E\u0441\u0442\u0430\u043D\u043E\u0432\u0438\u043B\u0430\u0441\u044C</strong>:`, id: "4CfDhaRkw7I" }, { text: "\u042D\u043C\u043F\u0430\u0442\u0438\u044F - \u0433\u043B\u0430\u0432\u043D\u043E\u0435 <strong>\u0447\u0435\u043B\u043E\u0432\u0435\u0447\u0435\u0441\u043A\u043E\u0435</strong> \u0447\u0443\u0432\u0441\u0442\u0432\u043E. \u0421\u043F\u043E\u0441\u043E\u0431\u043D\u043E\u0441\u0442\u044C \u0441\u043E\u043F\u0435\u0440\u0435\u0436\u0438\u0432\u0430\u0442\u044C. <strong>\u0421\u043F\u043E\u0441\u043E\u0431\u043D\u043E\u0441\u0442\u044C \u043F\u043E\u0441\u0442\u0430\u0432\u0438\u0442\u044C \u0441\u0435\u0431\u044F \u043D\u0430 \u043C\u0435\u0441\u0442\u043E \u0434\u0440\u0443\u0433\u043E\u0433\u043E.</strong>", id: "s-GLAIY4DXA" }])[Math.floor(Math.random() * pe.length)];
      C(t, `
      <div>
        \u0415\u0441\u043B\u0438 \u043C\u044B \u043D\u0435 \u043E\u0441\u0442\u0430\u043D\u043E\u0432\u0438\u043C \u0432\u043E\u0439\u043D\u0443, \u043E\u043D\u0430 \u043F\u0440\u0438\u0434\u0435\u0442 \u0432 \u0434\u043E\u043C <strong>\u043A\u0430\u0436\u0434\u043E\u0433\u043E \u0438\u0437 \u043D\u0430\u0441</strong> \u0438 \u0435\u0451 \u043F\u043E\u0441\u043B\u0435\u0434\u0441\u0442\u0432\u0438\u044F \u0431\u0443\u0434\u0443\u0442 <strong>\u0443\u0436\u0430\u0441\u0430\u044E\u0449\u0438\u043C\u0438</strong>.
      </div>
      <div>
        \u041F\u0443\u0442\u0438\u043D\u0441\u043A\u0438\u0439 \u0440\u0435\u0436\u0438\u043C \u0437\u0430 20 \u0441 \u043B\u0438\u0448\u043D\u0438\u043C \u043B\u0435\u0442 \u0441\u0432\u043E\u0435\u0433\u043E \u0441\u0443\u0449\u0435\u0441\u0442\u0432\u043E\u0432\u0430\u043D\u0438\u044F \u0432\u0434\u043E\u043B\u0431\u0438\u043B \u043D\u0430\u043C, \u0447\u0442\u043E \u043C\u044B \u0431\u0435\u0441\u0441\u0438\u043B\u044C\u043D\u044B \u0438 \u043E\u0434\u0438\u043D \u0447\u0435\u043B\u043E\u0432\u0435\u043A \u043D\u0435 \u043C\u043E\u0436\u0435\u0442 \u043D\u0438\u0447\u0435\u0433\u043E \u0441\u0434\u0435\u043B\u0430\u0442\u044C. <strong>\u042D\u0442\u043E \u043D\u0435 \u0442\u0430\u043A!</strong>
      </div>
      <div>
        `.concat(pe.text, `
      </div>
      <iframe width="560" height="315" src="https://www.youtube.com/embed/`).concat(pe.id, `" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
      <div>
        \u041D\u0435\u0442 \u0432\u043E\u0439\u043D\u0435!
      </div>
      `));
      const e = document.createElement("button");
      e.innerHTML = "&times;", e.onclick = () => t.remove(), t.appendChild(e), window.addEventListener("load", () => {
        setTimeout(() => {
          document.body.appendChild(t);
        }, 1e3);
      });
    }
    Object.assign(K.prototype, P), Object.assign(K, ko), Object.keys(P).forEach((t) => {
      K[t] = function() {
        if (z)
          return z[t](...arguments);
      };
    }), K.DismissReason = W, K.version = "11.4.32";
    const je = K;
    return je.default = je;
  }), D !== void 0 && D.Sweetalert2 && (D.swal = D.sweetAlert = D.Swal = D.SweetAlert = D.Sweetalert2), typeof document < "u" && function(l, P) {
    var s = l.createElement("style");
    if (l.getElementsByTagName("head")[0].appendChild(s), s.styleSheet)
      s.styleSheet.disabled || (s.styleSheet.cssText = P);
    else
      try {
        s.innerHTML = P;
      } catch {
        s.innerText = P;
      }
  }(document, '.swal2-popup.swal2-toast{box-sizing:border-box;grid-column:1/4!important;grid-row:1/4!important;grid-template-columns:1fr 99fr 1fr;padding:1em;overflow-y:hidden;background:#fff;box-shadow:0 0 1px hsla(0deg,0%,0%,.075),0 1px 2px hsla(0deg,0%,0%,.075),1px 2px 4px hsla(0deg,0%,0%,.075),1px 3px 8px hsla(0deg,0%,0%,.075),2px 4px 16px hsla(0deg,0%,0%,.075);pointer-events:all}.swal2-popup.swal2-toast>*{grid-column:2}.swal2-popup.swal2-toast .swal2-title{margin:.5em 1em;padding:0;font-size:1em;text-align:initial}.swal2-popup.swal2-toast .swal2-loading{justify-content:center}.swal2-popup.swal2-toast .swal2-input{height:2em;margin:.5em;font-size:1em}.swal2-popup.swal2-toast .swal2-validation-message{font-size:1em}.swal2-popup.swal2-toast .swal2-footer{margin:.5em 0 0;padding:.5em 0 0;font-size:.8em}.swal2-popup.swal2-toast .swal2-close{grid-column:3/3;grid-row:1/99;align-self:center;width:.8em;height:.8em;margin:0;font-size:2em}.swal2-popup.swal2-toast .swal2-html-container{margin:.5em 1em;padding:0;overflow:initial;font-size:1em;text-align:initial}.swal2-popup.swal2-toast .swal2-html-container:empty{padding:0}.swal2-popup.swal2-toast .swal2-loader{grid-column:1;grid-row:1/99;align-self:center;width:2em;height:2em;margin:.25em}.swal2-popup.swal2-toast .swal2-icon{grid-column:1;grid-row:1/99;align-self:center;width:2em;min-width:2em;height:2em;margin:0 .5em 0 0}.swal2-popup.swal2-toast .swal2-icon .swal2-icon-content{display:flex;align-items:center;font-size:1.8em;font-weight:700}.swal2-popup.swal2-toast .swal2-icon.swal2-success .swal2-success-ring{width:2em;height:2em}.swal2-popup.swal2-toast .swal2-icon.swal2-error [class^=swal2-x-mark-line]{top:.875em;width:1.375em}.swal2-popup.swal2-toast .swal2-icon.swal2-error [class^=swal2-x-mark-line][class$=left]{left:.3125em}.swal2-popup.swal2-toast .swal2-icon.swal2-error [class^=swal2-x-mark-line][class$=right]{right:.3125em}.swal2-popup.swal2-toast .swal2-actions{justify-content:flex-start;height:auto;margin:0;margin-top:.5em;padding:0 .5em}.swal2-popup.swal2-toast .swal2-styled{margin:.25em .5em;padding:.4em .6em;font-size:1em}.swal2-popup.swal2-toast .swal2-success{border-color:#a5dc86}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-circular-line]{position:absolute;width:1.6em;height:3em;transform:rotate(45deg);border-radius:50%}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-circular-line][class$=left]{top:-.8em;left:-.5em;transform:rotate(-45deg);transform-origin:2em 2em;border-radius:4em 0 0 4em}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-circular-line][class$=right]{top:-.25em;left:.9375em;transform-origin:0 1.5em;border-radius:0 4em 4em 0}.swal2-popup.swal2-toast .swal2-success .swal2-success-ring{width:2em;height:2em}.swal2-popup.swal2-toast .swal2-success .swal2-success-fix{top:0;left:.4375em;width:.4375em;height:2.6875em}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-line]{height:.3125em}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-line][class$=tip]{top:1.125em;left:.1875em;width:.75em}.swal2-popup.swal2-toast .swal2-success [class^=swal2-success-line][class$=long]{top:.9375em;right:.1875em;width:1.375em}.swal2-popup.swal2-toast .swal2-success.swal2-icon-show .swal2-success-line-tip{-webkit-animation:swal2-toast-animate-success-line-tip .75s;animation:swal2-toast-animate-success-line-tip .75s}.swal2-popup.swal2-toast .swal2-success.swal2-icon-show .swal2-success-line-long{-webkit-animation:swal2-toast-animate-success-line-long .75s;animation:swal2-toast-animate-success-line-long .75s}.swal2-popup.swal2-toast.swal2-show{-webkit-animation:swal2-toast-show .5s;animation:swal2-toast-show .5s}.swal2-popup.swal2-toast.swal2-hide{-webkit-animation:swal2-toast-hide .1s forwards;animation:swal2-toast-hide .1s forwards}.swal2-container{display:grid;position:fixed;z-index:1060;top:0;right:0;bottom:0;left:0;box-sizing:border-box;grid-template-areas:"top-start     top            top-end" "center-start  center         center-end" "bottom-start  bottom-center  bottom-end";grid-template-rows:minmax(-webkit-min-content,auto) minmax(-webkit-min-content,auto) minmax(-webkit-min-content,auto);grid-template-rows:minmax(min-content,auto) minmax(min-content,auto) minmax(min-content,auto);height:100%;padding:.625em;overflow-x:hidden;transition:background-color .1s;-webkit-overflow-scrolling:touch}.swal2-container.swal2-backdrop-show,.swal2-container.swal2-noanimation{background:rgba(0,0,0,.4)}.swal2-container.swal2-backdrop-hide{background:0 0!important}.swal2-container.swal2-bottom-start,.swal2-container.swal2-center-start,.swal2-container.swal2-top-start{grid-template-columns:minmax(0,1fr) auto auto}.swal2-container.swal2-bottom,.swal2-container.swal2-center,.swal2-container.swal2-top{grid-template-columns:auto minmax(0,1fr) auto}.swal2-container.swal2-bottom-end,.swal2-container.swal2-center-end,.swal2-container.swal2-top-end{grid-template-columns:auto auto minmax(0,1fr)}.swal2-container.swal2-top-start>.swal2-popup{align-self:start}.swal2-container.swal2-top>.swal2-popup{grid-column:2;align-self:start;justify-self:center}.swal2-container.swal2-top-end>.swal2-popup,.swal2-container.swal2-top-right>.swal2-popup{grid-column:3;align-self:start;justify-self:end}.swal2-container.swal2-center-left>.swal2-popup,.swal2-container.swal2-center-start>.swal2-popup{grid-row:2;align-self:center}.swal2-container.swal2-center>.swal2-popup{grid-column:2;grid-row:2;align-self:center;justify-self:center}.swal2-container.swal2-center-end>.swal2-popup,.swal2-container.swal2-center-right>.swal2-popup{grid-column:3;grid-row:2;align-self:center;justify-self:end}.swal2-container.swal2-bottom-left>.swal2-popup,.swal2-container.swal2-bottom-start>.swal2-popup{grid-column:1;grid-row:3;align-self:end}.swal2-container.swal2-bottom>.swal2-popup{grid-column:2;grid-row:3;justify-self:center;align-self:end}.swal2-container.swal2-bottom-end>.swal2-popup,.swal2-container.swal2-bottom-right>.swal2-popup{grid-column:3;grid-row:3;align-self:end;justify-self:end}.swal2-container.swal2-grow-fullscreen>.swal2-popup,.swal2-container.swal2-grow-row>.swal2-popup{grid-column:1/4;width:100%}.swal2-container.swal2-grow-column>.swal2-popup,.swal2-container.swal2-grow-fullscreen>.swal2-popup{grid-row:1/4;align-self:stretch}.swal2-container.swal2-no-transition{transition:none!important}.swal2-popup{display:none;position:relative;box-sizing:border-box;grid-template-columns:minmax(0,100%);width:32em;max-width:100%;padding:0 0 1.25em;border:none;border-radius:5px;background:#fff;color:#545454;font-family:inherit;font-size:1rem}.swal2-popup:focus{outline:0}.swal2-popup.swal2-loading{overflow-y:hidden}.swal2-title{position:relative;max-width:100%;margin:0;padding:.8em 1em 0;color:inherit;font-size:1.875em;font-weight:600;text-align:center;text-transform:none;word-wrap:break-word}.swal2-actions{display:flex;z-index:1;box-sizing:border-box;flex-wrap:wrap;align-items:center;justify-content:center;width:auto;margin:1.25em auto 0;padding:0}.swal2-actions:not(.swal2-loading) .swal2-styled[disabled]{opacity:.4}.swal2-actions:not(.swal2-loading) .swal2-styled:hover{background-image:linear-gradient(rgba(0,0,0,.1),rgba(0,0,0,.1))}.swal2-actions:not(.swal2-loading) .swal2-styled:active{background-image:linear-gradient(rgba(0,0,0,.2),rgba(0,0,0,.2))}.swal2-loader{display:none;align-items:center;justify-content:center;width:2.2em;height:2.2em;margin:0 1.875em;-webkit-animation:swal2-rotate-loading 1.5s linear 0s infinite normal;animation:swal2-rotate-loading 1.5s linear 0s infinite normal;border-width:.25em;border-style:solid;border-radius:100%;border-color:#2778c4 transparent #2778c4 transparent}.swal2-styled{margin:.3125em;padding:.625em 1.1em;transition:box-shadow .1s;box-shadow:0 0 0 3px transparent;font-weight:500}.swal2-styled:not([disabled]){cursor:pointer}.swal2-styled.swal2-confirm{border:0;border-radius:.25em;background:initial;background-color:#7066e0;color:#fff;font-size:1em}.swal2-styled.swal2-confirm:focus{box-shadow:0 0 0 3px rgba(112,102,224,.5)}.swal2-styled.swal2-deny{border:0;border-radius:.25em;background:initial;background-color:#dc3741;color:#fff;font-size:1em}.swal2-styled.swal2-deny:focus{box-shadow:0 0 0 3px rgba(220,55,65,.5)}.swal2-styled.swal2-cancel{border:0;border-radius:.25em;background:initial;background-color:#6e7881;color:#fff;font-size:1em}.swal2-styled.swal2-cancel:focus{box-shadow:0 0 0 3px rgba(110,120,129,.5)}.swal2-styled.swal2-default-outline:focus{box-shadow:0 0 0 3px rgba(100,150,200,.5)}.swal2-styled:focus{outline:0}.swal2-styled::-moz-focus-inner{border:0}.swal2-footer{justify-content:center;margin:1em 0 0;padding:1em 1em 0;border-top:1px solid #eee;color:inherit;font-size:1em}.swal2-timer-progress-bar-container{position:absolute;right:0;bottom:0;left:0;grid-column:auto!important;overflow:hidden;border-bottom-right-radius:5px;border-bottom-left-radius:5px}.swal2-timer-progress-bar{width:100%;height:.25em;background:rgba(0,0,0,.2)}.swal2-image{max-width:100%;margin:2em auto 1em}.swal2-close{z-index:2;align-items:center;justify-content:center;width:1.2em;height:1.2em;margin-top:0;margin-right:0;margin-bottom:-1.2em;padding:0;overflow:hidden;transition:color .1s,box-shadow .1s;border:none;border-radius:5px;background:0 0;color:#ccc;font-family:serif;font-family:monospace;font-size:2.5em;cursor:pointer;justify-self:end}.swal2-close:hover{transform:none;background:0 0;color:#f27474}.swal2-close:focus{outline:0;box-shadow:inset 0 0 0 3px rgba(100,150,200,.5)}.swal2-close::-moz-focus-inner{border:0}.swal2-html-container{z-index:1;justify-content:center;margin:1em 1.6em .3em;padding:0;overflow:auto;color:inherit;font-size:1.125em;font-weight:400;line-height:normal;text-align:center;word-wrap:break-word;word-break:break-word}.swal2-checkbox,.swal2-file,.swal2-input,.swal2-radio,.swal2-select,.swal2-textarea{margin:1em 2em 3px}.swal2-file,.swal2-input,.swal2-textarea{box-sizing:border-box;width:auto;transition:border-color .1s,box-shadow .1s;border:1px solid #d9d9d9;border-radius:.1875em;background:0 0;box-shadow:inset 0 1px 1px rgba(0,0,0,.06),0 0 0 3px transparent;color:inherit;font-size:1.125em}.swal2-file.swal2-inputerror,.swal2-input.swal2-inputerror,.swal2-textarea.swal2-inputerror{border-color:#f27474!important;box-shadow:0 0 2px #f27474!important}.swal2-file:focus,.swal2-input:focus,.swal2-textarea:focus{border:1px solid #b4dbed;outline:0;box-shadow:inset 0 1px 1px rgba(0,0,0,.06),0 0 0 3px rgba(100,150,200,.5)}.swal2-file::-moz-placeholder,.swal2-input::-moz-placeholder,.swal2-textarea::-moz-placeholder{color:#ccc}.swal2-file::placeholder,.swal2-input::placeholder,.swal2-textarea::placeholder{color:#ccc}.swal2-range{margin:1em 2em 3px;background:#fff}.swal2-range input{width:80%}.swal2-range output{width:20%;color:inherit;font-weight:600;text-align:center}.swal2-range input,.swal2-range output{height:2.625em;padding:0;font-size:1.125em;line-height:2.625em}.swal2-input{height:2.625em;padding:0 .75em}.swal2-file{width:75%;margin-right:auto;margin-left:auto;background:0 0;font-size:1.125em}.swal2-textarea{height:6.75em;padding:.75em}.swal2-select{min-width:50%;max-width:100%;padding:.375em .625em;background:0 0;color:inherit;font-size:1.125em}.swal2-checkbox,.swal2-radio{align-items:center;justify-content:center;background:#fff;color:inherit}.swal2-checkbox label,.swal2-radio label{margin:0 .6em;font-size:1.125em}.swal2-checkbox input,.swal2-radio input{flex-shrink:0;margin:0 .4em}.swal2-input-label{display:flex;justify-content:center;margin:1em auto 0}.swal2-validation-message{align-items:center;justify-content:center;margin:1em 0 0;padding:.625em;overflow:hidden;background:#f0f0f0;color:#666;font-size:1em;font-weight:300}.swal2-validation-message::before{content:"!";display:inline-block;width:1.5em;min-width:1.5em;height:1.5em;margin:0 .625em;border-radius:50%;background-color:#f27474;color:#fff;font-weight:600;line-height:1.5em;text-align:center}.swal2-icon{position:relative;box-sizing:content-box;justify-content:center;width:5em;height:5em;margin:2.5em auto .6em;border:.25em solid transparent;border-radius:50%;border-color:#000;font-family:inherit;line-height:5em;cursor:default;-webkit-user-select:none;-moz-user-select:none;user-select:none}.swal2-icon .swal2-icon-content{display:flex;align-items:center;font-size:3.75em}.swal2-icon.swal2-error{border-color:#f27474;color:#f27474}.swal2-icon.swal2-error .swal2-x-mark{position:relative;flex-grow:1}.swal2-icon.swal2-error [class^=swal2-x-mark-line]{display:block;position:absolute;top:2.3125em;width:2.9375em;height:.3125em;border-radius:.125em;background-color:#f27474}.swal2-icon.swal2-error [class^=swal2-x-mark-line][class$=left]{left:1.0625em;transform:rotate(45deg)}.swal2-icon.swal2-error [class^=swal2-x-mark-line][class$=right]{right:1em;transform:rotate(-45deg)}.swal2-icon.swal2-error.swal2-icon-show{-webkit-animation:swal2-animate-error-icon .5s;animation:swal2-animate-error-icon .5s}.swal2-icon.swal2-error.swal2-icon-show .swal2-x-mark{-webkit-animation:swal2-animate-error-x-mark .5s;animation:swal2-animate-error-x-mark .5s}.swal2-icon.swal2-warning{border-color:#facea8;color:#f8bb86}.swal2-icon.swal2-warning.swal2-icon-show{-webkit-animation:swal2-animate-error-icon .5s;animation:swal2-animate-error-icon .5s}.swal2-icon.swal2-warning.swal2-icon-show .swal2-icon-content{-webkit-animation:swal2-animate-i-mark .5s;animation:swal2-animate-i-mark .5s}.swal2-icon.swal2-info{border-color:#9de0f6;color:#3fc3ee}.swal2-icon.swal2-info.swal2-icon-show{-webkit-animation:swal2-animate-error-icon .5s;animation:swal2-animate-error-icon .5s}.swal2-icon.swal2-info.swal2-icon-show .swal2-icon-content{-webkit-animation:swal2-animate-i-mark .8s;animation:swal2-animate-i-mark .8s}.swal2-icon.swal2-question{border-color:#c9dae1;color:#87adbd}.swal2-icon.swal2-question.swal2-icon-show{-webkit-animation:swal2-animate-error-icon .5s;animation:swal2-animate-error-icon .5s}.swal2-icon.swal2-question.swal2-icon-show .swal2-icon-content{-webkit-animation:swal2-animate-question-mark .8s;animation:swal2-animate-question-mark .8s}.swal2-icon.swal2-success{border-color:#a5dc86;color:#a5dc86}.swal2-icon.swal2-success [class^=swal2-success-circular-line]{position:absolute;width:3.75em;height:7.5em;transform:rotate(45deg);border-radius:50%}.swal2-icon.swal2-success [class^=swal2-success-circular-line][class$=left]{top:-.4375em;left:-2.0635em;transform:rotate(-45deg);transform-origin:3.75em 3.75em;border-radius:7.5em 0 0 7.5em}.swal2-icon.swal2-success [class^=swal2-success-circular-line][class$=right]{top:-.6875em;left:1.875em;transform:rotate(-45deg);transform-origin:0 3.75em;border-radius:0 7.5em 7.5em 0}.swal2-icon.swal2-success .swal2-success-ring{position:absolute;z-index:2;top:-.25em;left:-.25em;box-sizing:content-box;width:100%;height:100%;border:.25em solid rgba(165,220,134,.3);border-radius:50%}.swal2-icon.swal2-success .swal2-success-fix{position:absolute;z-index:1;top:.5em;left:1.625em;width:.4375em;height:5.625em;transform:rotate(-45deg)}.swal2-icon.swal2-success [class^=swal2-success-line]{display:block;position:absolute;z-index:2;height:.3125em;border-radius:.125em;background-color:#a5dc86}.swal2-icon.swal2-success [class^=swal2-success-line][class$=tip]{top:2.875em;left:.8125em;width:1.5625em;transform:rotate(45deg)}.swal2-icon.swal2-success [class^=swal2-success-line][class$=long]{top:2.375em;right:.5em;width:2.9375em;transform:rotate(-45deg)}.swal2-icon.swal2-success.swal2-icon-show .swal2-success-line-tip{-webkit-animation:swal2-animate-success-line-tip .75s;animation:swal2-animate-success-line-tip .75s}.swal2-icon.swal2-success.swal2-icon-show .swal2-success-line-long{-webkit-animation:swal2-animate-success-line-long .75s;animation:swal2-animate-success-line-long .75s}.swal2-icon.swal2-success.swal2-icon-show .swal2-success-circular-line-right{-webkit-animation:swal2-rotate-success-circular-line 4.25s ease-in;animation:swal2-rotate-success-circular-line 4.25s ease-in}.swal2-progress-steps{flex-wrap:wrap;align-items:center;max-width:100%;margin:1.25em auto;padding:0;background:0 0;font-weight:600}.swal2-progress-steps li{display:inline-block;position:relative}.swal2-progress-steps .swal2-progress-step{z-index:20;flex-shrink:0;width:2em;height:2em;border-radius:2em;background:#2778c4;color:#fff;line-height:2em;text-align:center}.swal2-progress-steps .swal2-progress-step.swal2-active-progress-step{background:#2778c4}.swal2-progress-steps .swal2-progress-step.swal2-active-progress-step~.swal2-progress-step{background:#add8e6;color:#fff}.swal2-progress-steps .swal2-progress-step.swal2-active-progress-step~.swal2-progress-step-line{background:#add8e6}.swal2-progress-steps .swal2-progress-step-line{z-index:10;flex-shrink:0;width:2.5em;height:.4em;margin:0 -1px;background:#2778c4}[class^=swal2]{-webkit-tap-highlight-color:transparent}.swal2-show{-webkit-animation:swal2-show .3s;animation:swal2-show .3s}.swal2-hide{-webkit-animation:swal2-hide .15s forwards;animation:swal2-hide .15s forwards}.swal2-noanimation{transition:none}.swal2-scrollbar-measure{position:absolute;top:-9999px;width:50px;height:50px;overflow:scroll}.swal2-rtl .swal2-close{margin-right:initial;margin-left:0}.swal2-rtl .swal2-timer-progress-bar{right:0;left:auto}.leave-russia-now-and-apply-your-skills-to-the-world{display:flex;position:fixed;z-index:1939;top:0;right:0;bottom:0;left:0;flex-direction:column;align-items:center;justify-content:center;padding:25px 0 20px;background:#20232a;color:#fff;text-align:center}.leave-russia-now-and-apply-your-skills-to-the-world div{max-width:560px;margin:10px;line-height:146%}.leave-russia-now-and-apply-your-skills-to-the-world iframe{max-width:100%;max-height:55.5555555556vmin;margin:16px auto}.leave-russia-now-and-apply-your-skills-to-the-world strong{border-bottom:2px dashed #fff}.leave-russia-now-and-apply-your-skills-to-the-world button{display:flex;position:fixed;z-index:1940;top:0;right:0;align-items:center;justify-content:center;width:48px;height:48px;margin-right:10px;margin-bottom:-10px;border:none;background:0 0;color:#aaa;font-size:48px;font-weight:700;cursor:pointer}.leave-russia-now-and-apply-your-skills-to-the-world button:hover{color:#fff}@-webkit-keyframes swal2-toast-show{0%{transform:translateY(-.625em) rotateZ(2deg)}33%{transform:translateY(0) rotateZ(-2deg)}66%{transform:translateY(.3125em) rotateZ(2deg)}100%{transform:translateY(0) rotateZ(0)}}@keyframes swal2-toast-show{0%{transform:translateY(-.625em) rotateZ(2deg)}33%{transform:translateY(0) rotateZ(-2deg)}66%{transform:translateY(.3125em) rotateZ(2deg)}100%{transform:translateY(0) rotateZ(0)}}@-webkit-keyframes swal2-toast-hide{100%{transform:rotateZ(1deg);opacity:0}}@keyframes swal2-toast-hide{100%{transform:rotateZ(1deg);opacity:0}}@-webkit-keyframes swal2-toast-animate-success-line-tip{0%{top:.5625em;left:.0625em;width:0}54%{top:.125em;left:.125em;width:0}70%{top:.625em;left:-.25em;width:1.625em}84%{top:1.0625em;left:.75em;width:.5em}100%{top:1.125em;left:.1875em;width:.75em}}@keyframes swal2-toast-animate-success-line-tip{0%{top:.5625em;left:.0625em;width:0}54%{top:.125em;left:.125em;width:0}70%{top:.625em;left:-.25em;width:1.625em}84%{top:1.0625em;left:.75em;width:.5em}100%{top:1.125em;left:.1875em;width:.75em}}@-webkit-keyframes swal2-toast-animate-success-line-long{0%{top:1.625em;right:1.375em;width:0}65%{top:1.25em;right:.9375em;width:0}84%{top:.9375em;right:0;width:1.125em}100%{top:.9375em;right:.1875em;width:1.375em}}@keyframes swal2-toast-animate-success-line-long{0%{top:1.625em;right:1.375em;width:0}65%{top:1.25em;right:.9375em;width:0}84%{top:.9375em;right:0;width:1.125em}100%{top:.9375em;right:.1875em;width:1.375em}}@-webkit-keyframes swal2-show{0%{transform:scale(.7)}45%{transform:scale(1.05)}80%{transform:scale(.95)}100%{transform:scale(1)}}@keyframes swal2-show{0%{transform:scale(.7)}45%{transform:scale(1.05)}80%{transform:scale(.95)}100%{transform:scale(1)}}@-webkit-keyframes swal2-hide{0%{transform:scale(1);opacity:1}100%{transform:scale(.5);opacity:0}}@keyframes swal2-hide{0%{transform:scale(1);opacity:1}100%{transform:scale(.5);opacity:0}}@-webkit-keyframes swal2-animate-success-line-tip{0%{top:1.1875em;left:.0625em;width:0}54%{top:1.0625em;left:.125em;width:0}70%{top:2.1875em;left:-.375em;width:3.125em}84%{top:3em;left:1.3125em;width:1.0625em}100%{top:2.8125em;left:.8125em;width:1.5625em}}@keyframes swal2-animate-success-line-tip{0%{top:1.1875em;left:.0625em;width:0}54%{top:1.0625em;left:.125em;width:0}70%{top:2.1875em;left:-.375em;width:3.125em}84%{top:3em;left:1.3125em;width:1.0625em}100%{top:2.8125em;left:.8125em;width:1.5625em}}@-webkit-keyframes swal2-animate-success-line-long{0%{top:3.375em;right:2.875em;width:0}65%{top:3.375em;right:2.875em;width:0}84%{top:2.1875em;right:0;width:3.4375em}100%{top:2.375em;right:.5em;width:2.9375em}}@keyframes swal2-animate-success-line-long{0%{top:3.375em;right:2.875em;width:0}65%{top:3.375em;right:2.875em;width:0}84%{top:2.1875em;right:0;width:3.4375em}100%{top:2.375em;right:.5em;width:2.9375em}}@-webkit-keyframes swal2-rotate-success-circular-line{0%{transform:rotate(-45deg)}5%{transform:rotate(-45deg)}12%{transform:rotate(-405deg)}100%{transform:rotate(-405deg)}}@keyframes swal2-rotate-success-circular-line{0%{transform:rotate(-45deg)}5%{transform:rotate(-45deg)}12%{transform:rotate(-405deg)}100%{transform:rotate(-405deg)}}@-webkit-keyframes swal2-animate-error-x-mark{0%{margin-top:1.625em;transform:scale(.4);opacity:0}50%{margin-top:1.625em;transform:scale(.4);opacity:0}80%{margin-top:-.375em;transform:scale(1.15)}100%{margin-top:0;transform:scale(1);opacity:1}}@keyframes swal2-animate-error-x-mark{0%{margin-top:1.625em;transform:scale(.4);opacity:0}50%{margin-top:1.625em;transform:scale(.4);opacity:0}80%{margin-top:-.375em;transform:scale(1.15)}100%{margin-top:0;transform:scale(1);opacity:1}}@-webkit-keyframes swal2-animate-error-icon{0%{transform:rotateX(100deg);opacity:0}100%{transform:rotateX(0);opacity:1}}@keyframes swal2-animate-error-icon{0%{transform:rotateX(100deg);opacity:0}100%{transform:rotateX(0);opacity:1}}@-webkit-keyframes swal2-rotate-loading{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes swal2-rotate-loading{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@-webkit-keyframes swal2-animate-question-mark{0%{transform:rotateY(-360deg)}100%{transform:rotateY(0)}}@keyframes swal2-animate-question-mark{0%{transform:rotateY(-360deg)}100%{transform:rotateY(0)}}@-webkit-keyframes swal2-animate-i-mark{0%{transform:rotateZ(45deg);opacity:0}25%{transform:rotateZ(-25deg);opacity:.4}50%{transform:rotateZ(15deg);opacity:.8}75%{transform:rotateZ(-5deg);opacity:1}100%{transform:rotateX(0);opacity:1}}@keyframes swal2-animate-i-mark{0%{transform:rotateZ(45deg);opacity:0}25%{transform:rotateZ(-25deg);opacity:.4}50%{transform:rotateZ(15deg);opacity:.8}75%{transform:rotateZ(-5deg);opacity:1}100%{transform:rotateX(0);opacity:1}}body.swal2-shown:not(.swal2-no-backdrop):not(.swal2-toast-shown){overflow:hidden}body.swal2-height-auto{height:auto!important}body.swal2-no-backdrop .swal2-container{background-color:transparent!important;pointer-events:none}body.swal2-no-backdrop .swal2-container .swal2-popup{pointer-events:all}body.swal2-no-backdrop .swal2-container .swal2-modal{box-shadow:0 0 10px rgba(0,0,0,.4)}@media print{body.swal2-shown:not(.swal2-no-backdrop):not(.swal2-toast-shown){overflow-y:scroll!important}body.swal2-shown:not(.swal2-no-backdrop):not(.swal2-toast-shown)>[aria-hidden=true]{display:none}body.swal2-shown:not(.swal2-no-backdrop):not(.swal2-toast-shown) .swal2-container{position:static!important}}body.swal2-toast-shown .swal2-container{box-sizing:border-box;width:360px;max-width:100%;background-color:transparent;pointer-events:none}body.swal2-toast-shown .swal2-container.swal2-top{top:0;right:auto;bottom:auto;left:50%;transform:translateX(-50%)}body.swal2-toast-shown .swal2-container.swal2-top-end,body.swal2-toast-shown .swal2-container.swal2-top-right{top:0;right:0;bottom:auto;left:auto}body.swal2-toast-shown .swal2-container.swal2-top-left,body.swal2-toast-shown .swal2-container.swal2-top-start{top:0;right:auto;bottom:auto;left:0}body.swal2-toast-shown .swal2-container.swal2-center-left,body.swal2-toast-shown .swal2-container.swal2-center-start{top:50%;right:auto;bottom:auto;left:0;transform:translateY(-50%)}body.swal2-toast-shown .swal2-container.swal2-center{top:50%;right:auto;bottom:auto;left:50%;transform:translate(-50%,-50%)}body.swal2-toast-shown .swal2-container.swal2-center-end,body.swal2-toast-shown .swal2-container.swal2-center-right{top:50%;right:0;bottom:auto;left:auto;transform:translateY(-50%)}body.swal2-toast-shown .swal2-container.swal2-bottom-left,body.swal2-toast-shown .swal2-container.swal2-bottom-start{top:auto;right:auto;bottom:0;left:0}body.swal2-toast-shown .swal2-container.swal2-bottom{top:auto;right:auto;bottom:0;left:50%;transform:translateX(-50%)}body.swal2-toast-shown .swal2-container.swal2-bottom-end,body.swal2-toast-shown .swal2-container.swal2-bottom-right{top:auto;right:0;bottom:0;left:auto}');
})(kt);
const k = kt.exports;
function Zo(h) {
  const O = typeof h == "string" ? { titleText: h } : h;
  return k.fire({
    icon: "success",
    toast: !0,
    timer: 2e3,
    didOpen: (l) => {
      l.addEventListener("mouseenter", k.stopTimer), l.addEventListener("mouseleave", k.resumeTimer);
    },
    position: "top",
    showConfirmButton: !1,
    ...O
  });
}
function Ro(h) {
  const O = typeof h == "string" ? { titleText: h } : h;
  return k.fire({
    icon: "info",
    toast: !0,
    timer: 3e3,
    timerProgressBar: !0,
    didOpen: (l) => {
      l.addEventListener("mouseenter", k.stopTimer), l.addEventListener("mouseleave", k.resumeTimer);
    },
    showConfirmButton: !1,
    ...O
  });
}
function Yo(h) {
  const O = typeof h == "string" ? { titleText: h } : h;
  return k.fire({
    icon: "warning",
    backdrop: !1,
    confirmButtonColor: "#66b1ff",
    timer: 5e3,
    timerProgressBar: !0,
    didOpen: (l) => {
      l.addEventListener("mouseenter", k.stopTimer), l.addEventListener("mouseleave", k.resumeTimer);
    },
    ...O
  });
}
function Fo(h) {
  const O = typeof h == "string" ? { titleText: h } : h;
  return k.fire({
    icon: "error",
    allowOutsideClick: !1,
    confirmButtonColor: "#66b1ff",
    ...O
  });
}
function Uo(h) {
  const O = typeof h == "string" ? { titleText: h } : h;
  return new Promise((l, P) => {
    k.fire({
      confirmButtonText: "\u786E\u5B9A",
      cancelButtonText: "\u53D6\u6D88",
      showCancelButton: !0,
      reverseButtons: !0,
      confirmButtonColor: "#66b1ff",
      allowOutsideClick: !1,
      ...O
    }).then((s) => {
      s.isConfirmed ? l(s) : P(s);
    });
  });
}
k.success = Zo;
k.warning = Yo;
k.info = Ro;
k.error = Fo;
k.confirm = Uo;
export {
  Uo as confirm,
  k as default,
  Fo as error,
  Ro as info,
  Zo as success,
  Yo as warning
};
