// axios-shortcut/dist/axios-shortcut.es.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b2) => {
  for (var prop in b2 || (b2 = {}))
    if (__hasOwnProp.call(b2, prop))
      __defNormalProp(a, prop, b2[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b2)) {
      if (__propIsEnum.call(b2, prop))
        __defNormalProp(a, prop, b2[prop]);
    }
  return a;
};
var __spreadProps = (a, b2) => __defProps(a, __getOwnPropDescs(b2));
import { stringify } from "qs";
var freeGlobal = typeof global == "object" && global && global.Object === Object && global;
var freeGlobal$1 = freeGlobal;
var freeSelf = typeof self == "object" && self && self.Object === Object && self;
var root = freeGlobal$1 || freeSelf || Function("return this")();
var root$1 = root;
var Symbol$1 = root$1.Symbol;
var Symbol$2 = Symbol$1;
var objectProto$2 = Object.prototype;
var hasOwnProperty$1 = objectProto$2.hasOwnProperty;
var nativeObjectToString$1 = objectProto$2.toString;
var symToStringTag$1 = Symbol$2 ? Symbol$2.toStringTag : void 0;
function getRawTag(value) {
  var isOwn = hasOwnProperty$1.call(value, symToStringTag$1), tag = value[symToStringTag$1];
  try {
    value[symToStringTag$1] = void 0;
    var unmasked = true;
  } catch (e) {
  }
  var result = nativeObjectToString$1.call(value);
  if (unmasked) {
    if (isOwn) {
      value[symToStringTag$1] = tag;
    } else {
      delete value[symToStringTag$1];
    }
  }
  return result;
}
var objectProto$1 = Object.prototype;
var nativeObjectToString = objectProto$1.toString;
function objectToString(value) {
  return nativeObjectToString.call(value);
}
var nullTag = "[object Null]", undefinedTag = "[object Undefined]";
var symToStringTag = Symbol$2 ? Symbol$2.toStringTag : void 0;
function baseGetTag(value) {
  if (value == null) {
    return value === void 0 ? undefinedTag : nullTag;
  }
  return symToStringTag && symToStringTag in Object(value) ? getRawTag(value) : objectToString(value);
}
function isObjectLike(value) {
  return value != null && typeof value == "object";
}
function identity(value) {
  return value;
}
function overArg(func, transform) {
  return function(arg) {
    return func(transform(arg));
  };
}
var getPrototype = overArg(Object.getPrototypeOf, Object);
var getPrototype$1 = getPrototype;
var objectTag = "[object Object]";
var funcProto = Function.prototype, objectProto = Object.prototype;
var funcToString = funcProto.toString;
var hasOwnProperty = objectProto.hasOwnProperty;
var objectCtorString = funcToString.call(Object);
function isPlainObject(value) {
  if (!isObjectLike(value) || baseGetTag(value) != objectTag) {
    return false;
  }
  var proto = getPrototype$1(value);
  if (proto === null) {
    return true;
  }
  var Ctor = hasOwnProperty.call(proto, "constructor") && proto.constructor;
  return typeof Ctor == "function" && Ctor instanceof Ctor && funcToString.call(Ctor) == objectCtorString;
}
typeof require != "undefined" && require;
var h = (e) => ({ object: () => e === null || Array.isArray(e) && e.length === 0 || isPlainObject(e) && Object.getOwnPropertyNames(e).length === 0, number: () => Number.isNaN(e), string: () => e === "", undefined: () => true, boolean: () => e === false, symbol: () => false, bigint: () => false })[typeof e](), m = (e) => !h(e);
function b(e) {
  return {}.toString.call(e).slice(8, -1).toLowerCase();
}
var g = (e, t = identity) => {
  const r = new FormData();
  let n = true;
  for (let o in e) {
    const i = t(e[o], o);
    ["string", "blob", "file"].includes(b(i)) || (n = false), i !== void 0 && r.append(o, i);
  }
  return n || console.warn("[Kayran] FormData\u7684value\u7C7B\u578B\u4EC5\u80FD\u4E3Astring\u6216blob"), r;
};
var N, j;
(function() {
  for (var e = [], t = 0; t < 256; ++t)
    e.push("%" + ((t < 16 ? "0" : "") + t.toString(16)).toUpperCase());
  return e;
})();
(j = N || (N = {})).history = "history", j.hash = "hash";
const te = (e) => {
  let t = "";
  return m(e) && (/^(((ht|f)tps?):\/\/)?[\w-]+(\.[\w-]+)+([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E")), t;
};
Object.freeze({ __proto__: null, [Symbol.toStringTag]: "Module", lng: (e) => {
  let t = "";
  return typeof e == "number" && Number.isNaN(e) ? t = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && (e = e.toString(), /^-?(\d+|\d+\.\d+)$/.test(e) ? ((e = Number(e)) > 180 || e < -180) && (t = "\u7ECF\u5EA6\u7684\u8303\u56F4\u4E3A[-180, 180]") : t = "\u683C\u5F0F\u4E0D\u6B63\u786E"), t;
}, lat: (e) => {
  let t = "";
  return typeof e == "number" && Number.isNaN(e) ? t = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && (e = e.toString(), /^-?(\d+|\d+\.\d+)$/.test(e) ? ((e = Number(e)) > 90 || e < -90) && (t = "\u7EAC\u5EA6\u7684\u8303\u56F4\u4E3A[-90, 90]") : t = "\u683C\u5F0F\u4E0D\u6B63\u786E"), t;
}, int: (e, t) => {
  let r = "";
  if (typeof e == "number" && Number.isNaN(e))
    r = "\u683C\u5F0F\u4E0D\u6B63\u786E";
  else if (m(e)) {
    let n, o, i, a, l, c;
    if (typeof t == "object" ? (l = t.range, c = t.positiveSign) : l = t, c || typeof e != "number" || console.warn("[kayran] positiveSign\u4EC5\u5BF9\u5B57\u7B26\u4E32\u751F\u6548"), l) {
      if (!/^[\[(](-?\d*|-∞),(-?\d*|\+∞)[\])]$/.test(l))
        throw Error("[kayran] range\u53C2\u6570\u683C\u5F0F\u4E0D\u6B63\u786E");
      const [e2, t2] = l.substr(1, l.length - 2).split(",");
      ["", "-\u221E"].includes(e2) || (l.startsWith("(") ? i = e2 : n = e2), ["", "+\u221E"].includes(t2) || (l.endsWith(")") ? a = t2 : o = t2);
    }
    const s = Number(e), f = e.toString();
    switch (true) {
      case !(c ? /^[+\-]?\d+$/ : /^\-?\d+$/).test(f):
        r = "\u9700\u4E3A\u6574\u6570";
        break;
      case (m(n) && s < n):
        r = "\u4E0D\u80FD\u5C0F\u4E8E" + n;
        break;
      case (m(o) && s > o):
        r = "\u4E0D\u80FD\u5927\u4E8E" + o;
        break;
      case (m(i) && s <= i):
        r = "\u9700\u5927\u4E8E" + i;
        break;
      case (m(a) && s >= a):
        r = "\u9700\u5C0F\u4E8E" + a;
    }
  }
  return r;
}, decimal: (e, t) => {
  var r, n, o, i, a;
  let l = "";
  if (typeof e == "number" && Number.isNaN(e))
    l = "\u683C\u5F0F\u4E0D\u6B63\u786E";
  else if (m(e)) {
    let c, s, f, d, u, p;
    if (typeof t == "object" ? (c = t.range, s = t.decimalPlaces) : c = t, c) {
      if (!/^[\[(](-?\d*\.?\d*|-∞),(-?\d*\.?\d*|\+∞)[\])]$/.test(c))
        throw Error("[kayran] range\u53C2\u6570\u683C\u5F0F\u4E0D\u6B63\u786E");
      const [e2, t2] = c.substr(1, c.length - 2).split(",");
      ["", "-\u221E"].includes(e2) || (c.startsWith("(") ? u = e2 : f = e2), ["", "+\u221E"].includes(t2) || (c.endsWith(")") ? p = t2 : d = t2);
    }
    const y = Number(e), h2 = e.toString();
    switch (true) {
      case !/^-?(\d+|\d+\.\d+)$/.test(h2):
        l = "\u9700\u4E3A\u6570\u5B57";
        break;
      case (m(f) && y < f):
        l = "\u4E0D\u80FD\u5C0F\u4E8E" + f;
        break;
      case (m(d) && y > d):
        l = "\u4E0D\u80FD\u5927\u4E8E" + d;
        break;
      case (m(u) && y <= u):
        l = "\u9700\u5927\u4E8E" + u;
        break;
      case (m(p) && y >= p):
        l = "\u9700\u5C0F\u4E8E" + p;
        break;
      default:
        const e2 = h2.split(".");
        if (typeof s == "number")
          switch (true) {
            case s < 0:
              throw Error("decimalPlaces\u4E0D\u80FD\u5C0F\u4E8E0");
            case s === 0:
              m(e2[1]) && (l = "\u9700\u4E3A\u6574\u6570");
              break;
            default:
              ((r = e2[1]) == null ? void 0 : r.length) !== s && (l = "\u5C0F\u6570\u4F4D\u6570\u9700\u4E3A" + s + "\u4F4D");
          }
        else if (Array.isArray(s)) {
          const [t2, r2] = s;
          if (r2 < t2)
            throw Error("[kayran] decimalPlaces\u7684\u4E0A\u9650\u4E0D\u80FD\u5C0F\u4E8E\u4E0B\u9650");
          if (Number.isNaN(r2))
            throw Error("[kayran] decimalPlaces[1]\u4E0D\u80FD\u4E3ANaN");
          if (r2 < 0)
            throw Error("[kayran] decimalPlaces\u7684\u4E0A\u9650\u4E0D\u80FD\u4E3A\u8D1F");
          if (((o = (n = e2[1]) == null ? void 0 : n.length) != null ? o : 0) > r2 && (l = "\u6700\u591A" + r2 + "\u4F4D\u5C0F\u6570"), Number.isNaN(t2))
            throw Error("[kayran] decimalPlaces[0]\u4E0D\u80FD\u4E3ANaN");
          if (t2 < 0)
            throw Error("[kayran] decimalPlaces[0]\u4E0D\u80FD\u4E3A\u8D1F");
          ((a = (i = e2[1]) == null ? void 0 : i.length) != null ? a : 0) < t2 && (l = "\u6700\u5C11" + t2 + "\u4F4D\u5C0F\u6570");
        }
    }
  }
  return l;
}, len: (e, t) => {
  let r, n, o, i = "";
  if (typeof t == "number") {
    if (Number.isNaN(t))
      throw Error("[kayran] options\u4E0D\u80FD\u4E3ANaN");
    r = t;
  } else
    Array.isArray(t) ? (n = t[0], o = t[1]) : t && typeof t == "object" && (n = t.min, o = t.max);
  if (Number.isNaN(n))
    throw Error("[kayran] min\u4E0D\u80FD\u4E3ANaN");
  if (Number.isNaN(o))
    throw Error("[kayran] max\u4E0D\u80FD\u4E3ANaN");
  if (o < 0)
    throw Error("[kayran] max\u4E0D\u80FD\u4E3A\u8D1F");
  if (n < 0)
    throw Error("[kayran] min\u4E0D\u80FD\u4E3A\u8D1F");
  if (o < n)
    throw Error("[kayran] max\u4E0D\u80FD\u5C0F\u4E8Emin");
  if (h(r) && h(n) && h(o))
    throw Error("[kayran] \u672A\u6307\u5B9A\u4EFB\u4F55\u6821\u9A8C\u6761\u4EF6");
  return (e === "" || m(e)) && (e.length > o ? i = "\u4E0D\u80FD\u8D85\u8FC7" + o + "\u4E2A\u5B57\u7B26" : n && e.length < n ? i = "\u4E0D\u80FD\u4F4E\u4E8E" + n + "\u4E2A\u5B57\u7B26" : m(r) && e.length !== r && (i = "\u5B57\u7B26\u6570\u9700\u4E3A" + n)), i;
}, phone: (e) => {
  let t = "";
  return typeof e == "number" && Number.isNaN(e) ? t = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && (e = e.toString(), /^(?:(?:\+|00)86)?1[3-9]\d{9}$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E")), t;
}, tel: (e, { multiple: t = true } = {}) => {
  const r = t ? /^((?:\d{3}-)?\d{8};?)+$|^((?:\d{4}-)?\d{7,8};?)+$/ : /^(?:\d{3}-)?\d{8}$|^(?:\d{4}-)?\d{7,8}$/;
  let n = "";
  return typeof e == "number" && Number.isNaN(e) ? n = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && ((e = e.toString()).length > 50 ? n = "\u4E0D\u80FD\u8D85\u8FC750\u4E2A\u5B57\u7B26" : r.test(e) || (n = "\u683C\u5F0F\u4E0D\u6B63\u786E" + (t ? "\uFF0C\u5982\u6709\u591A\u4E2A\u8BF7\u7528\u82F1\u6587\u5206\u53F7\u9694\u5F00" : ""))), n;
}, email: (e) => {
  let t = "";
  return /^[\u4e00-\u9fa5\dA-Za-z_-]+@[\dA-Za-z_-]+(\.[\dA-Za-z_-]+)+$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E"), t;
}, idCard: (e) => {
  let t = "";
  return typeof e == "number" && Number.isNaN(e) ? t = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && (e = e.toString(), /^[1-9]\d{5}(?:18|19|20)\d{2}(?:0[1-9]|10|11|12)(?:0[1-9]|[1-2]\d|30|31)\d{3}[\dXx]$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E")), t;
}, ipv4: (e) => {
  let t = "";
  return /^((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])(?::(?:[0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]))?$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E"), t;
}, ipv6: (e) => {
  let t = "";
  return /^(?:(?:(?:[0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}:[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){5}:([0-9A-Fa-f]{1,4}:)?[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){4}:([0-9A-Fa-f]{1,4}:){0,2}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){3}:([0-9A-Fa-f]{1,4}:){0,3}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){2}:([0-9A-Fa-f]{1,4}:){0,4}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(([0-9A-Fa-f]{1,4}:){0,5}:((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(::([0-9A-Fa-f]{1,4}:){0,5}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|([0-9A-Fa-f]{1,4}::([0-9A-Fa-f]{1,4}:){0,5}[0-9A-Fa-f]{1,4})|(::([0-9A-Fa-f]{1,4}:){0,6}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){1,7}:))|\[(?:(?:(?:[0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}:[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){5}:([0-9A-Fa-f]{1,4}:)?[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){4}:([0-9A-Fa-f]{1,4}:){0,2}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){3}:([0-9A-Fa-f]{1,4}:){0,3}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){2}:([0-9A-Fa-f]{1,4}:){0,4}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){6}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(([0-9A-Fa-f]{1,4}:){0,5}:((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|(::([0-9A-Fa-f]{1,4}:){0,5}((\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b)\.){3}(\b((25[0-5])|(1\d{2})|(2[0-4]\d)|(\d{1,2}))\b))|([0-9A-Fa-f]{1,4}::([0-9A-Fa-f]{1,4}:){0,5}[0-9A-Fa-f]{1,4})|(::([0-9A-Fa-f]{1,4}:){0,6}[0-9A-Fa-f]{1,4})|(([0-9A-Fa-f]{1,4}:){1,7}:))\](?::(?:[0-9]|[1-9][0-9]{1,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]))?$/i.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E"), t;
}, url: te, postcode: (e) => {
  let t = "";
  return typeof e == "number" && Number.isNaN(e) ? t = "\u683C\u5F0F\u4E0D\u6B63\u786E" : m(e) && (e = e.toString(), /^(0[1-7]|1[0-356]|2[0-7]|3[0-6]|4[0-7]|5[1-7]|6[1-7]|7[0-5]|8[013-6])\d{4}$/.test(e) || (t = "\u683C\u5F0F\u4E0D\u6B63\u786E")), t;
}, base64: (e, { mediaType: t = "", scheme: r = true } = {}) => {
  let n = "";
  if (m(e))
    if (r && !e.startsWith(`data:${t}`))
      n = "\u7F3A\u5931scheme";
    else {
      const t2 = r ? e.split(",")[1] : e;
      t2 && /^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$/.test(t2) || (n = "\u683C\u5F0F\u4E0D\u6B63\u786E");
    }
  return n;
} });
var le, ce;
(ce = le || (le = {})).deep = "deep", ce.shallow = "shallow";
const CONSOLE_PREFIX = "[axios-shortcut] ";
const METHODS_WITH_REQUEST_BODY = ["PUT", "POST", "DELETE", "PATCH"];
const METHODS_WITHOUT_REQUEST_BODY = ["GET", "HEAD"];
const METHODS = METHODS_WITH_REQUEST_BODY.concat(METHODS_WITHOUT_REQUEST_BODY);
const createAxiosShortcut = (axios) => {
  let result = {};
  const download = (url, data, config = {}) => {
    if (config.method) {
      return axios(__spreadValues(__spreadValues({
        responseType: "blob",
        url
      }, METHODS_WITH_REQUEST_BODY.includes(config.method.toUpperCase()) ? { data } : { params: data }), config));
    } else {
      window.open(url + stringify(data, { addQueryPrefix: true }));
    }
  };
  result.DOWNLOAD = download;
  const upload = (url, data, config) => {
    return axios(__spreadValues({
      url,
      method: "POST",
      data: g(data)
    }, config));
  };
  METHODS.map((v) => {
    const value = (url, data, config = {}) => {
      return axios(__spreadValues(__spreadValues({
        method: v,
        url
      }, METHODS_WITH_REQUEST_BODY.includes(v.toUpperCase()) ? { data } : { params: data }), config));
    };
    value.download = function() {
      const [url, data, config] = arguments;
      if (config == null ? void 0 : config.method) {
        console.warn(`${CONSOLE_PREFIX}method\u65E0\u6CD5\u91CD\u590D\u6307\u5B9A`);
      }
      return download(url, data, __spreadProps(__spreadValues({}, config), {
        method: v
      }));
    };
    value.upload = function() {
      const [url, data, config] = arguments;
      if (config == null ? void 0 : config.method) {
        console.warn(`${CONSOLE_PREFIX}method\u65E0\u6CD5\u91CD\u590D\u6307\u5B9A`);
      }
      return upload(url, data, __spreadProps(__spreadValues({}, config), {
        method: v
      }));
    };
    result[v] = value;
  });
  return result;
};
export { createAxiosShortcut as default };
