// kikimore/dist/kikimore.mjs
import Ar from "vue";
Ar.util.warn;
function Bt(t, e = []) {
  const r = {}, n = {}, o = {}, i = {}, a = Array.isArray(e) ? e : Object.keys(e);
  for (const s in t)
    if (s.startsWith("@")) {
      const l = s.substring(1);
      l.startsWith("hook:") ? i[l] = t[s] : o[l] = t[s];
    } else
      a.includes(s) ? r[s] = t[s] : n[s] = t[s];
  return {
    props: r,
    attrs: n,
    listeners: o,
    hooks: i
  };
}
var Un = typeof global == "object" && global && global.Object === Object && global;
const Cr = Un;
var Gn = typeof self == "object" && self && self.Object === Object && self, Nn = Cr || Gn || Function("return this")();
const C = Nn;
var zn = C.Symbol;
const S = zn;
var Er = Object.prototype, Vn = Er.hasOwnProperty, Wn = Er.toString, ct = S ? S.toStringTag : void 0;
function Hn(t) {
  var e = Vn.call(t, ct), r = t[ct];
  try {
    t[ct] = void 0;
    var n = !0;
  } catch {
  }
  var o = Wn.call(t);
  return n && (e ? t[ct] = r : delete t[ct]), o;
}
var Kn = Object.prototype, qn = Kn.toString;
function Jn(t) {
  return qn.call(t);
}
var Zn = "[object Null]", Qn = "[object Undefined]", xe = S ? S.toStringTag : void 0;
function H(t) {
  return t == null ? t === void 0 ? Qn : Zn : xe && xe in Object(t) ? Hn(t) : Jn(t);
}
function A(t) {
  return t != null && typeof t == "object";
}
var Xn = "[object Symbol]";
function le(t) {
  return typeof t == "symbol" || A(t) && H(t) == Xn;
}
function Yn(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length, o = Array(n); ++r < n; )
    o[r] = e(t[r], r, t);
  return o;
}
var to = Array.isArray;
const O = to;
var eo = 1 / 0, Le = S ? S.prototype : void 0, Fe = Le ? Le.toString : void 0;
function xr(t) {
  if (typeof t == "string")
    return t;
  if (O(t))
    return Yn(t, xr) + "";
  if (le(t))
    return Fe ? Fe.call(t) : "";
  var e = t + "";
  return e == "0" && 1 / t == -eo ? "-0" : e;
}
function E(t) {
  var e = typeof t;
  return t != null && (e == "object" || e == "function");
}
function ce(t) {
  return t;
}
var ro = "[object AsyncFunction]", no = "[object Function]", oo = "[object GeneratorFunction]", io = "[object Proxy]";
function ue(t) {
  if (!E(t))
    return !1;
  var e = H(t);
  return e == no || e == oo || e == ro || e == io;
}
var ao = C["__core-js_shared__"];
const Wt = ao;
var Ie = function() {
  var t = /[^.]+$/.exec(Wt && Wt.keys && Wt.keys.IE_PROTO || "");
  return t ? "Symbol(src)_1." + t : "";
}();
function so(t) {
  return !!Ie && Ie in t;
}
var lo = Function.prototype, co = lo.toString;
function K(t) {
  if (t != null) {
    try {
      return co.call(t);
    } catch {
    }
    try {
      return t + "";
    } catch {
    }
  }
  return "";
}
var uo = /[\\^$.*+?()[\]{}|]/g, fo = /^\[object .+?Constructor\]$/, po = Function.prototype, ho = Object.prototype, bo = po.toString, yo = ho.hasOwnProperty, vo = RegExp(
  "^" + bo.call(yo).replace(uo, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function go(t) {
  if (!E(t) || so(t))
    return !1;
  var e = ue(t) ? vo : fo;
  return e.test(K(t));
}
function _o(t, e) {
  return t == null ? void 0 : t[e];
}
function q(t, e) {
  var r = _o(t, e);
  return go(r) ? r : void 0;
}
var mo = q(C, "WeakMap");
const Qt = mo;
var Be = Object.create, jo = function() {
  function t() {
  }
  return function(e) {
    if (!E(e))
      return {};
    if (Be)
      return Be(e);
    t.prototype = e;
    var r = new t();
    return t.prototype = void 0, r;
  };
}();
const wo = jo;
function Po(t, e, r) {
  switch (r.length) {
    case 0:
      return t.call(e);
    case 1:
      return t.call(e, r[0]);
    case 2:
      return t.call(e, r[0], r[1]);
    case 3:
      return t.call(e, r[0], r[1], r[2]);
  }
  return t.apply(e, r);
}
function Lr(t, e) {
  var r = -1, n = t.length;
  for (e || (e = Array(n)); ++r < n; )
    e[r] = t[r];
  return e;
}
var $o = 800, Oo = 16, To = Date.now;
function So(t) {
  var e = 0, r = 0;
  return function() {
    var n = To(), o = Oo - (n - r);
    if (r = n, o > 0) {
      if (++e >= $o)
        return arguments[0];
    } else
      e = 0;
    return t.apply(void 0, arguments);
  };
}
function Ao(t) {
  return function() {
    return t;
  };
}
var Co = function() {
  try {
    var t = q(Object, "defineProperty");
    return t({}, "", {}), t;
  } catch {
  }
}();
const Ct = Co;
var Eo = Ct ? function(t, e) {
  return Ct(t, "toString", {
    configurable: !0,
    enumerable: !1,
    value: Ao(e),
    writable: !0
  });
} : ce;
const xo = Eo;
var Lo = So(xo);
const Fo = Lo;
function Io(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length; ++r < n && e(t[r], r, t) !== !1; )
    ;
  return t;
}
var Bo = 9007199254740991, Do = /^(?:0|[1-9]\d*)$/;
function fe(t, e) {
  var r = typeof t;
  return e = e == null ? Bo : e, !!e && (r == "number" || r != "symbol" && Do.test(t)) && t > -1 && t % 1 == 0 && t < e;
}
function Dt(t, e, r) {
  e == "__proto__" && Ct ? Ct(t, e, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : t[e] = r;
}
function mt(t, e) {
  return t === e || t !== t && e !== e;
}
var Mo = Object.prototype, Ro = Mo.hasOwnProperty;
function Fr(t, e, r) {
  var n = t[e];
  (!(Ro.call(t, e) && mt(n, r)) || r === void 0 && !(e in t)) && Dt(t, e, r);
}
function rt(t, e, r, n) {
  var o = !r;
  r || (r = {});
  for (var i = -1, a = e.length; ++i < a; ) {
    var s = e[i], l = n ? n(r[s], t[s], s, r, t) : void 0;
    l === void 0 && (l = t[s]), o ? Dt(r, s, l) : Fr(r, s, l);
  }
  return r;
}
var De = Math.max;
function ko(t, e, r) {
  return e = De(e === void 0 ? t.length - 1 : e, 0), function() {
    for (var n = arguments, o = -1, i = De(n.length - e, 0), a = Array(i); ++o < i; )
      a[o] = n[e + o];
    o = -1;
    for (var s = Array(e + 1); ++o < e; )
      s[o] = n[o];
    return s[e] = r(a), Po(t, this, s);
  };
}
function Uo(t, e) {
  return Fo(ko(t, e, ce), t + "");
}
var Go = 9007199254740991;
function pe(t) {
  return typeof t == "number" && t > -1 && t % 1 == 0 && t <= Go;
}
function Mt(t) {
  return t != null && pe(t.length) && !ue(t);
}
function No(t, e, r) {
  if (!E(r))
    return !1;
  var n = typeof e;
  return (n == "number" ? Mt(r) && fe(e, r.length) : n == "string" && e in r) ? mt(r[e], t) : !1;
}
function Ir(t) {
  return Uo(function(e, r) {
    var n = -1, o = r.length, i = o > 1 ? r[o - 1] : void 0, a = o > 2 ? r[2] : void 0;
    for (i = t.length > 3 && typeof i == "function" ? (o--, i) : void 0, a && No(r[0], r[1], a) && (i = o < 3 ? void 0 : i, o = 1), e = Object(e); ++n < o; ) {
      var s = r[n];
      s && t(e, s, n, i);
    }
    return e;
  });
}
var zo = Object.prototype;
function he(t) {
  var e = t && t.constructor, r = typeof e == "function" && e.prototype || zo;
  return t === r;
}
function Vo(t, e) {
  for (var r = -1, n = Array(t); ++r < t; )
    n[r] = e(r);
  return n;
}
var Wo = "[object Arguments]";
function Me(t) {
  return A(t) && H(t) == Wo;
}
var Br = Object.prototype, Ho = Br.hasOwnProperty, Ko = Br.propertyIsEnumerable, qo = Me(function() {
  return arguments;
}()) ? Me : function(t) {
  return A(t) && Ho.call(t, "callee") && !Ko.call(t, "callee");
};
const Et = qo;
function Jo() {
  return !1;
}
var Dr = typeof exports == "object" && exports && !exports.nodeType && exports, Re = Dr && typeof module == "object" && module && !module.nodeType && module, Zo = Re && Re.exports === Dr, ke = Zo ? C.Buffer : void 0, Qo = ke ? ke.isBuffer : void 0, Xo = Qo || Jo;
const dt = Xo;
var Yo = "[object Arguments]", ti = "[object Array]", ei = "[object Boolean]", ri = "[object Date]", ni = "[object Error]", oi = "[object Function]", ii = "[object Map]", ai = "[object Number]", si = "[object Object]", li = "[object RegExp]", ci = "[object Set]", ui = "[object String]", fi = "[object WeakMap]", pi = "[object ArrayBuffer]", hi = "[object DataView]", di = "[object Float32Array]", bi = "[object Float64Array]", yi = "[object Int8Array]", vi = "[object Int16Array]", gi = "[object Int32Array]", _i = "[object Uint8Array]", mi = "[object Uint8ClampedArray]", ji = "[object Uint16Array]", wi = "[object Uint32Array]", m = {};
m[di] = m[bi] = m[yi] = m[vi] = m[gi] = m[_i] = m[mi] = m[ji] = m[wi] = !0;
m[Yo] = m[ti] = m[pi] = m[ei] = m[hi] = m[ri] = m[ni] = m[oi] = m[ii] = m[ai] = m[si] = m[li] = m[ci] = m[ui] = m[fi] = !1;
function Pi(t) {
  return A(t) && pe(t.length) && !!m[H(t)];
}
function de(t) {
  return function(e) {
    return t(e);
  };
}
var Mr = typeof exports == "object" && exports && !exports.nodeType && exports, pt = Mr && typeof module == "object" && module && !module.nodeType && module, $i = pt && pt.exports === Mr, Ht = $i && Cr.process, Oi = function() {
  try {
    var t = pt && pt.require && pt.require("util").types;
    return t || Ht && Ht.binding && Ht.binding("util");
  } catch {
  }
}();
const Y = Oi;
var Ue = Y && Y.isTypedArray, Ti = Ue ? de(Ue) : Pi;
const be = Ti;
var Si = Object.prototype, Ai = Si.hasOwnProperty;
function Rr(t, e) {
  var r = O(t), n = !r && Et(t), o = !r && !n && dt(t), i = !r && !n && !o && be(t), a = r || n || o || i, s = a ? Vo(t.length, String) : [], l = s.length;
  for (var c in t)
    (e || Ai.call(t, c)) && !(a && (c == "length" || o && (c == "offset" || c == "parent") || i && (c == "buffer" || c == "byteLength" || c == "byteOffset") || fe(c, l))) && s.push(c);
  return s;
}
function kr(t, e) {
  return function(r) {
    return t(e(r));
  };
}
var Ci = kr(Object.keys, Object);
const Ei = Ci;
var xi = Object.prototype, Li = xi.hasOwnProperty;
function Fi(t) {
  if (!he(t))
    return Ei(t);
  var e = [];
  for (var r in Object(t))
    Li.call(t, r) && r != "constructor" && e.push(r);
  return e;
}
function jt(t) {
  return Mt(t) ? Rr(t) : Fi(t);
}
function Ii(t) {
  var e = [];
  if (t != null)
    for (var r in Object(t))
      e.push(r);
  return e;
}
var Bi = Object.prototype, Di = Bi.hasOwnProperty;
function Mi(t) {
  if (!E(t))
    return Ii(t);
  var e = he(t), r = [];
  for (var n in t)
    n == "constructor" && (e || !Di.call(t, n)) || r.push(n);
  return r;
}
function nt(t) {
  return Mt(t) ? Rr(t, !0) : Mi(t);
}
var Ri = Ir(function(t, e, r, n) {
  rt(e, nt(e), t, n);
});
const ki = Ri;
var Ui = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Gi = /^\w*$/;
function ye(t, e) {
  if (O(t))
    return !1;
  var r = typeof t;
  return r == "number" || r == "symbol" || r == "boolean" || t == null || le(t) ? !0 : Gi.test(t) || !Ui.test(t) || e != null && t in Object(e);
}
var Ni = q(Object, "create");
const bt = Ni;
function zi() {
  this.__data__ = bt ? bt(null) : {}, this.size = 0;
}
function Vi(t) {
  var e = this.has(t) && delete this.__data__[t];
  return this.size -= e ? 1 : 0, e;
}
var Wi = "__lodash_hash_undefined__", Hi = Object.prototype, Ki = Hi.hasOwnProperty;
function qi(t) {
  var e = this.__data__;
  if (bt) {
    var r = e[t];
    return r === Wi ? void 0 : r;
  }
  return Ki.call(e, t) ? e[t] : void 0;
}
var Ji = Object.prototype, Zi = Ji.hasOwnProperty;
function Qi(t) {
  var e = this.__data__;
  return bt ? e[t] !== void 0 : Zi.call(e, t);
}
var Xi = "__lodash_hash_undefined__";
function Yi(t, e) {
  var r = this.__data__;
  return this.size += this.has(t) ? 0 : 1, r[t] = bt && e === void 0 ? Xi : e, this;
}
function V(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
V.prototype.clear = zi;
V.prototype.delete = Vi;
V.prototype.get = qi;
V.prototype.has = Qi;
V.prototype.set = Yi;
function ta() {
  this.__data__ = [], this.size = 0;
}
function Rt(t, e) {
  for (var r = t.length; r--; )
    if (mt(t[r][0], e))
      return r;
  return -1;
}
var ea = Array.prototype, ra = ea.splice;
function na(t) {
  var e = this.__data__, r = Rt(e, t);
  if (r < 0)
    return !1;
  var n = e.length - 1;
  return r == n ? e.pop() : ra.call(e, r, 1), --this.size, !0;
}
function oa(t) {
  var e = this.__data__, r = Rt(e, t);
  return r < 0 ? void 0 : e[r][1];
}
function ia(t) {
  return Rt(this.__data__, t) > -1;
}
function aa(t, e) {
  var r = this.__data__, n = Rt(r, t);
  return n < 0 ? (++this.size, r.push([t, e])) : r[n][1] = e, this;
}
function B(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
B.prototype.clear = ta;
B.prototype.delete = na;
B.prototype.get = oa;
B.prototype.has = ia;
B.prototype.set = aa;
var sa = q(C, "Map");
const yt = sa;
function la() {
  this.size = 0, this.__data__ = {
    hash: new V(),
    map: new (yt || B)(),
    string: new V()
  };
}
function ca(t) {
  var e = typeof t;
  return e == "string" || e == "number" || e == "symbol" || e == "boolean" ? t !== "__proto__" : t === null;
}
function kt(t, e) {
  var r = t.__data__;
  return ca(e) ? r[typeof e == "string" ? "string" : "hash"] : r.map;
}
function ua(t) {
  var e = kt(this, t).delete(t);
  return this.size -= e ? 1 : 0, e;
}
function fa(t) {
  return kt(this, t).get(t);
}
function pa(t) {
  return kt(this, t).has(t);
}
function ha(t, e) {
  var r = kt(this, t), n = r.size;
  return r.set(t, e), this.size += r.size == n ? 0 : 1, this;
}
function D(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
D.prototype.clear = la;
D.prototype.delete = ua;
D.prototype.get = fa;
D.prototype.has = pa;
D.prototype.set = ha;
var da = "Expected a function";
function ve(t, e) {
  if (typeof t != "function" || e != null && typeof e != "function")
    throw new TypeError(da);
  var r = function() {
    var n = arguments, o = e ? e.apply(this, n) : n[0], i = r.cache;
    if (i.has(o))
      return i.get(o);
    var a = t.apply(this, n);
    return r.cache = i.set(o, a) || i, a;
  };
  return r.cache = new (ve.Cache || D)(), r;
}
ve.Cache = D;
var ba = 500;
function ya(t) {
  var e = ve(t, function(n) {
    return r.size === ba && r.clear(), n;
  }), r = e.cache;
  return e;
}
var va = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, ga = /\\(\\)?/g, _a = ya(function(t) {
  var e = [];
  return t.charCodeAt(0) === 46 && e.push(""), t.replace(va, function(r, n, o, i) {
    e.push(o ? i.replace(ga, "$1") : n || r);
  }), e;
});
const ma = _a;
function ja(t) {
  return t == null ? "" : xr(t);
}
function Ur(t, e) {
  return O(t) ? t : ye(t, e) ? [t] : ma(ja(t));
}
var wa = 1 / 0;
function Ut(t) {
  if (typeof t == "string" || le(t))
    return t;
  var e = t + "";
  return e == "0" && 1 / t == -wa ? "-0" : e;
}
function Gr(t, e) {
  e = Ur(e, t);
  for (var r = 0, n = e.length; t != null && r < n; )
    t = t[Ut(e[r++])];
  return r && r == n ? t : void 0;
}
function Pa(t, e, r) {
  var n = t == null ? void 0 : Gr(t, e);
  return n === void 0 ? r : n;
}
function Nr(t, e) {
  for (var r = -1, n = e.length, o = t.length; ++r < n; )
    t[o + r] = e[r];
  return t;
}
var $a = kr(Object.getPrototypeOf, Object);
const ge = $a;
var Oa = "[object Object]", Ta = Function.prototype, Sa = Object.prototype, zr = Ta.toString, Aa = Sa.hasOwnProperty, Ca = zr.call(Object);
function Vr(t) {
  if (!A(t) || H(t) != Oa)
    return !1;
  var e = ge(t);
  if (e === null)
    return !0;
  var r = Aa.call(e, "constructor") && e.constructor;
  return typeof r == "function" && r instanceof r && zr.call(r) == Ca;
}
function Ea() {
  this.__data__ = new B(), this.size = 0;
}
function xa(t) {
  var e = this.__data__, r = e.delete(t);
  return this.size = e.size, r;
}
function La(t) {
  return this.__data__.get(t);
}
function Fa(t) {
  return this.__data__.has(t);
}
var Ia = 200;
function Ba(t, e) {
  var r = this.__data__;
  if (r instanceof B) {
    var n = r.__data__;
    if (!yt || n.length < Ia - 1)
      return n.push([t, e]), this.size = ++r.size, this;
    r = this.__data__ = new D(n);
  }
  return r.set(t, e), this.size = r.size, this;
}
function T(t) {
  var e = this.__data__ = new B(t);
  this.size = e.size;
}
T.prototype.clear = Ea;
T.prototype.delete = xa;
T.prototype.get = La;
T.prototype.has = Fa;
T.prototype.set = Ba;
function Da(t, e) {
  return t && rt(e, jt(e), t);
}
function Ma(t, e) {
  return t && rt(e, nt(e), t);
}
var Wr = typeof exports == "object" && exports && !exports.nodeType && exports, Ge = Wr && typeof module == "object" && module && !module.nodeType && module, Ra = Ge && Ge.exports === Wr, Ne = Ra ? C.Buffer : void 0, ze = Ne ? Ne.allocUnsafe : void 0;
function Hr(t, e) {
  if (e)
    return t.slice();
  var r = t.length, n = ze ? ze(r) : new t.constructor(r);
  return t.copy(n), n;
}
function ka(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length, o = 0, i = []; ++r < n; ) {
    var a = t[r];
    e(a, r, t) && (i[o++] = a);
  }
  return i;
}
function Kr() {
  return [];
}
var Ua = Object.prototype, Ga = Ua.propertyIsEnumerable, Ve = Object.getOwnPropertySymbols, Na = Ve ? function(t) {
  return t == null ? [] : (t = Object(t), ka(Ve(t), function(e) {
    return Ga.call(t, e);
  }));
} : Kr;
const _e = Na;
function za(t, e) {
  return rt(t, _e(t), e);
}
var Va = Object.getOwnPropertySymbols, Wa = Va ? function(t) {
  for (var e = []; t; )
    Nr(e, _e(t)), t = ge(t);
  return e;
} : Kr;
const qr = Wa;
function Ha(t, e) {
  return rt(t, qr(t), e);
}
function Jr(t, e, r) {
  var n = e(t);
  return O(t) ? n : Nr(n, r(t));
}
function Xt(t) {
  return Jr(t, jt, _e);
}
function Ka(t) {
  return Jr(t, nt, qr);
}
var qa = q(C, "DataView");
const Yt = qa;
var Ja = q(C, "Promise");
const te = Ja;
var Za = q(C, "Set");
const ee = Za;
var We = "[object Map]", Qa = "[object Object]", He = "[object Promise]", Ke = "[object Set]", qe = "[object WeakMap]", Je = "[object DataView]", Xa = K(Yt), Ya = K(yt), ts = K(te), es = K(ee), rs = K(Qt), N = H;
(Yt && N(new Yt(new ArrayBuffer(1))) != Je || yt && N(new yt()) != We || te && N(te.resolve()) != He || ee && N(new ee()) != Ke || Qt && N(new Qt()) != qe) && (N = function(t) {
  var e = H(t), r = e == Qa ? t.constructor : void 0, n = r ? K(r) : "";
  if (n)
    switch (n) {
      case Xa:
        return Je;
      case Ya:
        return We;
      case ts:
        return He;
      case es:
        return Ke;
      case rs:
        return qe;
    }
  return e;
});
const vt = N;
var ns = Object.prototype, os = ns.hasOwnProperty;
function is(t) {
  var e = t.length, r = new t.constructor(e);
  return e && typeof t[0] == "string" && os.call(t, "index") && (r.index = t.index, r.input = t.input), r;
}
var as = C.Uint8Array;
const xt = as;
function me(t) {
  var e = new t.constructor(t.byteLength);
  return new xt(e).set(new xt(t)), e;
}
function ss(t, e) {
  var r = e ? me(t.buffer) : t.buffer;
  return new t.constructor(r, t.byteOffset, t.byteLength);
}
var ls = /\w*$/;
function cs(t) {
  var e = new t.constructor(t.source, ls.exec(t));
  return e.lastIndex = t.lastIndex, e;
}
var Ze = S ? S.prototype : void 0, Qe = Ze ? Ze.valueOf : void 0;
function us(t) {
  return Qe ? Object(Qe.call(t)) : {};
}
function Zr(t, e) {
  var r = e ? me(t.buffer) : t.buffer;
  return new t.constructor(r, t.byteOffset, t.length);
}
var fs = "[object Boolean]", ps = "[object Date]", hs = "[object Map]", ds = "[object Number]", bs = "[object RegExp]", ys = "[object Set]", vs = "[object String]", gs = "[object Symbol]", _s = "[object ArrayBuffer]", ms = "[object DataView]", js = "[object Float32Array]", ws = "[object Float64Array]", Ps = "[object Int8Array]", $s = "[object Int16Array]", Os = "[object Int32Array]", Ts = "[object Uint8Array]", Ss = "[object Uint8ClampedArray]", As = "[object Uint16Array]", Cs = "[object Uint32Array]";
function Es(t, e, r) {
  var n = t.constructor;
  switch (e) {
    case _s:
      return me(t);
    case fs:
    case ps:
      return new n(+t);
    case ms:
      return ss(t, r);
    case js:
    case ws:
    case Ps:
    case $s:
    case Os:
    case Ts:
    case Ss:
    case As:
    case Cs:
      return Zr(t, r);
    case hs:
      return new n();
    case ds:
    case vs:
      return new n(t);
    case bs:
      return cs(t);
    case ys:
      return new n();
    case gs:
      return us(t);
  }
}
function Qr(t) {
  return typeof t.constructor == "function" && !he(t) ? wo(ge(t)) : {};
}
var xs = "[object Map]";
function Ls(t) {
  return A(t) && vt(t) == xs;
}
var Xe = Y && Y.isMap, Fs = Xe ? de(Xe) : Ls;
const Is = Fs;
var Bs = "[object Set]";
function Ds(t) {
  return A(t) && vt(t) == Bs;
}
var Ye = Y && Y.isSet, Ms = Ye ? de(Ye) : Ds;
const Rs = Ms;
var ks = 1, Us = 2, Gs = 4, Xr = "[object Arguments]", Ns = "[object Array]", zs = "[object Boolean]", Vs = "[object Date]", Ws = "[object Error]", Yr = "[object Function]", Hs = "[object GeneratorFunction]", Ks = "[object Map]", qs = "[object Number]", tn = "[object Object]", Js = "[object RegExp]", Zs = "[object Set]", Qs = "[object String]", Xs = "[object Symbol]", Ys = "[object WeakMap]", tl = "[object ArrayBuffer]", el = "[object DataView]", rl = "[object Float32Array]", nl = "[object Float64Array]", ol = "[object Int8Array]", il = "[object Int16Array]", al = "[object Int32Array]", sl = "[object Uint8Array]", ll = "[object Uint8ClampedArray]", cl = "[object Uint16Array]", ul = "[object Uint32Array]", v = {};
v[Xr] = v[Ns] = v[tl] = v[el] = v[zs] = v[Vs] = v[rl] = v[nl] = v[ol] = v[il] = v[al] = v[Ks] = v[qs] = v[tn] = v[Js] = v[Zs] = v[Qs] = v[Xs] = v[sl] = v[ll] = v[cl] = v[ul] = !0;
v[Ws] = v[Yr] = v[Ys] = !1;
function Tt(t, e, r, n, o, i) {
  var a, s = e & ks, l = e & Us, c = e & Gs;
  if (r && (a = o ? r(t, n, o, i) : r(t)), a !== void 0)
    return a;
  if (!E(t))
    return t;
  var p = O(t);
  if (p) {
    if (a = is(t), !s)
      return Lr(t, a);
  } else {
    var f = vt(t), u = f == Yr || f == Hs;
    if (dt(t))
      return Hr(t, s);
    if (f == tn || f == Xr || u && !o) {
      if (a = l || u ? {} : Qr(t), !s)
        return l ? Ha(t, Ma(a, t)) : za(t, Da(a, t));
    } else {
      if (!v[f])
        return o ? t : {};
      a = Es(t, f, s);
    }
  }
  i || (i = new T());
  var d = i.get(t);
  if (d)
    return d;
  i.set(t, a), Rs(t) ? t.forEach(function(b) {
    a.add(Tt(b, e, r, b, t, i));
  }) : Is(t) && t.forEach(function(b, h) {
    a.set(h, Tt(b, e, r, h, t, i));
  });
  var _ = c ? l ? Ka : Xt : l ? nt : jt, w = p ? void 0 : _(t);
  return Io(w || t, function(b, h) {
    w && (h = b, b = t[h]), Fr(a, h, Tt(b, e, r, h, t, i));
  }), a;
}
var fl = 1, pl = 4;
function hl(t) {
  return Tt(t, fl | pl);
}
var dl = "__lodash_hash_undefined__";
function bl(t) {
  return this.__data__.set(t, dl), this;
}
function yl(t) {
  return this.__data__.has(t);
}
function Lt(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.__data__ = new D(); ++e < r; )
    this.add(t[e]);
}
Lt.prototype.add = Lt.prototype.push = bl;
Lt.prototype.has = yl;
function vl(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length; ++r < n; )
    if (e(t[r], r, t))
      return !0;
  return !1;
}
function gl(t, e) {
  return t.has(e);
}
var _l = 1, ml = 2;
function en(t, e, r, n, o, i) {
  var a = r & _l, s = t.length, l = e.length;
  if (s != l && !(a && l > s))
    return !1;
  var c = i.get(t), p = i.get(e);
  if (c && p)
    return c == e && p == t;
  var f = -1, u = !0, d = r & ml ? new Lt() : void 0;
  for (i.set(t, e), i.set(e, t); ++f < s; ) {
    var _ = t[f], w = e[f];
    if (n)
      var b = a ? n(w, _, f, e, t, i) : n(_, w, f, t, e, i);
    if (b !== void 0) {
      if (b)
        continue;
      u = !1;
      break;
    }
    if (d) {
      if (!vl(e, function(h, $) {
        if (!gl(d, $) && (_ === h || o(_, h, r, n, i)))
          return d.push($);
      })) {
        u = !1;
        break;
      }
    } else if (!(_ === w || o(_, w, r, n, i))) {
      u = !1;
      break;
    }
  }
  return i.delete(t), i.delete(e), u;
}
function jl(t) {
  var e = -1, r = Array(t.size);
  return t.forEach(function(n, o) {
    r[++e] = [o, n];
  }), r;
}
function wl(t) {
  var e = -1, r = Array(t.size);
  return t.forEach(function(n) {
    r[++e] = n;
  }), r;
}
var Pl = 1, $l = 2, Ol = "[object Boolean]", Tl = "[object Date]", Sl = "[object Error]", Al = "[object Map]", Cl = "[object Number]", El = "[object RegExp]", xl = "[object Set]", Ll = "[object String]", Fl = "[object Symbol]", Il = "[object ArrayBuffer]", Bl = "[object DataView]", tr = S ? S.prototype : void 0, Kt = tr ? tr.valueOf : void 0;
function Dl(t, e, r, n, o, i, a) {
  switch (r) {
    case Bl:
      if (t.byteLength != e.byteLength || t.byteOffset != e.byteOffset)
        return !1;
      t = t.buffer, e = e.buffer;
    case Il:
      return !(t.byteLength != e.byteLength || !i(new xt(t), new xt(e)));
    case Ol:
    case Tl:
    case Cl:
      return mt(+t, +e);
    case Sl:
      return t.name == e.name && t.message == e.message;
    case El:
    case Ll:
      return t == e + "";
    case Al:
      var s = jl;
    case xl:
      var l = n & Pl;
      if (s || (s = wl), t.size != e.size && !l)
        return !1;
      var c = a.get(t);
      if (c)
        return c == e;
      n |= $l, a.set(t, e);
      var p = en(s(t), s(e), n, o, i, a);
      return a.delete(t), p;
    case Fl:
      if (Kt)
        return Kt.call(t) == Kt.call(e);
  }
  return !1;
}
var Ml = 1, Rl = Object.prototype, kl = Rl.hasOwnProperty;
function Ul(t, e, r, n, o, i) {
  var a = r & Ml, s = Xt(t), l = s.length, c = Xt(e), p = c.length;
  if (l != p && !a)
    return !1;
  for (var f = l; f--; ) {
    var u = s[f];
    if (!(a ? u in e : kl.call(e, u)))
      return !1;
  }
  var d = i.get(t), _ = i.get(e);
  if (d && _)
    return d == e && _ == t;
  var w = !0;
  i.set(t, e), i.set(e, t);
  for (var b = a; ++f < l; ) {
    u = s[f];
    var h = t[u], $ = e[u];
    if (n)
      var Pt = a ? n($, h, u, e, t, i) : n(h, $, u, t, e, i);
    if (!(Pt === void 0 ? h === $ || o(h, $, r, n, i) : Pt)) {
      w = !1;
      break;
    }
    b || (b = u == "constructor");
  }
  if (w && !b) {
    var lt = t.constructor, Q = e.constructor;
    lt != Q && "constructor" in t && "constructor" in e && !(typeof lt == "function" && lt instanceof lt && typeof Q == "function" && Q instanceof Q) && (w = !1);
  }
  return i.delete(t), i.delete(e), w;
}
var Gl = 1, er = "[object Arguments]", rr = "[object Array]", $t = "[object Object]", Nl = Object.prototype, nr = Nl.hasOwnProperty;
function zl(t, e, r, n, o, i) {
  var a = O(t), s = O(e), l = a ? rr : vt(t), c = s ? rr : vt(e);
  l = l == er ? $t : l, c = c == er ? $t : c;
  var p = l == $t, f = c == $t, u = l == c;
  if (u && dt(t)) {
    if (!dt(e))
      return !1;
    a = !0, p = !1;
  }
  if (u && !p)
    return i || (i = new T()), a || be(t) ? en(t, e, r, n, o, i) : Dl(t, e, l, r, n, o, i);
  if (!(r & Gl)) {
    var d = p && nr.call(t, "__wrapped__"), _ = f && nr.call(e, "__wrapped__");
    if (d || _) {
      var w = d ? t.value() : t, b = _ ? e.value() : e;
      return i || (i = new T()), o(w, b, r, n, i);
    }
  }
  return u ? (i || (i = new T()), Ul(t, e, r, n, o, i)) : !1;
}
function je(t, e, r, n, o) {
  return t === e ? !0 : t == null || e == null || !A(t) && !A(e) ? t !== t && e !== e : zl(t, e, r, n, je, o);
}
var Vl = 1, Wl = 2;
function Hl(t, e, r, n) {
  var o = r.length, i = o, a = !n;
  if (t == null)
    return !i;
  for (t = Object(t); o--; ) {
    var s = r[o];
    if (a && s[2] ? s[1] !== t[s[0]] : !(s[0] in t))
      return !1;
  }
  for (; ++o < i; ) {
    s = r[o];
    var l = s[0], c = t[l], p = s[1];
    if (a && s[2]) {
      if (c === void 0 && !(l in t))
        return !1;
    } else {
      var f = new T();
      if (n)
        var u = n(c, p, l, t, e, f);
      if (!(u === void 0 ? je(p, c, Vl | Wl, n, f) : u))
        return !1;
    }
  }
  return !0;
}
function rn(t) {
  return t === t && !E(t);
}
function Kl(t) {
  for (var e = jt(t), r = e.length; r--; ) {
    var n = e[r], o = t[n];
    e[r] = [n, o, rn(o)];
  }
  return e;
}
function nn(t, e) {
  return function(r) {
    return r == null ? !1 : r[t] === e && (e !== void 0 || t in Object(r));
  };
}
function ql(t) {
  var e = Kl(t);
  return e.length == 1 && e[0][2] ? nn(e[0][0], e[0][1]) : function(r) {
    return r === t || Hl(r, t, e);
  };
}
function Jl(t, e) {
  return t != null && e in Object(t);
}
function Zl(t, e, r) {
  e = Ur(e, t);
  for (var n = -1, o = e.length, i = !1; ++n < o; ) {
    var a = Ut(e[n]);
    if (!(i = t != null && r(t, a)))
      break;
    t = t[a];
  }
  return i || ++n != o ? i : (o = t == null ? 0 : t.length, !!o && pe(o) && fe(a, o) && (O(t) || Et(t)));
}
function Ql(t, e) {
  return t != null && Zl(t, e, Jl);
}
var Xl = 1, Yl = 2;
function tc(t, e) {
  return ye(t) && rn(e) ? nn(Ut(t), e) : function(r) {
    var n = Pa(r, t);
    return n === void 0 && n === e ? Ql(r, t) : je(e, n, Xl | Yl);
  };
}
function ec(t) {
  return function(e) {
    return e == null ? void 0 : e[t];
  };
}
function rc(t) {
  return function(e) {
    return Gr(e, t);
  };
}
function nc(t) {
  return ye(t) ? ec(Ut(t)) : rc(t);
}
function oc(t) {
  return typeof t == "function" ? t : t == null ? ce : typeof t == "object" ? O(t) ? tc(t[0], t[1]) : ql(t) : nc(t);
}
function ic(t) {
  return function(e, r, n) {
    for (var o = -1, i = Object(e), a = n(e), s = a.length; s--; ) {
      var l = a[t ? s : ++o];
      if (r(i[l], l, i) === !1)
        break;
    }
    return e;
  };
}
var ac = ic();
const on = ac;
function sc(t, e) {
  return t && on(t, e, jt);
}
function re(t, e, r) {
  (r !== void 0 && !mt(t[e], r) || r === void 0 && !(e in t)) && Dt(t, e, r);
}
function lc(t) {
  return A(t) && Mt(t);
}
function ne(t, e) {
  if (!(e === "constructor" && typeof t[e] == "function") && e != "__proto__")
    return t[e];
}
function cc(t) {
  return rt(t, nt(t));
}
function uc(t, e, r, n, o, i, a) {
  var s = ne(t, r), l = ne(e, r), c = a.get(l);
  if (c) {
    re(t, r, c);
    return;
  }
  var p = i ? i(s, l, r + "", t, e, a) : void 0, f = p === void 0;
  if (f) {
    var u = O(l), d = !u && dt(l), _ = !u && !d && be(l);
    p = l, u || d || _ ? O(s) ? p = s : lc(s) ? p = Lr(s) : d ? (f = !1, p = Hr(l, !0)) : _ ? (f = !1, p = Zr(l, !0)) : p = [] : Vr(l) || Et(l) ? (p = s, Et(s) ? p = cc(s) : (!E(s) || ue(s)) && (p = Qr(l))) : f = !1;
  }
  f && (a.set(l, p), o(p, l, n, i, a), a.delete(l)), re(t, r, p);
}
function an(t, e, r, n, o) {
  t !== e && on(e, function(i, a) {
    if (o || (o = new T()), E(i))
      uc(t, e, a, r, an, n, o);
    else {
      var s = n ? n(ne(t, a), i, a + "", t, e, o) : void 0;
      s === void 0 && (s = i), re(t, a, s);
    }
  }, nt);
}
var fc = Ir(function(t, e, r, n) {
  an(t, e, r, n);
});
const pc = fc;
function hc(t, e) {
  var r = {};
  return e = oc(e), sc(t, function(n, o, i) {
    Dt(r, e(n, o, i), n);
  }), r;
}
var Ft = function() {
  return Ft = Object.assign || function(t) {
    for (var e, r = 1, n = arguments.length; r < n; r++) {
      e = arguments[r];
      for (var o in e)
        Object.prototype.hasOwnProperty.call(e, o) && (t[o] = e[o]);
    }
    return t;
  }, Ft.apply(this, arguments);
};
function dc(t) {
  return t.toLowerCase();
}
var bc = [/([a-z0-9])([A-Z])/g, /([A-Z])([A-Z][a-z])/g], yc = /[^A-Z0-9]+/gi;
function vc(t, e) {
  e === void 0 && (e = {});
  for (var r = e.splitRegexp, n = r === void 0 ? bc : r, o = e.stripRegexp, i = o === void 0 ? yc : o, a = e.transform, s = a === void 0 ? dc : a, l = e.delimiter, c = l === void 0 ? " " : l, p = or(or(t, n, "$1\0$2"), i, "\0"), f = 0, u = p.length; p.charAt(f) === "\0"; )
    f++;
  for (; p.charAt(u - 1) === "\0"; )
    u--;
  return p.slice(f, u).split("\0").map(s).join(c);
}
function or(t, e, r) {
  return e instanceof RegExp ? t.replace(e, r) : e.reduce(function(n, o) {
    return n.replace(o, r);
  }, t);
}
function sn(t, e) {
  var r = t.charAt(0), n = t.substr(1).toLowerCase();
  return e > 0 && r >= "0" && r <= "9" ? "_" + r + n : "" + r.toUpperCase() + n;
}
function gc(t, e) {
  return e === void 0 && (e = {}), vc(t, Ft({ delimiter: "", transform: sn }, e));
}
function _c(t, e) {
  return e === 0 ? t.toLowerCase() : sn(t, e);
}
function mc(t, e) {
  return e === void 0 && (e = {}), gc(t, Ft({ transform: _c }, e));
}
function jc(t, e) {
  const r = /* @__PURE__ */ Object.create(null), n = t.split(",");
  for (let o = 0; o < n.length; o++)
    r[n[o]] = !0;
  return e ? (o) => !!r[o.toLowerCase()] : (o) => !!r[o];
}
const wc = /* @__PURE__ */ jc(
  "String,Number,Boolean,Function,Symbol,BigInt"
);
function Pc(t) {
  const e = t && t.toString().match(/^\s*function (\w+)/);
  return e ? e[1] : t === null ? "null" : "";
}
function $c(t, e) {
  let r;
  const n = Pc(e);
  if (wc(n)) {
    const o = typeof t;
    r = o === n.toLowerCase(), !r && o === "object" && (r = t instanceof e);
  } else
    n === "Object" ? r = E(t) : n === "Array" ? r = Array.isArray(t) : n === "null" ? r = t === null : r = t instanceof e;
  return {
    valid: r,
    expectedType: n
  };
}
function Oc({
  prop: t,
  type: e,
  validator: r
}) {
  if (![void 0, null].includes(t) && e) {
    let n = !1;
    const o = Array.isArray(e) ? e : [e], i = [];
    for (let a = 0; a < o.length && !n; a++) {
      const { valid: s, expectedType: l } = $c(t, o[a]);
      i.push(l || ""), n = s;
    }
    if (!n)
      throw new Error(`Invalid prop: type check failed, expecting [${i.join(", ")}], receiving: ${t}`);
  }
  if (r && !r(t))
    throw new Error(`Invalid prop: validator check failed, receiving: ${t}`);
}
function Tc(t, {
  mergeObject: e,
  mergeFunction: r
}) {
  const n = [];
  for (let i = t.length - 1; i >= 0; i--)
    n.push(t[i]);
  const o = r ? (i, a) => i instanceof Function && a instanceof Function ? r(a, i) : void 0 : void 0;
  return e === "deep" ? pc(...n, o) : ki(...n, o);
}
function Sc(t, {
  mergeFunction: e
}) {
  return t.reduce(e, () => {
  });
}
function y(t, e = {}) {
  const {
    type: r,
    default: n,
    defaultIsDynamic: o = !1,
    required: i = !1,
    validator: a,
    camelCase: s = !0,
    mergeObjectApplyOnlyToDefault: l = !1,
    mergeFunctionApplyOnlyToDefault: c = !0
  } = e;
  let {
    mergeObject: p = "deep",
    mergeFunction: f = !1
  } = e;
  const u = [];
  let d, _ = !1, w = !1;
  const b = (h) => {
    if (h !== void 0) {
      Oc({ type: r, prop: h, validator: a });
      const $ = Vr(h), Pt = h instanceof Function;
      return _ = $, w = Pt, $ ? (h = hl(h), s ? hc(h, (lt, Q) => mc(Q, {
        stripRegexp: /-/g
      })) : h) : h;
    }
  };
  for (const h of t)
    u.push(b(h));
  if (!o)
    u.push(b(n));
  else if (!(n instanceof Function))
    throw new TypeError(`Invalid option: config.default should be Function when config.defaultIsDynamic enabled, receiving: ${n}`);
  _ || (p = !1, w || (f = !1));
  for (let h = 0; h < u.length; h++) {
    const $ = u[h];
    if ($ !== void 0) {
      h === u.length - 1 ? d = $ : p ? d = Tc(l ? [$, n] : u, {
        mergeObject: p,
        mergeFunction: f
      }) : f ? d = Sc(c ? [$, n] : u, {
        mergeFunction: f
      }) : d = $;
      break;
    }
  }
  if (i && [void 0, null].includes(d))
    throw new Error("Missing required prop");
  return o ? y(
    t,
    {
      ...e,
      default: n(d),
      defaultIsDynamic: !1
    }
  ) : d;
}
function Ac(t, e) {
  return Object.keys(t).filter((r) => e(r)).reduce((r, n) => (r[n] = t[n], r), {});
}
function Cc(t) {
  return Ac(t, (e) => !e.startsWith("hook:"));
}
var Ec = typeof global == "object" && global && global.Object === Object && global;
const ln = Ec;
var xc = typeof self == "object" && self && self.Object === Object && self, Lc = ln || xc || Function("return this")();
const x = Lc;
var Fc = x.Symbol;
const tt = Fc;
var cn = Object.prototype, Ic = cn.hasOwnProperty, Bc = cn.toString, ut = tt ? tt.toStringTag : void 0;
function Dc(t) {
  var e = Ic.call(t, ut), r = t[ut];
  try {
    t[ut] = void 0;
    var n = !0;
  } catch {
  }
  var o = Bc.call(t);
  return n && (e ? t[ut] = r : delete t[ut]), o;
}
var Mc = Object.prototype, Rc = Mc.toString;
function kc(t) {
  return Rc.call(t);
}
var Uc = "[object Null]", Gc = "[object Undefined]", ir = tt ? tt.toStringTag : void 0;
function ot(t) {
  return t == null ? t === void 0 ? Gc : Uc : ir && ir in Object(t) ? Dc(t) : kc(t);
}
function it(t) {
  return t != null && typeof t == "object";
}
var Nc = Array.isArray;
const we = Nc;
function wt(t) {
  var e = typeof t;
  return t != null && (e == "object" || e == "function");
}
var zc = "[object AsyncFunction]", Vc = "[object Function]", Wc = "[object GeneratorFunction]", Hc = "[object Proxy]";
function un(t) {
  if (!wt(t))
    return !1;
  var e = ot(t);
  return e == Vc || e == Wc || e == zc || e == Hc;
}
var Kc = x["__core-js_shared__"];
const qt = Kc;
var ar = function() {
  var t = /[^.]+$/.exec(qt && qt.keys && qt.keys.IE_PROTO || "");
  return t ? "Symbol(src)_1." + t : "";
}();
function qc(t) {
  return !!ar && ar in t;
}
var Jc = Function.prototype, Zc = Jc.toString;
function J(t) {
  if (t != null) {
    try {
      return Zc.call(t);
    } catch {
    }
    try {
      return t + "";
    } catch {
    }
  }
  return "";
}
var Qc = /[\\^$.*+?()[\]{}|]/g, Xc = /^\[object .+?Constructor\]$/, Yc = Function.prototype, tu = Object.prototype, eu = Yc.toString, ru = tu.hasOwnProperty, nu = RegExp(
  "^" + eu.call(ru).replace(Qc, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function ou(t) {
  if (!wt(t) || qc(t))
    return !1;
  var e = un(t) ? nu : Xc;
  return e.test(J(t));
}
function iu(t, e) {
  return t == null ? void 0 : t[e];
}
function Z(t, e) {
  var r = iu(t, e);
  return ou(r) ? r : void 0;
}
var au = Z(x, "WeakMap");
const oe = au;
var sr = Object.create, su = function() {
  function t() {
  }
  return function(e) {
    if (!wt(e))
      return {};
    if (sr)
      return sr(e);
    t.prototype = e;
    var r = new t();
    return t.prototype = void 0, r;
  };
}();
const lu = su;
function cu(t, e) {
  var r = -1, n = t.length;
  for (e || (e = Array(n)); ++r < n; )
    e[r] = t[r];
  return e;
}
var uu = function() {
  try {
    var t = Z(Object, "defineProperty");
    return t({}, "", {}), t;
  } catch {
  }
}();
const lr = uu;
function fu(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length; ++r < n && e(t[r], r, t) !== !1; )
    ;
  return t;
}
var pu = 9007199254740991, hu = /^(?:0|[1-9]\d*)$/;
function du(t, e) {
  var r = typeof t;
  return e = e == null ? pu : e, !!e && (r == "number" || r != "symbol" && hu.test(t)) && t > -1 && t % 1 == 0 && t < e;
}
function fn(t, e, r) {
  e == "__proto__" && lr ? lr(t, e, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : t[e] = r;
}
function pn(t, e) {
  return t === e || t !== t && e !== e;
}
var bu = Object.prototype, yu = bu.hasOwnProperty;
function hn(t, e, r) {
  var n = t[e];
  (!(yu.call(t, e) && pn(n, r)) || r === void 0 && !(e in t)) && fn(t, e, r);
}
function Gt(t, e, r, n) {
  var o = !r;
  r || (r = {});
  for (var i = -1, a = e.length; ++i < a; ) {
    var s = e[i], l = n ? n(r[s], t[s], s, r, t) : void 0;
    l === void 0 && (l = t[s]), o ? fn(r, s, l) : hn(r, s, l);
  }
  return r;
}
var vu = 9007199254740991;
function dn(t) {
  return typeof t == "number" && t > -1 && t % 1 == 0 && t <= vu;
}
function bn(t) {
  return t != null && dn(t.length) && !un(t);
}
var gu = Object.prototype;
function Pe(t) {
  var e = t && t.constructor, r = typeof e == "function" && e.prototype || gu;
  return t === r;
}
function _u(t, e) {
  for (var r = -1, n = Array(t); ++r < t; )
    n[r] = e(r);
  return n;
}
var mu = "[object Arguments]";
function cr(t) {
  return it(t) && ot(t) == mu;
}
var yn = Object.prototype, ju = yn.hasOwnProperty, wu = yn.propertyIsEnumerable, Pu = cr(function() {
  return arguments;
}()) ? cr : function(t) {
  return it(t) && ju.call(t, "callee") && !wu.call(t, "callee");
};
const $u = Pu;
function Ou() {
  return !1;
}
var vn = typeof exports == "object" && exports && !exports.nodeType && exports, ur = vn && typeof module == "object" && module && !module.nodeType && module, Tu = ur && ur.exports === vn, fr = Tu ? x.Buffer : void 0, Su = fr ? fr.isBuffer : void 0, Au = Su || Ou;
const gn = Au;
var Cu = "[object Arguments]", Eu = "[object Array]", xu = "[object Boolean]", Lu = "[object Date]", Fu = "[object Error]", Iu = "[object Function]", Bu = "[object Map]", Du = "[object Number]", Mu = "[object Object]", Ru = "[object RegExp]", ku = "[object Set]", Uu = "[object String]", Gu = "[object WeakMap]", Nu = "[object ArrayBuffer]", zu = "[object DataView]", Vu = "[object Float32Array]", Wu = "[object Float64Array]", Hu = "[object Int8Array]", Ku = "[object Int16Array]", qu = "[object Int32Array]", Ju = "[object Uint8Array]", Zu = "[object Uint8ClampedArray]", Qu = "[object Uint16Array]", Xu = "[object Uint32Array]", j = {};
j[Vu] = j[Wu] = j[Hu] = j[Ku] = j[qu] = j[Ju] = j[Zu] = j[Qu] = j[Xu] = !0;
j[Cu] = j[Eu] = j[Nu] = j[xu] = j[zu] = j[Lu] = j[Fu] = j[Iu] = j[Bu] = j[Du] = j[Mu] = j[Ru] = j[ku] = j[Uu] = j[Gu] = !1;
function Yu(t) {
  return it(t) && dn(t.length) && !!j[ot(t)];
}
function $e(t) {
  return function(e) {
    return t(e);
  };
}
var _n = typeof exports == "object" && exports && !exports.nodeType && exports, ht = _n && typeof module == "object" && module && !module.nodeType && module, tf = ht && ht.exports === _n, Jt = tf && ln.process, ef = function() {
  try {
    var t = ht && ht.require && ht.require("util").types;
    return t || Jt && Jt.binding && Jt.binding("util");
  } catch {
  }
}();
const et = ef;
var pr = et && et.isTypedArray, rf = pr ? $e(pr) : Yu;
const nf = rf;
var of = Object.prototype, af = of.hasOwnProperty;
function mn(t, e) {
  var r = we(t), n = !r && $u(t), o = !r && !n && gn(t), i = !r && !n && !o && nf(t), a = r || n || o || i, s = a ? _u(t.length, String) : [], l = s.length;
  for (var c in t)
    (e || af.call(t, c)) && !(a && (c == "length" || o && (c == "offset" || c == "parent") || i && (c == "buffer" || c == "byteLength" || c == "byteOffset") || du(c, l))) && s.push(c);
  return s;
}
function jn(t, e) {
  return function(r) {
    return t(e(r));
  };
}
var sf = jn(Object.keys, Object);
const lf = sf;
var cf = Object.prototype, uf = cf.hasOwnProperty;
function ff(t) {
  if (!Pe(t))
    return lf(t);
  var e = [];
  for (var r in Object(t))
    uf.call(t, r) && r != "constructor" && e.push(r);
  return e;
}
function Oe(t) {
  return bn(t) ? mn(t) : ff(t);
}
function pf(t) {
  var e = [];
  if (t != null)
    for (var r in Object(t))
      e.push(r);
  return e;
}
var hf = Object.prototype, df = hf.hasOwnProperty;
function bf(t) {
  if (!wt(t))
    return pf(t);
  var e = Pe(t), r = [];
  for (var n in t)
    n == "constructor" && (e || !df.call(t, n)) || r.push(n);
  return r;
}
function Te(t) {
  return bn(t) ? mn(t, !0) : bf(t);
}
var yf = Z(Object, "create");
const gt = yf;
function vf() {
  this.__data__ = gt ? gt(null) : {}, this.size = 0;
}
function gf(t) {
  var e = this.has(t) && delete this.__data__[t];
  return this.size -= e ? 1 : 0, e;
}
var _f = "__lodash_hash_undefined__", mf = Object.prototype, jf = mf.hasOwnProperty;
function wf(t) {
  var e = this.__data__;
  if (gt) {
    var r = e[t];
    return r === _f ? void 0 : r;
  }
  return jf.call(e, t) ? e[t] : void 0;
}
var Pf = Object.prototype, $f = Pf.hasOwnProperty;
function Of(t) {
  var e = this.__data__;
  return gt ? e[t] !== void 0 : $f.call(e, t);
}
var Tf = "__lodash_hash_undefined__";
function Sf(t, e) {
  var r = this.__data__;
  return this.size += this.has(t) ? 0 : 1, r[t] = gt && e === void 0 ? Tf : e, this;
}
function W(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
W.prototype.clear = vf;
W.prototype.delete = gf;
W.prototype.get = wf;
W.prototype.has = Of;
W.prototype.set = Sf;
function Af() {
  this.__data__ = [], this.size = 0;
}
function Nt(t, e) {
  for (var r = t.length; r--; )
    if (pn(t[r][0], e))
      return r;
  return -1;
}
var Cf = Array.prototype, Ef = Cf.splice;
function xf(t) {
  var e = this.__data__, r = Nt(e, t);
  if (r < 0)
    return !1;
  var n = e.length - 1;
  return r == n ? e.pop() : Ef.call(e, r, 1), --this.size, !0;
}
function Lf(t) {
  var e = this.__data__, r = Nt(e, t);
  return r < 0 ? void 0 : e[r][1];
}
function Ff(t) {
  return Nt(this.__data__, t) > -1;
}
function If(t, e) {
  var r = this.__data__, n = Nt(r, t);
  return n < 0 ? (++this.size, r.push([t, e])) : r[n][1] = e, this;
}
function M(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
M.prototype.clear = Af;
M.prototype.delete = xf;
M.prototype.get = Lf;
M.prototype.has = Ff;
M.prototype.set = If;
var Bf = Z(x, "Map");
const _t = Bf;
function Df() {
  this.size = 0, this.__data__ = {
    hash: new W(),
    map: new (_t || M)(),
    string: new W()
  };
}
function Mf(t) {
  var e = typeof t;
  return e == "string" || e == "number" || e == "symbol" || e == "boolean" ? t !== "__proto__" : t === null;
}
function zt(t, e) {
  var r = t.__data__;
  return Mf(e) ? r[typeof e == "string" ? "string" : "hash"] : r.map;
}
function Rf(t) {
  var e = zt(this, t).delete(t);
  return this.size -= e ? 1 : 0, e;
}
function kf(t) {
  return zt(this, t).get(t);
}
function Uf(t) {
  return zt(this, t).has(t);
}
function Gf(t, e) {
  var r = zt(this, t), n = r.size;
  return r.set(t, e), this.size += r.size == n ? 0 : 1, this;
}
function at(t) {
  var e = -1, r = t == null ? 0 : t.length;
  for (this.clear(); ++e < r; ) {
    var n = t[e];
    this.set(n[0], n[1]);
  }
}
at.prototype.clear = Df;
at.prototype.delete = Rf;
at.prototype.get = kf;
at.prototype.has = Uf;
at.prototype.set = Gf;
function wn(t, e) {
  for (var r = -1, n = e.length, o = t.length; ++r < n; )
    t[o + r] = e[r];
  return t;
}
var Nf = jn(Object.getPrototypeOf, Object);
const Se = Nf;
var zf = "[object Object]", Vf = Function.prototype, Wf = Object.prototype, Pn = Vf.toString, Hf = Wf.hasOwnProperty, Kf = Pn.call(Object);
function $n(t) {
  if (!it(t) || ot(t) != zf)
    return !1;
  var e = Se(t);
  if (e === null)
    return !0;
  var r = Hf.call(e, "constructor") && e.constructor;
  return typeof r == "function" && r instanceof r && Pn.call(r) == Kf;
}
function qf() {
  this.__data__ = new M(), this.size = 0;
}
function Jf(t) {
  var e = this.__data__, r = e.delete(t);
  return this.size = e.size, r;
}
function Zf(t) {
  return this.__data__.get(t);
}
function Qf(t) {
  return this.__data__.has(t);
}
var Xf = 200;
function Yf(t, e) {
  var r = this.__data__;
  if (r instanceof M) {
    var n = r.__data__;
    if (!_t || n.length < Xf - 1)
      return n.push([t, e]), this.size = ++r.size, this;
    r = this.__data__ = new at(n);
  }
  return r.set(t, e), this.size = r.size, this;
}
function st(t) {
  var e = this.__data__ = new M(t);
  this.size = e.size;
}
st.prototype.clear = qf;
st.prototype.delete = Jf;
st.prototype.get = Zf;
st.prototype.has = Qf;
st.prototype.set = Yf;
function tp(t, e) {
  return t && Gt(e, Oe(e), t);
}
function ep(t, e) {
  return t && Gt(e, Te(e), t);
}
var On = typeof exports == "object" && exports && !exports.nodeType && exports, hr = On && typeof module == "object" && module && !module.nodeType && module, rp = hr && hr.exports === On, dr = rp ? x.Buffer : void 0, br = dr ? dr.allocUnsafe : void 0;
function np(t, e) {
  if (e)
    return t.slice();
  var r = t.length, n = br ? br(r) : new t.constructor(r);
  return t.copy(n), n;
}
function op(t, e) {
  for (var r = -1, n = t == null ? 0 : t.length, o = 0, i = []; ++r < n; ) {
    var a = t[r];
    e(a, r, t) && (i[o++] = a);
  }
  return i;
}
function Tn() {
  return [];
}
var ip = Object.prototype, ap = ip.propertyIsEnumerable, yr = Object.getOwnPropertySymbols, sp = yr ? function(t) {
  return t == null ? [] : (t = Object(t), op(yr(t), function(e) {
    return ap.call(t, e);
  }));
} : Tn;
const Ae = sp;
function lp(t, e) {
  return Gt(t, Ae(t), e);
}
var cp = Object.getOwnPropertySymbols, up = cp ? function(t) {
  for (var e = []; t; )
    wn(e, Ae(t)), t = Se(t);
  return e;
} : Tn;
const Sn = up;
function fp(t, e) {
  return Gt(t, Sn(t), e);
}
function An(t, e, r) {
  var n = e(t);
  return we(t) ? n : wn(n, r(t));
}
function pp(t) {
  return An(t, Oe, Ae);
}
function hp(t) {
  return An(t, Te, Sn);
}
var dp = Z(x, "DataView");
const ie = dp;
var bp = Z(x, "Promise");
const ae = bp;
var yp = Z(x, "Set");
const se = yp;
var vr = "[object Map]", vp = "[object Object]", gr = "[object Promise]", _r = "[object Set]", mr = "[object WeakMap]", jr = "[object DataView]", gp = J(ie), _p = J(_t), mp = J(ae), jp = J(se), wp = J(oe), z = ot;
(ie && z(new ie(new ArrayBuffer(1))) != jr || _t && z(new _t()) != vr || ae && z(ae.resolve()) != gr || se && z(new se()) != _r || oe && z(new oe()) != mr) && (z = function(t) {
  var e = ot(t), r = e == vp ? t.constructor : void 0, n = r ? J(r) : "";
  if (n)
    switch (n) {
      case gp:
        return jr;
      case _p:
        return vr;
      case mp:
        return gr;
      case jp:
        return _r;
      case wp:
        return mr;
    }
  return e;
});
const Ce = z;
var Pp = Object.prototype, $p = Pp.hasOwnProperty;
function Op(t) {
  var e = t.length, r = new t.constructor(e);
  return e && typeof t[0] == "string" && $p.call(t, "index") && (r.index = t.index, r.input = t.input), r;
}
var Tp = x.Uint8Array;
const wr = Tp;
function Ee(t) {
  var e = new t.constructor(t.byteLength);
  return new wr(e).set(new wr(t)), e;
}
function Sp(t, e) {
  var r = e ? Ee(t.buffer) : t.buffer;
  return new t.constructor(r, t.byteOffset, t.byteLength);
}
var Ap = /\w*$/;
function Cp(t) {
  var e = new t.constructor(t.source, Ap.exec(t));
  return e.lastIndex = t.lastIndex, e;
}
var Pr = tt ? tt.prototype : void 0, $r = Pr ? Pr.valueOf : void 0;
function Ep(t) {
  return $r ? Object($r.call(t)) : {};
}
function xp(t, e) {
  var r = e ? Ee(t.buffer) : t.buffer;
  return new t.constructor(r, t.byteOffset, t.length);
}
var Lp = "[object Boolean]", Fp = "[object Date]", Ip = "[object Map]", Bp = "[object Number]", Dp = "[object RegExp]", Mp = "[object Set]", Rp = "[object String]", kp = "[object Symbol]", Up = "[object ArrayBuffer]", Gp = "[object DataView]", Np = "[object Float32Array]", zp = "[object Float64Array]", Vp = "[object Int8Array]", Wp = "[object Int16Array]", Hp = "[object Int32Array]", Kp = "[object Uint8Array]", qp = "[object Uint8ClampedArray]", Jp = "[object Uint16Array]", Zp = "[object Uint32Array]";
function Qp(t, e, r) {
  var n = t.constructor;
  switch (e) {
    case Up:
      return Ee(t);
    case Lp:
    case Fp:
      return new n(+t);
    case Gp:
      return Sp(t, r);
    case Np:
    case zp:
    case Vp:
    case Wp:
    case Hp:
    case Kp:
    case qp:
    case Jp:
    case Zp:
      return xp(t, r);
    case Ip:
      return new n();
    case Bp:
    case Rp:
      return new n(t);
    case Dp:
      return Cp(t);
    case Mp:
      return new n();
    case kp:
      return Ep(t);
  }
}
function Xp(t) {
  return typeof t.constructor == "function" && !Pe(t) ? lu(Se(t)) : {};
}
var Yp = "[object Map]";
function th(t) {
  return it(t) && Ce(t) == Yp;
}
var Or = et && et.isMap, eh = Or ? $e(Or) : th;
const rh = eh;
var nh = "[object Set]";
function oh(t) {
  return it(t) && Ce(t) == nh;
}
var Tr = et && et.isSet, ih = Tr ? $e(Tr) : oh;
const ah = ih;
var sh = 1, lh = 2, ch = 4, Cn = "[object Arguments]", uh = "[object Array]", fh = "[object Boolean]", ph = "[object Date]", hh = "[object Error]", En = "[object Function]", dh = "[object GeneratorFunction]", bh = "[object Map]", yh = "[object Number]", xn = "[object Object]", vh = "[object RegExp]", gh = "[object Set]", _h = "[object String]", mh = "[object Symbol]", jh = "[object WeakMap]", wh = "[object ArrayBuffer]", Ph = "[object DataView]", $h = "[object Float32Array]", Oh = "[object Float64Array]", Th = "[object Int8Array]", Sh = "[object Int16Array]", Ah = "[object Int32Array]", Ch = "[object Uint8Array]", Eh = "[object Uint8ClampedArray]", xh = "[object Uint16Array]", Lh = "[object Uint32Array]", g = {};
g[Cn] = g[uh] = g[wh] = g[Ph] = g[fh] = g[ph] = g[$h] = g[Oh] = g[Th] = g[Sh] = g[Ah] = g[bh] = g[yh] = g[xn] = g[vh] = g[gh] = g[_h] = g[mh] = g[Ch] = g[Eh] = g[xh] = g[Lh] = !0;
g[hh] = g[En] = g[jh] = !1;
function St(t, e, r, n, o, i) {
  var a, s = e & sh, l = e & lh, c = e & ch;
  if (r && (a = o ? r(t, n, o, i) : r(t)), a !== void 0)
    return a;
  if (!wt(t))
    return t;
  var p = we(t);
  if (p) {
    if (a = Op(t), !s)
      return cu(t, a);
  } else {
    var f = Ce(t), u = f == En || f == dh;
    if (gn(t))
      return np(t, s);
    if (f == xn || f == Cn || u && !o) {
      if (a = l || u ? {} : Xp(t), !s)
        return l ? fp(t, ep(a, t)) : lp(t, tp(a, t));
    } else {
      if (!g[f])
        return o ? t : {};
      a = Qp(t, f, s);
    }
  }
  i || (i = new st());
  var d = i.get(t);
  if (d)
    return d;
  i.set(t, a), ah(t) ? t.forEach(function(b) {
    a.add(St(b, e, r, b, t, i));
  }) : rh(t) && t.forEach(function(b, h) {
    a.set(h, St(b, e, r, h, t, i));
  });
  var _ = c ? l ? hp : pp : l ? Te : Oe, w = p ? void 0 : _(t);
  return fu(w || t, function(b, h) {
    w && (h = b, b = t[h]), hn(a, h, St(b, e, r, h, t, i));
  }), a;
}
var Fh = 1, Ih = 4;
function It(t) {
  return St(t, Fh | Ih);
}
function Bh(t) {
  let e = 0;
  if (t)
    for (const r of t)
      e += r.charCodeAt(0) > 255 ? 2 : 1;
  return e;
}
function Ln(t) {
  for (const e in t)
    t[e] = t[e].bind(this);
  return y([Cc(this.$listeners)], {
    default: t,
    mergeFunction: (e, r) => (...n) => {
      e(...n), r(...n);
    }
  });
}
function F(t) {
  return {
    object: () => t === null || Array.isArray(t) && t.length === 0 || $n(t) && Object.getOwnPropertyNames(t).length === 0,
    number: () => Number.isNaN(t),
    string: () => t === "",
    undefined: () => !0,
    boolean: () => t === !1,
    symbol: () => !1,
    bigint: () => !1,
    function: () => !0
  }[typeof t]();
}
function L(t) {
  return !F(t);
}
const Dh = (t) => {
  const { top: e, right: r, bottom: n, left: o } = t.getBoundingClientRect(), i = e >= 0 && e <= window.innerHeight && n >= 0 && n <= window.innerHeight, a = o >= 0 && o <= window.innerWidth && r >= 0 && r <= window.innerWidth;
  return i && a;
}, Mh = (t = ".el-form .el-form-item.is-error", e = window) => {
  const r = (o) => {
    o.scrollIntoView({
      behavior: "smooth",
      block: "center"
    });
  }, n = (o, i) => new Promise((a, s) => {
    if (o)
      for (const l of o instanceof NodeList ? o : [o]) {
        l.classList.add("animate__animated", i);
        const c = () => {
          l.classList.remove("animate__animated", i), l.removeEventListener("animationend", c), a();
        };
        l.addEventListener("animationend", c);
      }
    else
      s();
  });
  setTimeout(() => {
    const o = typeof t == "string" ? document.querySelectorAll(t) : t;
    if (o[0])
      if (Dh(o[0]))
        n(o, "animate__headShake").catch((i) => {
          console.warn(i);
        });
      else {
        let i = function() {
          clearTimeout(a), a = setTimeout(() => {
            n(o, "animate__headShake").catch((s) => {
              console.warn(s);
            }), e.removeEventListener("scroll", i);
          }, 100);
        }, a;
        e.addEventListener("scroll", i), r(o[0]);
      }
  }, 0);
};
function Vt(t, e, r, n, o, i, a, s) {
  var l = typeof t == "function" ? t.options : t;
  e && (l.render = e, l.staticRenderFns = r, l._compiled = !0), n && (l.functional = !0), i && (l._scopeId = "data-v-" + i);
  var c;
  if (a ? (c = function(u) {
    u = u || this.$vnode && this.$vnode.ssrContext || this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext, !u && typeof __VUE_SSR_CONTEXT__ < "u" && (u = __VUE_SSR_CONTEXT__), o && o.call(this, u), u && u._registeredComponents && u._registeredComponents.add(a);
  }, l._ssrRegister = c) : o && (c = s ? function() {
    o.call(
      this,
      (l.functional ? this.parent : this).$root.$options.shadowRoot
    );
  } : o), c)
    if (l.functional) {
      l._injectStyles = c;
      var p = l.render;
      l.render = function(d, _) {
        return c.call(_), p(d, _);
      };
    } else {
      var f = l.beforeCreate;
      l.beforeCreate = f ? [].concat(f, c) : [c];
    }
  return {
    exports: t,
    options: l
  };
}
const Rh = {
  name: "KiFormDialog",
  props: {
    show: {
      type: Boolean,
      required: !0
    },
    value: {},
    elFormProps: {},
    retrieve: {},
    submit: {},
    readonly: {
      type: Boolean,
      default: void 0
    },
    loading: {
      type: Boolean,
      default: void 0
    },
    title: {},
    getContainer: {},
    allowClose: {
      type: Boolean,
      default: void 0
    }
  },
  data() {
    return {
      retrieving: !0,
      submitting: !1,
      closing: !1,
      initiated: !1,
      styleTag: null,
      scrollbar: null,
      showConfirmButton: !1,
      beforeCloseIsPassed: !1,
      labelWidth: void 0,
      key: 0
    };
  },
  computed: {
    ValueIsPlainObject() {
      return $n(this.value);
    },
    AllowClose() {
      return y([this.allowClose, I.allowClose, !0]);
    },
    Title() {
      return y([this.title, I.title], {
        type: String
      });
    },
    Listeners() {
      return Ln.call(this, In);
    },
    Loading() {
      return y([this.loading, I.loading, this.retrieving], {
        type: Boolean
      });
    },
    Retrieve() {
      return y([this.retrieve, I.retrieve], {
        type: Function
      });
    },
    Submit() {
      return y([this.submit, I.submit], {
        type: Function
      });
    },
    Readonly() {
      return y([this.readonly, I.readonly, !1], {
        type: Boolean
      });
    },
    ElDialogProps() {
      return y([this.AllowClose ? void 0 : {
        closeOnClickModal: !1,
        showClose: !1,
        closeOnPressEscape: !1
      }, this.$attrs, Fn], {
        default: (t) => (this.beforeCloseIsPassed = Boolean(t.beforeClose), {
          closeOnClickModal: !1,
          ...!this.beforeCloseIsPassed && {
            beforeClose: () => {
              this.$emit("update:show", !1);
            }
          }
        }),
        defaultIsDynamic: !0
      });
    },
    ElFormProps() {
      return y([
        {
          model: this.value,
          ref: "elFormRef"
        },
        this.elFormProps,
        I.elFormProps,
        {
          disabled: this.readonly || this.submitting
        }
      ], {
        type: Object
      });
    },
    GetContainer() {
      return y([this.getContainer, I.getContainer, ["", !0].includes(this.ElDialogProps.appendToBody) ? "body" : void 0], {
        type: [String, Function]
      });
    }
  },
  watch: {
    show: {
      immediate: !0,
      handler(t) {
        var e;
        if (t) {
          this.retrieving = !0;
          const r = (e = this.Retrieve) == null ? void 0 : e.call(this);
          r instanceof Promise ? r.catch((n) => {
            console.error(n), this.close();
          }).finally((n) => {
            this.retrieving = !1;
          }) : this.retrieving = !1, this.computeLabelWidth();
        } else
          this.initiated && (this.closing = !0);
        this.GetContainer && this.$nextTick(() => {
          (typeof this.GetContainer == "function" ? this.GetContainer() : document.querySelector(this.GetContainer)).appendChild(this.$el);
        }), this.initiated = !0;
      }
    },
    Readonly: {
      immediate: !0,
      handler(t) {
        this.closing || (this.showConfirmButton = !t);
      }
    }
  },
  created() {
    this.value__ = It(this.value);
  },
  updated() {
    this.computeLabelWidth();
  },
  unmounted() {
    var t;
    this.GetContainer && ((t = this.$el) == null ? void 0 : t.parentNode) && this.$el.parentNode.removeChild(this.$el);
  },
  methods: {
    computeLabelWidth() {
      const { labelWidth: t, labelPosition: e } = this.ElFormProps;
      e !== "top" && [void 0, "auto"].includes(t) && this.$nextTick(() => {
        var n;
        let r = 0;
        (n = this.$refs.elFormRef) == null || n.$el.querySelectorAll(".el-form-item__label").forEach((o) => {
          const i = o.style.width;
          o.style.width = "unset";
          const a = Math.ceil(parseFloat(window.getComputedStyle(o).width));
          a > r && (r = a), o.style.width = i;
        }), r && (this.labelWidth = `${r}px`);
      });
    },
    onClosed() {
      var t;
      this.submitting = !1, this.$emit("input", It(this.value__)), (t = this.$refs.elFormRef) == null || t.clearValidate(), this.closing = !1, this.showConfirmButton = !this.Readonly, ["", !0].includes(this.ElDialogProps.destroyOnClose) && this.key++;
    },
    close() {
      this.beforeCloseIsPassed ? this.$refs.elDialogRef.beforeClose() : this.$emit("update:show", !1);
    },
    confirm() {
      var e;
      const t = () => {
        if (typeof this.Submit == "function") {
          const r = this.Submit();
          r instanceof Promise ? (this.submitting = !0, r.then((n) => {
            (n == null ? void 0 : n.show) === !0 ? this.submitting = !1 : this.close();
          }).catch((n) => {
            console.error(n), this.submitting = !1;
          })) : (r == null ? void 0 : r.show) !== !0 && this.close();
        } else
          this.close();
      };
      (e = this.$refs.elFormRef) == null || e.validate((r) => {
        r ? t() : this.highlightError(void 0, this.$refs.overlayScrollbar);
      });
    },
    highlightError: Mh
  }
};
var kh = function() {
  var e = this, r = e._self._c;
  return r("el-dialog", e._g(e._b({ key: e.key, ref: "elDialogRef", attrs: { visible: e.show, title: e.Title, destroyOnClose: !1, appendToBody: !1 }, on: { "update:visible": function(n) {
    e.show = n;
  }, closed: e.onClosed }, scopedSlots: e._u([{ key: "title", fn: function() {
    return [e._t("title")];
  }, proxy: !0 }, { key: "footer", fn: function() {
    return [e._t("footer", function() {
      return [e.AllowClose ? r("el-button", { attrs: { disabled: e.closing }, on: { click: e.close } }, [e._v(" " + e._s(e.showConfirmButton ? "\u53D6 \u6D88" : "\u5173 \u95ED") + " ")]) : e._e(), e.showConfirmButton ? r("el-button", { attrs: { type: "primary", disabled: e.closing, loading: e.submitting }, on: { click: e.confirm } }, [e._v(" \u786E \u5B9A ")]) : e._e()];
    }, { close: e.close, closing: e.closing, confirm: e.confirm, submitting: e.submitting })];
  }, proxy: !0 }], null, !0) }, "el-dialog", e.ElDialogProps, !1), e.Listeners), [r("div", { directives: [{ name: "loading", rawName: "v-loading", value: e.Loading, expression: "Loading" }], staticClass: "body", attrs: { overflow: "y-hidden", flex: "~ col" } }, [r("div", { ref: "overlayScrollbar", staticClass: "overflow-y-auto px-40px pb-85px pt-25px", staticStyle: { "max-height": "calc(100vh - 45px)" } }, [e.ValueIsPlainObject ? r("el-form", e._g(e._b({ class: !e.showConfirmButton && "readonly", attrs: { labelWidth: e.labelWidth } }, "el-form", e.ElFormProps, !1), e.Listeners), [e._t("default", null, { elFormRef: e.$refs.elFormRef })], 2) : e._t("default")], 2)])]);
}, Uh = [], Gh = /* @__PURE__ */ Vt(
  Rh,
  kh,
  Uh,
  !1,
  null,
  "23bf0776",
  null,
  null
);
const R = Gh.exports;
let I = {}, Fn = {}, In = {};
R.install = (t, e = {}) => {
  if (R.name) {
    if (R.install.installed) {
      console.warn(`${R.name} has been registered.`);
      return;
    }
  } else
    throw new Error("Name is required for a global component.");
  const { props: r, attrs: n, listeners: o, hooks: i } = Bt(e, R.props);
  I = r, Fn = n, In = o, t.component(R.name, R), R.install.installed = !0;
};
function Bn(t, e, r) {
  this.$children.forEach((n) => {
    var o = n.$options.componentName;
    o === t ? n.$emit.apply(n, [e].concat(r)) : Bn.apply(n, [t, e].concat([r]));
  });
}
const Nh = {
  methods: {
    dispatch(t, e, r) {
      for (var n = this.$parent || this.$root, o = n.$options.componentName; n && (!o || o !== t); )
        n = n.$parent, n && (o = n.$options.componentName);
      n && n.$emit.apply(n, [e].concat(r));
    },
    broadcast(t, e, r) {
      Bn.call(this, t, e, r);
    }
  }
};
let Ot;
const zh = new Uint8Array(16);
function Vh() {
  if (!Ot && (Ot = typeof crypto < "u" && crypto.getRandomValues && crypto.getRandomValues.bind(crypto), !Ot))
    throw new Error("crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported");
  return Ot(zh);
}
const P = [];
for (let t = 0; t < 256; ++t)
  P.push((t + 256).toString(16).slice(1));
function Wh(t, e = 0) {
  return (P[t[e + 0]] + P[t[e + 1]] + P[t[e + 2]] + P[t[e + 3]] + "-" + P[t[e + 4]] + P[t[e + 5]] + "-" + P[t[e + 6]] + P[t[e + 7]] + "-" + P[t[e + 8]] + P[t[e + 9]] + "-" + P[t[e + 10]] + P[t[e + 11]] + P[t[e + 12]] + P[t[e + 13]] + P[t[e + 14]] + P[t[e + 15]]).toLowerCase();
}
const Hh = typeof crypto < "u" && crypto.randomUUID && crypto.randomUUID.bind(crypto), Sr = {
  randomUUID: Hh
};
function Zt(t, e, r) {
  if (Sr.randomUUID && !e && !t)
    return Sr.randomUUID();
  t = t || {};
  const n = t.random || (t.rng || Vh)();
  if (n[6] = n[6] & 15 | 64, n[8] = n[8] & 63 | 128, e) {
    r = r || 0;
    for (let o = 0; o < 16; ++o)
      e[r + o] = n[o];
    return e;
  }
  return Wh(n);
}
const Kh = {
  name: "KiSelect",
  mixins: [Nh],
  inject: {
    elForm: {
      default: {}
    }
  },
  props: {
    value: {},
    label: {},
    options: {
      type: Array
    },
    props: {},
    ellipsis: {
      type: Boolean,
      default: void 0
    },
    allowSelectAll: {
      type: Boolean,
      default: void 0
    },
    search: {},
    searchImmediately: {
      type: Boolean,
      default: void 0
    }
  },
  data() {
    return {
      value__: this.value,
      initialValue: void 0,
      popper: null,
      unwatchOptions: null,
      loading: void 0,
      options__: [],
      optionGroupPropsList: [],
      optionPropsList: [],
      allSelected: !1,
      indeterminate: !1,
      valueInitializedWhenMultiple: !1,
      previousQuery: null
    };
  },
  computed: {
    AllowSelectAll() {
      return y([this.allowSelectAll, X.allowSelectAll, !0]);
    },
    Listeners() {
      return Ln.call(this, Mn);
    },
    grouped() {
      return L(this.Props.groupOptions);
    },
    itemTypeIsJSON() {
      var t;
      return typeof ((t = this.options__) == null ? void 0 : t[0]) == "object";
    },
    valueComesFromObject() {
      return F(this.Props.value) || this.valueType === "function" ? !1 : this.itemTypeIsJSON;
    },
    ScopedSlots() {
      const t = {};
      for (const e in this.$slots)
        e !== "default" && (t[e] = this.$slots[e]);
      return t;
    },
    ElSelectProps() {
      const t = Boolean(this.Search), e = t ? "\u641C\u7D22" : "\u8BF7\u9009\u62E9";
      return y([this.$attrs, Dn, {
        clearable: !0,
        filterable: !0,
        remote: t,
        reserveKeyword: !0,
        remoteMethod: this.remoteMethod,
        valueKey: this.valueComesFromObject ? this.Props.value : void 0,
        loading: this.loading,
        placeholder: e
      }], {
        type: Object
      });
    },
    Ellipsis() {
      var e;
      y([this.ellipsis, X.ellipsis, !1], {
        type: Boolean
      }) ? this.$nextTick(() => {
        this.popper = this.$refs.elSelect.$refs.popper, this.unwatchOptions = this.$watch("options__", (r) => {
          r && r.length && setTimeout(() => {
            this.popper && (this.popper.$el.style.maxWidth = this.popper.minWidth);
          });
        }, {
          immediate: !0
        });
      }) : (e = this.unwatchOptions) == null || e.call(this);
    },
    valueType() {
      return this.validateProps("value");
    },
    labelType() {
      return this.validateProps("label");
    },
    labelRightType() {
      return this.validateProps("labelRight");
    },
    disabledType() {
      return this.validateProps("disabled");
    },
    groupLabelType() {
      return this.validateProps("groupLabel");
    },
    groupDisabledType() {
      return this.validateProps("groupDisabled");
    },
    groupOptionsType() {
      return this.validateProps("groupOptions");
    },
    Props() {
      return y([
        this.props,
        X.props,
        {
          disabled: "disabled",
          groupDisabled: "disabled"
        }
      ], {
        type: Object
      });
    },
    Search() {
      return y([this.search, X.search], {
        type: Function
      });
    },
    SearchImmediately() {
      return y([this.searchImmediately, X.searchImmediately, !0], {
        type: Boolean
      });
    },
    isMultiple() {
      return [!0, ""].includes(this.ElSelectProps.multiple);
    }
  },
  watch: {
    value: {
      immediate: !0,
      handler(t, e) {
        this.value__ = t, this.showLabel(), this.syncSelectAllBtn(t);
      }
    },
    value__: {
      handler(t, e) {
        if (this.isMultiple) {
          if (!this.valueInitializedWhenMultiple)
            return;
          this.valueInitializedWhenMultiple = !0;
        }
        F(t) && (this.$emit("update:index", void 0), this.remoteMethod());
      }
    },
    options: {
      immediate: !0,
      handler(t, e) {
        this.setOptions__(t);
      }
    }
  },
  created() {
    this.SearchImmediately && this.remoteMethod(void 0, !0);
  },
  mounted() {
    this.initialValue = It(this.value), this.dispatch("ElForm", "el.form.addField", [this]);
  },
  methods: {
    showLabel() {
      this.label && !this.isMultiple && this.$nextTick(() => {
        this.$refs.elSelect.selected instanceof Ar || (this.$refs.elSelect.selectedLabel = this.label);
      });
    },
    onVisibleChange(t) {
      t || (this.showLabel(), F(this.value__) && this.previousQuery && setTimeout(() => {
        this.remoteMethod();
      }, 100));
    },
    setOptions__(t) {
      y([t], { type: Array }), this.options__ = t || [], this.grouped ? this.optionGroupPropsList = Array.from(t || [], (e, r) => {
        const n = this.getGroupOptions(e, r);
        return {
          key: Zt(),
          label: this.getGroupLabel(e, r),
          disabled: this.isGroupDisabled(e, r),
          options: n,
          optionPropsList: Array.from(n || [], (o, i) => ({
            key: Zt(),
            value: this.getValue(o, i),
            label: this.getLabel(o, i),
            labelRight: this.getLabelRight(o, i),
            disabled: this.isDisabled(o, i)
          }))
        };
      }) : this.optionPropsList = Array.from(t || [], (e, r) => ({
        key: Zt(),
        value: this.getValue(e, r),
        label: this.getLabel(e, r),
        labelRight: this.getLabelRight(e, r),
        disabled: this.isDisabled(e, r)
      })), L(t) && this.showLabel(), this.$emit("update:options", t);
    },
    selectAll() {
      if (this.allSelected) {
        const t = [];
        this.options__.map((e, r) => {
          this.isDisabled(e, r) || t.push(this.getValue(e, r));
        }), this.value__ = t;
      } else
        this.value__ = [];
      this.onChange(this.value__), this.$emit("input", this.value__);
    },
    resetField() {
      const t = It(this.initialValue);
      this.value__ = t, this.onChange(t);
    },
    validate(t, e) {
      e();
    },
    clearValidate() {
    },
    getRules() {
      return [];
    },
    getFilteredRule() {
      return [];
    },
    validateProps(t) {
      return y([this.Props[t]], {
        type: [Boolean, Symbol, String, Number, Function]
      }), typeof this.Props[t];
    },
    onOptionClick(t, e) {
      this.$emit("update:index", e);
    },
    remoteMethod(t, e = !1) {
      if (!this.Search)
        return;
      this.loading = !0, this.previousQuery = t;
      const r = this.Search(t, e);
      r instanceof Promise ? r.then((n) => {
        this.setOptions__(n);
      }).finally(() => {
        this.loading = !1;
      }) : (this.setOptions__(r), this.loading = !1);
    },
    syncSelectAllBtn(t) {
      if (this.isMultiple && !this.grouped) {
        const e = t ? t.length : 0, r = this.options__.length;
        this.allSelected = e > 0 && e === r, this.indeterminate = e > 0 && e < r;
      }
    },
    onChange(t) {
      this.syncSelectAllBtn(t), this.$nextTick(() => {
        this.$emit("update:label", this.$refs.elSelect.selectedLabel);
      });
    },
    getValue(t, e) {
      let r = t;
      if (this.valueType === "function")
        r = this.Props.value(t, e);
      else if (this.itemTypeIsJSON)
        if (L(this.Props.value))
          r = t == null ? void 0 : t[this.Props.value];
        else {
          if (F(this.ElSelectProps.valueKey))
            throw new Error("'value-key' of 'el-select' is required when binding value is an object.");
          if (L(this.value) && typeof this.value != "object")
            throw new Error("Binding value must be an object when 'options' is an object[] and 'props.value' is unset.");
        }
      return r;
    },
    getLabel(t, e) {
      let r = t;
      return this.labelType === "function" ? r = this.Props.label(t, e) : this.itemTypeIsJSON && (L(this.Props.label) ? r = t == null ? void 0 : t[this.Props.label] : r = JSON.stringify(t)), F(r) ? "" : String(r);
    },
    getLabelRight(t, e) {
      let r;
      return this.labelRightType === "function" ? r = this.Props.labelRight(t, e) : this.itemTypeIsJSON && L(this.Props.labelRight) && (r = t == null ? void 0 : t[this.Props.labelRight]), F(r) ? "" : String(r);
    },
    getGroupLabel(t, e) {
      let r = t;
      return this.groupLabelType === "function" ? r = this.Props.groupLabel(t, e) : this.itemTypeIsJSON && (L(this.Props.groupLabel) ? r = t == null ? void 0 : t[this.Props.groupLabel] : r = JSON.stringify(t)), F(r) ? "" : String(r);
    },
    isDisabled(t, e) {
      let r = !1;
      return this.disabledType === "function" ? r = this.Props.disabled(t, e) : this.itemTypeIsJSON && L(this.Props.disabled) && (r = t == null ? void 0 : t[this.Props.disabled]), Boolean(r);
    },
    isGroupDisabled(t, e) {
      let r = !1;
      return this.groupDisabledType === "function" ? r = this.Props.groupDisabled(t, e) : this.itemTypeIsJSON && L(this.Props.groupDisabled) && (r = t == null ? void 0 : t[this.Props.groupDisabled]), Boolean(r);
    },
    getGroupOptions(t, e) {
      let r;
      return this.groupOptionsType === "function" ? r = this.Props.groupOptions(t, e) : this.itemTypeIsJSON && (r = t == null ? void 0 : t[this.Props.groupOptions]), F(r) ? [] : Array.isArray(r) ? r : (console.warn("groupOptions \u4EC5\u80FD\u4E3A any[] \u7C7B\u578B"), []);
    }
  }
};
var qh = function() {
  var e = this, r = e._self._c;
  return r("el-select", e._g(e._b({ ref: "elSelect", on: { change: e.onChange, "visible-change": e.onVisibleChange }, scopedSlots: e._u([e._l(e.ScopedSlots, function(n, o) {
    return { key: o, fn: function() {
      return [e._t(o)];
    }, proxy: !0 };
  })], null, !0), model: { value: e.value__, callback: function(n) {
    e.value__ = n;
  }, expression: "value__" } }, "el-select", e.ElSelectProps, !1), e.Listeners), [e.grouped ? e._l(e.options__, function(n, o) {
    return r("el-option-group", { key: e.optionGroupPropsList[o].key, attrs: { label: e.optionGroupPropsList[o].label, disabled: e.optionGroupPropsList[o].disabled } }, e._l(e.optionGroupPropsList[o].options, function(i, a) {
      return r("el-option", { key: e.optionGroupPropsList[o].optionPropsList[a].key, attrs: { label: e.optionGroupPropsList[o].optionPropsList[a].label, value: e.optionGroupPropsList[o].optionPropsList[a].value, disabled: e.optionGroupPropsList[o].optionPropsList[a].disabled }, nativeOn: { click: function(s) {
        e.optionGroupPropsList[o].optionPropsList[a].disabled || e.onOptionClick(n, o);
      } } }, [e.$slots.default ? e._t("default", null, { option: i, index: a }) : [r("el-tooltip", { attrs: { disabled: !e.Ellipsis, effect: "dark", placement: "right", content: e.optionGroupPropsList[o].optionPropsList[a].label } }, [r("span", { staticClass: "label-left" }, [e._v(e._s(e.optionGroupPropsList[o].optionPropsList[a].label))])]), r("span", { staticClass: "label-right" }, [e._v(e._s(e.optionGroupPropsList[o].optionPropsList[a].labelRight))])]], 2);
    }), 1);
  }) : [e.AllowSelectAll && e.isMultiple && e.options__.length > 1 ? r("el-checkbox", { staticClass: "px-20px py-10px", attrs: { indeterminate: e.indeterminate }, on: { change: e.selectAll }, model: { value: e.allSelected, callback: function(n) {
    e.allSelected = n;
  }, expression: "allSelected" } }, [e._v(" \u5168\u9009 ")]) : e._e(), e._l(e.options__, function(n, o) {
    return r("el-option", { key: e.optionPropsList[o].key, attrs: { label: e.optionPropsList[o].label, value: e.optionPropsList[o].value, disabled: e.optionPropsList[o].disabled }, nativeOn: { click: function(i) {
      e.optionPropsList[o].disabled || e.onOptionClick(n, o);
    } } }, [e.$slots.default ? e._t("default", null, { option: n, index: o }) : [r("el-tooltip", { attrs: { disabled: !e.Ellipsis, effect: "dark", placement: "right", content: e.optionPropsList[o].label } }, [r("span", { staticClass: "label-left" }, [e._v(e._s(e.optionPropsList[o].label))])]), r("span", { staticClass: "label-right" }, [e._v(e._s(e.optionPropsList[o].labelRight))])]], 2);
  })]], 2);
}, Jh = [], Zh = /* @__PURE__ */ Vt(
  Kh,
  qh,
  Jh,
  !1,
  null,
  null,
  null,
  null
);
const k = Zh.exports;
let X = {}, Dn = {}, Mn = {};
k.install = (t, e = {}) => {
  if (k.name) {
    if (k.install.installed) {
      console.warn(`${k.name} has been registered.`);
      return;
    }
  } else
    throw new Error("Name is required for a global component.");
  const { props: r, attrs: n, listeners: o, hooks: i } = Bt(e, k.props);
  X = r, Dn = n, Mn = o, t.component(k.name, k), k.install.installed = !0;
};
const Qh = {
  name: "KiPopButton",
  props: {
    elPopconfirmProps: {},
    elTooltipProps: {},
    elPopoverProps: {}
  },
  computed: {
    ElButtonProps() {
      return y([this.$attrs, Rn]);
    },
    ElPopoverProps() {
      const t = y([
        this.elPopoverProps,
        At.elPopoverProps
      ], {
        type: Object
      }), { title: e, content: r } = t || {};
      return {
        disabled: !(e || r),
        ...t
      };
    },
    ElPopconfirmProps() {
      const t = y([
        this.elPopconfirmProps,
        At.elPopconfirmProps
      ], {
        type: Object
      });
      return {
        popperClass: "pop-button-popper",
        disabled: !(t != null && t.title),
        ...t
      };
    },
    ElTooltipProps() {
      const t = y([
        this.elTooltipProps,
        At.elTooltipProps
      ], {
        type: Object
      });
      return {
        disabled: !(t != null && t.content),
        ...t
      };
    }
  },
  methods: {
    onClick(t) {
      this.$refs.elTooltip.manual || (this.$refs.elTooltip.showPopper = !1), this.ElPopconfirmProps.disabled && this.$emit("click", t);
    }
  }
};
var Xh = function() {
  var e = this, r = e._self._c;
  return r("el-tooltip", e._b({ ref: "elTooltip", staticClass: "pop-button" }, "el-tooltip", e.ElTooltipProps, !1), [r("div", { attrs: { slot: "content" }, domProps: { innerHTML: e._s(e.ElTooltipProps.content) }, slot: "content" }), r("el-popover", e._b({}, "el-popover", e.ElPopoverProps, !1), [r("div", { domProps: { innerHTML: e._s(e.ElPopoverProps.content) } }), r("el-popconfirm", e._b({ attrs: { slot: "reference" }, on: { confirm: function(n) {
    return e.$emit("click", n);
  }, onConfirm: function(n) {
    return e.$emit("click", n);
  } }, slot: "reference" }, "el-popconfirm", e.ElPopconfirmProps, !1), [r("el-button", e._b({ attrs: { slot: "reference" }, on: { click: e.onClick }, slot: "reference" }, "el-button", e.ElButtonProps, !1), [e._t("default")], 2)], 1)], 1)], 1);
}, Yh = [], td = /* @__PURE__ */ Vt(
  Qh,
  Xh,
  Yh,
  !1,
  null,
  "4aca3107",
  null,
  null
);
const U = td.exports;
let At = {}, Rn = {};
U.install = (t, e = {}) => {
  if (U.name) {
    if (U.install.installed) {
      console.warn(`${U.name} has been registered.`);
      return;
    }
  } else
    throw new Error("Name is required for a global component.");
  const { props: r, attrs: n, listeners: o, hooks: i } = Bt(e, U.props);
  At = r, Rn = n, t.component(U.name, U), U.install.installed = !0;
};
const ed = {
  name: "KiPopSwitch",
  model: {
    prop: "value",
    event: "change"
  },
  props: {
    value: {},
    inlinePrompt: {
      type: Boolean,
      default: void 0
    },
    elPopconfirmProps: {},
    elTooltipProps: {},
    elPopoverProps: {}
  },
  computed: {
    InlinePrompt() {
      return y([this.inlinePrompt, ft.inlinePrompt, !0], {
        type: Boolean
      });
    },
    ElSwitchProps() {
      return y([this.$attrs, kn], {
        default: (t) => {
          let e = 0;
          return ["active-text", "inactive-text", "activeText", "inactiveText"].map((r) => {
            const n = Bh(t[r]);
            n > e && (e = n);
          }), {
            ...this.InlinePrompt && { width: 30 + e * 6 },
            ...t
          };
        },
        defaultIsDynamic: !0
      });
    },
    ElPopoverProps() {
      return y([
        this.elPopoverProps,
        ft.elPopoverProps
      ], {
        type: Object,
        default: (t) => ({
          popperClass: "pop-switch",
          disabled: !(t && (t.title || t.content))
        }),
        defaultIsDynamic: !0
      });
    },
    ElPopconfirmProps() {
      return y([
        this.elPopconfirmProps,
        ft.elPopconfirmProps
      ], {
        type: Object,
        default: (t) => ({
          popperClass: "pop-switch",
          disabled: [!0, ""].includes(this.ElSwitchProps.disabled) || !(t != null && t.title)
        }),
        defaultIsDynamic: !0
      });
    },
    ElTooltipProps() {
      return y([
        this.elTooltipProps,
        ft.elTooltipProps
      ], {
        type: Object,
        default: (t) => ({
          disabled: !((t == null ? void 0 : t.content) || this.$slots.elTooltipContent)
        }),
        defaultIsDynamic: !0
      });
    }
  },
  methods: {
    onConfirm() {
      const { checked: t, inactiveValue: e, activeValue: r } = this.$refs.elSwitch;
      this.$emit("change", t ? e : r);
    },
    onClick() {
      this.$refs.elTooltip.manual || (this.$refs.elTooltip.showPopper = !1), this.ElSwitchProps.disabled === !1 && this.ElPopconfirmProps.disabled && this.onConfirm();
    }
  }
};
var rd = function() {
  var e = this, r = e._self._c;
  return r("el-tooltip", e._b({ ref: "elTooltip" }, "el-tooltip", e.ElTooltipProps, !1), [r("div", { attrs: { slot: "content" }, domProps: { innerHTML: e._s(e.ElTooltipProps.content) }, slot: "content" }), r("el-popover", e._b({}, "el-popover", e.ElPopoverProps, !1), [r("div", { domProps: { innerHTML: e._s(e.ElPopoverProps.content) } }), r("el-popconfirm", e._b({ attrs: { slot: "reference" }, on: { confirm: e.onConfirm, onConfirm: e.onConfirm }, slot: "reference" }, "el-popconfirm", e.ElPopconfirmProps, !1), [r("el-switch", e._b({ ref: "elSwitch", class: e.InlinePrompt && "text-inside", attrs: { slot: "reference", value: e.value }, nativeOn: { click: function(n) {
    return e.onClick.apply(null, arguments);
  } }, slot: "reference" }, "el-switch", e.ElSwitchProps, !1))], 1)], 1)], 1);
}, nd = [], od = /* @__PURE__ */ Vt(
  ed,
  rd,
  nd,
  !1,
  null,
  "a05c355d",
  null,
  null
);
const G = od.exports;
let ft = {}, kn = {};
G.install = (t, e = {}) => {
  if (G.name) {
    if (G.install.installed) {
      console.warn(`${G.name} has been registered.`);
      return;
    }
  } else
    throw new Error("Name is required for a global component.");
  const { props: r, attrs: n, listeners: o, hooks: i } = Bt(e, G.props);
  ft = r, kn = n, t.component(G.name, G), G.install.installed = !0;
};
export {
  R as FormDialog,
  U as PopButton,
  G as PopSwitch,
  k as Select
};
